<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Michael Noe" />

<meta name="date" content="2024-03-10" />

<title>Figure2.Rmd</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">noe2024</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure2.Rmd</h1>
<h4 class="author">Michael Noe</h4>
<h4 class="date">2024-03-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>noe2024/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024tree9ff0bc6401db701ba9eeb4433ba4a85568a190f2targetblank9ff0bc6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/cancer-genomics/noe2024/tree/9ff0bc6401db701ba9eeb4433ba4a85568a190f2" target="_blank">9ff0bc6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024tree9ff0bc6401db701ba9eeb4433ba4a85568a190f2targetblank9ff0bc6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/cancer-genomics/noe2024/tree/9ff0bc6401db701ba9eeb4433ba4a85568a190f2" target="_blank">9ff0bc6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/results/

Untracked files:
    Untracked:  analysis/temp3.Rmd
    Untracked:  analysis/tempS10.Rmd
    Untracked:  code/Pre_Supplementary_Figure10.Rmd
    Untracked:  code/Pre_Supplementary_FigureX2.Rmd
    Untracked:  code/Pre_Supplementary_FigureX3.Rmd
    Untracked:  code/Supplementary_Table1.Rmd
    Untracked:  code/Supplementary_Table2.Rmd
    Untracked:  code/Supplementary_Table4.Rmd
    Untracked:  code/Supplementary_Table5.Rmd
    Untracked:  code/Supplementary_Table6.Rmd
    Untracked:  code/Supplementary_Table7.Rmd
    Untracked:  code/Supplementary_Table9.Rmd
    Untracked:  code/cpg_500000_topbot1000_beta.R
    Untracked:  code/healthy_cpggroups.R
    Untracked:  code/healthy_cpggroups.sh
    Untracked:  code/tss_500000_topbot1000_beta.R
    Untracked:  data/Figure_2B_stats.rds
    Untracked:  data/Figure_2C_stats.rds
    Untracked:  data/carvalho_motifs_ML.rds
    Untracked:  data/cpg_500000_topbot1000_beta2.rds
    Untracked:  data/mouse_permutations_mean.rds
    Untracked:  data/mouse_permutations_median.rds
    Untracked:  data/tss_500000_topbot1000_beta.rds

Unstaged changes:
    Modified:   analysis/Figure2.Rmd
    Modified:   analysis/Figure3.Rmd
    Modified:   analysis/Supplementary_Figure10.Rmd
    Modified:   analysis/Supplementary_Figure4.Rmd
    Modified:   analysis/Supplementary_Figure5.Rmd
    Deleted:    analysis/Supplementary_Table7.Rmd
    Deleted:    analysis/Supplementary_Table9.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    code/SuppTable1.Rmd
    Deleted:    code/SuppTable2.Rmd
    Deleted:    code/SuppTable4.Rmd
    Deleted:    code/SuppTable5.Rmd
    Deleted:    code/SuppTable6.Rmd
    Modified:   data/mouse_frag.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Figure2.Rmd</code>) and HTML
(<code>docs/Figure2.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/8f3f870aa5437d3d6f5c79d921e7f225b4514773/analysis/Figure2.Rmd" target="_blank">8f3f870</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-06
</td>
<td>
Correct stat table in Fig 2DE
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/e2712c2174370e09a641030a5487b6a18338704f/analysis/Figure2.Rmd" target="_blank">e2712c2</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-06
</td>
<td>
Reorder figures: add S4,S6,S13
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/e2712c2174370e09a641030a5487b6a18338704f/docs/Figure2.html" target="_blank">e2712c2</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-06
</td>
<td>
Reorder figures: add S4,S6,S13
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/9d3f606bb5ef1c1a9c3c587cddd0f2250b76198f/docs/Figure2.html" target="_blank">9d3f606</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-24
</td>
<td>
Add Supp Fig 9, reorder Fig 3 figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/dc39d3fb10aa7b5335c6182588b4ef4a906fb458/analysis/Figure2.Rmd" target="_blank">dc39d3f</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-21
</td>
<td>
Add supplementary figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/dc39d3fb10aa7b5335c6182588b4ef4a906fb458/docs/Figure2.html" target="_blank">dc39d3f</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-21
</td>
<td>
Add supplementary figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/e1f36dcdb578c47e206926e77af40fe0d2c7053d/analysis/Figure2.Rmd" target="_blank">e1f36dc</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-15
</td>
<td>
Add supp. figs and tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/e1f36dcdb578c47e206926e77af40fe0d2c7053d/docs/Figure2.html" target="_blank">e1f36dc</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-15
</td>
<td>
Add supp. figs and tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/de3561665de4eb4a67334ac71cee69954c0f14f5/analysis/Figure2.Rmd" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/de3561665de4eb4a67334ac71cee69954c0f14f5/docs/Figure2.html" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="pre-requisites" class="section level2">
<h2>Pre-requisites</h2>
<p>These packages are necessary to process the data to make these
figures</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)
library(&quot;annotatr&quot;)
library(&quot;cowplot&quot;)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;

The following object is masked from &#39;package:lubridate&#39;:

    stamp</code></pre>
<pre class="r"><code>library(&quot;data.table&quot;)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last

The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(&quot;GenomicRanges&quot;)</code></pre>
<pre><code>Loading required package: stats4
Loading required package: BiocGenerics

Attaching package: &#39;BiocGenerics&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    intersect, setdiff, union

The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union

The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs

The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors

Attaching package: &#39;S4Vectors&#39;

The following objects are masked from &#39;package:data.table&#39;:

    first, second

The following objects are masked from &#39;package:lubridate&#39;:

    second, second&lt;-

The following objects are masked from &#39;package:dplyr&#39;:

    first, rename

The following object is masked from &#39;package:tidyr&#39;:

    expand

The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: &#39;IRanges&#39;

The following object is masked from &#39;package:data.table&#39;:

    shift

The following object is masked from &#39;package:lubridate&#39;:

    %within%

The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice

The following object is masked from &#39;package:purrr&#39;:

    reduce

Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>library(&quot;grid&quot;)
library(&quot;ggpubr&quot;)</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;

The following object is masked from &#39;package:cowplot&#39;:

    get_legend</code></pre>
<pre class="r"><code>library(&quot;tidymodels&quot;)</code></pre>
<pre><code>── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──
✔ broom        1.0.5     ✔ rsample      1.2.0
✔ dials        1.2.0     ✔ tune         1.1.2
✔ infer        1.0.6     ✔ workflows    1.1.3
✔ modeldata    1.3.0     ✔ workflowsets 1.0.1
✔ parsnip      1.1.1     ✔ yardstick    1.3.0
✔ recipes      1.0.9     
── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ data.table::between()    masks dplyr::between()
✖ IRanges::collapse()      masks dplyr::collapse()
✖ BiocGenerics::combine()  masks dplyr::combine()
✖ IRanges::desc()          masks dplyr::desc()
✖ scales::discard()        masks purrr::discard()
✖ S4Vectors::expand()      masks tidyr::expand()
✖ dplyr::filter()          masks stats::filter()
✖ S4Vectors::first()       masks data.table::first(), dplyr::first()
✖ recipes::fixed()         masks stringr::fixed()
✖ dplyr::lag()             masks stats::lag()
✖ data.table::last()       masks dplyr::last()
✖ BiocGenerics::Position() masks ggplot2::Position(), base::Position()
✖ GenomicRanges::reduce()  masks IRanges::reduce(), purrr::reduce()
✖ S4Vectors::rename()      masks dplyr::rename()
✖ IRanges::slice()         masks dplyr::slice()
✖ yardstick::spec()        masks readr::spec()
✖ recipes::step()          masks stats::step()
✖ data.table::transpose()  masks purrr::transpose()
✖ recipes::update()        masks GenomicRanges::update(), stats4::update(), stats::update()
• Dig deeper into tidy modeling with R at https://www.tmwr.org</code></pre>
<pre class="r"><code>library(&quot;knitr&quot;)
library(&quot;parallel&quot;)
library(&quot;foreach&quot;)</code></pre>
<pre><code>
Attaching package: &#39;foreach&#39;

The following objects are masked from &#39;package:purrr&#39;:

    accumulate, when</code></pre>
<pre class="r"><code>library(&quot;doRNG&quot;)</code></pre>
<pre><code>Loading required package: rngtools</code></pre>
<pre class="r"><code>library(&quot;here&quot;)</code></pre>
<pre><code>here() starts at /Users/skoul/Documents/Projects/cfepi_wflow/noe2024</code></pre>
</div>
<div id="figure-2a" class="section level2">
<h2>Figure 2A</h2>
<pre class="r"><code>t &lt;- readRDS(here(&quot;data/position_in_read.rds&quot;))
t_normalizer &lt;- unique(t[,c(5,6)])
t_normalizer$normal &lt;- t_normalizer$somatic_all / mean(t_normalizer$somatic_all)
t$normal &lt;- t_normalizer[match(t$case, t_normalizer$case),]$normal
t$n &lt;- t$n / t$normal
t_b_small &lt;- tibble()
for (i in c(0.1, 0.2, 0.3, 0.4, 0.5)) {
  t_b_small_temp &lt;- t[which(t$beta &lt;= i),] %&gt;% 
    group_by(diff, case, motif) %&gt;%
    summarise(n = sum(n))
  t_b_small_temp$beta &lt;- i
  t_b_small &lt;- rbind(t_b_small, t_b_small_temp)
}</code></pre>
<pre><code>`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>#t_b_small$beta &lt;- paste0(&quot;&lt;= &quot;, t_b_small$beta)
t_b_large &lt;- tibble()
for (i in c(0.5, 0.6, 0.7, 0.8, 0.9)) {
  t_b_large_temp &lt;- t[which(t$beta &gt;= i),] %&gt;% 
    group_by(diff, case, motif) %&gt;%
    summarise(n = sum(n))
  t_b_large_temp$beta &lt;- 1 - i
  t_b_large &lt;- rbind(t_b_large, t_b_large_temp)
}</code></pre>
<pre><code>`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.
`summarise()` has grouped output by &#39;diff&#39;, &#39;case&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>#t_b_large$beta &lt;- paste0(&quot;&gt;= &quot;, t_b_large$beta)


t_b_small$group &lt;- &quot;unmethylated&quot;
t_b_large$group &lt;- &quot;methylated&quot;

t_b &lt;- rbind(t_b_small, t_b_large)
t_b$beta &lt;- factor(t_b$beta, levels = c(0.1, 0.2, 0.3, 0.4, 0.5))


for (i in unique(t_b$beta)) {
  t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CG&quot;) &amp; (t_b$group == &quot;unmethylated&quot;)),]$n &lt;- t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CG&quot;) &amp; (t_b$group == &quot;unmethylated&quot;)),]$n * (mean(t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CG&quot;) &amp; (t_b$group == &quot;methylated&quot;)),]$n) / mean(t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CG&quot;) &amp; (t_b$group == &quot;unmethylated&quot;)),]$n))
    t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CCG&quot;) &amp; (t_b$group == &quot;unmethylated&quot;)),]$n &lt;- t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CCG&quot;) &amp; (t_b$group == &quot;unmethylated&quot;)),]$n * (mean(t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CCG&quot;) &amp; (t_b$group == &quot;methylated&quot;)),]$n) / mean(t_b[which((t_b$beta == i) &amp; (t_b$motif == &quot;CCG&quot;) &amp; (t_b$group == &quot;unmethylated&quot;)),]$n))
}

t_b.summary &lt;- t_b %&gt;%
  group_by(diff, beta, motif, group) %&gt;%
  summarise(
    sd = sd(n, na.rm = TRUE),
    n = mean(n)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;diff&#39;, &#39;beta&#39;, &#39;motif&#39;. You can override
using the `.groups` argument.</code></pre>
<pre class="r"><code>t_b.summary$beta &lt;- factor(t_b.summary$beta, levels = c(0.1, 0.2, 0.3, 0.4, 0.5))
t_b$group_beta &lt;- paste0(t_b$group, &quot;_&quot;, t_b$beta)
t_b.summary$group_beta &lt;- paste0(t_b.summary$group, &quot;_&quot;, t_b.summary$beta)
t_b$beta &lt;- as.numeric(as.character(t_b$beta))
t_b$beta &lt;- paste0(&quot;methylated: beta &gt;= &quot;, 1 - t_b$beta, &quot;;\nunmethylated: beta &lt;= &quot;, t_b$beta)
t_b.summary$beta &lt;- as.numeric(as.character(t_b.summary$beta))
t_b.summary$beta &lt;- paste0(&quot;methylated: beta &gt;= &quot;, 1 - t_b.summary$beta, &quot;;\nunmethylated: beta &lt;= &quot;, t_b.summary$beta)

plot_cg &lt;- ggplot(t_b[which((t_b$diff &gt;= 1) &amp; (t_b$diff &lt;= 40) &amp; (t_b$motif == &quot;CG&quot;)),], aes(x=diff, y=n, colour=group_beta, fill=group_beta)) +
  geom_point(position=position_jitter(h=0.01, w=0.4), alpha = 1, size = 0.1) +
  geom_bar(stat = &quot;identity&quot;, data = t_b.summary[which((t_b.summary$diff&gt;= 1) &amp; (t_b.summary$diff &lt;= 40) &amp; (t_b.summary$motif == &quot;CG&quot;)),], colour=&quot;black&quot;, alpha=0.7) +
   geom_errorbar(
    aes(ymin = n-sd, ymax = n+sd),
    data = t_b.summary[which((t_b.summary$diff&gt;= 1) &amp; (t_b.summary$diff &lt;= 40) &amp; (t_b.summary$motif == &quot;CG&quot;)),], width = 0.2, colour=&quot;black&quot;) +
  facet_grid(rows = vars(beta),
             cols = vars(group),
             scales = &quot;free_y&quot;,
             space = &quot;fixed&quot;) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=16),
        axis.title.y = element_text(margin = margin(l = 20, r = 20)),
        axis.title.x = element_text(margin = margin(t = 20, b = 40)),
        #plot.title = element_text(hjust = 0.5, margin = margin(t = 20, b = 20)),
        strip.background=element_rect(color = NA),
        panel.spacing=unit(2, &quot;lines&quot;),
        strip.placement = &quot;top&quot;,
        legend.position = &quot;none&quot;,
        strip.text.y.right = element_text(angle = 0)) +
  scale_fill_manual(values=c(&quot;#004766&quot;, &quot;#255672&quot;, &quot;#3d667e&quot;, &quot;#53768A&quot;, &quot;#698696&quot;, &quot;#77c8dd&quot;, &quot;#93cfe3&quot;, &quot;#acd7e7&quot;, &quot;#c2deeb&quot;, &quot;#d7e6ee&quot;)) +
  scale_colour_manual(values=c(&quot;#004766&quot;, &quot;#255672&quot;, &quot;#3d667e&quot;, &quot;#53768A&quot;, &quot;#698696&quot;, &quot;#77c8dd&quot;, &quot;#93cfe3&quot;, &quot;#acd7e7&quot;, &quot;#c2deeb&quot;, &quot;#d7e6ee&quot;)) +
  ylab(&quot;&quot;) +
  xlab(&quot;Position in fragment&quot;) +
  #ggtitle(&quot;Position of \&quot;CG\&quot; in the fragment, based on methylation-status of that \&quot;CG\&quot;.&quot;) +
  ggtitle(&quot;&quot;)

#ggsave(here(&quot;docs/figure/Supplementary_Fig_2.jpg&quot;), plot = plot_cg, device = &quot;jpeg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=300)

plot_ccg &lt;- ggplot(t_b[which((t_b$diff &gt;= 1) &amp; (t_b$diff &lt;= 40) &amp; (t_b$motif == &quot;CCG&quot;)),], aes(x=diff, y=n, colour=group_beta, fill=group_beta)) +
  geom_point(position=position_jitter(h=0.01, w=0.4), alpha = 1, size = 0.1) +
  geom_bar(stat = &quot;identity&quot;, data = t_b.summary[which((t_b.summary$diff&gt;= 1) &amp; (t_b.summary$diff &lt;= 40) &amp; (t_b.summary$motif == &quot;CCG&quot;)),], colour=&quot;black&quot;, alpha=0.7) +
   geom_errorbar(
    aes(ymin = n-sd, ymax = n+sd),
    data = t_b.summary[which((t_b.summary$diff&gt;= 1) &amp; (t_b.summary$diff &lt;= 40) &amp; (t_b.summary$motif == &quot;CCG&quot;)),], width = 0.2, colour=&quot;black&quot;) +
  facet_grid(rows = vars(beta),
             cols = vars(group),
             scales = &quot;free_y&quot;,
             space = &quot;fixed&quot;) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=16),
        axis.title.y = element_text(margin = margin(l = 20, r = 20)),
        axis.title.x = element_text(margin = margin(t = 20, b = 40)),
        plot.title = element_text(hjust = 0.5, margin = margin(t = 20, b = 20)),
        strip.background=element_rect(color = NA),
        panel.spacing=unit(2, &quot;lines&quot;),
        strip.placement = &quot;top&quot;,
        legend.position = &quot;none&quot;,
        strip.text.y.right = element_text(angle = 0)) +
  scale_fill_manual(values=c(&quot;#004766&quot;, &quot;#255672&quot;, &quot;#3d667e&quot;, &quot;#53768A&quot;, &quot;#698696&quot;, &quot;#77c8dd&quot;, &quot;#93cfe3&quot;, &quot;#acd7e7&quot;, &quot;#c2deeb&quot;, &quot;#d7e6ee&quot;)) +
  scale_colour_manual(values=c(&quot;#004766&quot;, &quot;#255672&quot;, &quot;#3d667e&quot;, &quot;#53768A&quot;, &quot;#698696&quot;, &quot;#77c8dd&quot;, &quot;#93cfe3&quot;, &quot;#acd7e7&quot;, &quot;#c2deeb&quot;, &quot;#d7e6ee&quot;)) +
  ylab(&quot;&quot;) +
  xlab(&quot;Position in fragment&quot;) +
  ggtitle(&quot;Position of \&quot;CCG\&quot; in the fragment, based on methylation-status of that \&quot;CCG\&quot;.&quot;)

#ggsave(here(&quot;docs/figure/reviewer1_plot1_ccg.jpg&quot;), plot = plot_ccg, device = &quot;jpeg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=300)

plot2A &lt;- ggplot(t_b[which((t_b$diff &gt;= 1) &amp; (t_b$diff &lt;= 40) &amp; (t_b$motif == &quot;CG&quot;) &amp; (t_b$beta == &quot;methylated: beta &gt;= 0.7;\nunmethylated: beta &lt;= 0.3&quot;)),], aes(x=diff, y=n, colour=group_beta, fill=group_beta)) +
  geom_point(position=position_jitter(h=0.01, w=0.4), alpha = 1, size = 0.1) +
  geom_bar(stat = &quot;identity&quot;, data = t_b.summary[which((t_b.summary$diff&gt;= 1) &amp; (t_b.summary$diff &lt;= 40) &amp; (t_b.summary$motif == &quot;CG&quot;) &amp; (t_b.summary$beta == &quot;methylated: beta &gt;= 0.7;\nunmethylated: beta &lt;= 0.3&quot;)),], colour=&quot;black&quot;, alpha=0.7) +
   geom_errorbar(
    aes(ymin = n-sd, ymax = n+sd),
    data = t_b.summary[which((t_b.summary$diff&gt;= 1) &amp; (t_b.summary$diff &lt;= 40) &amp; (t_b.summary$motif == &quot;CG&quot;) &amp; (t_b.summary$beta == &quot;methylated: beta &gt;= 0.7;\nunmethylated: beta &lt;= 0.3&quot;)),], width = 0.2, colour=&quot;black&quot;) +
  facet_grid(cols = vars(group),
             scales = &quot;free_y&quot;,
             space = &quot;fixed&quot;) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=16),
        axis.title.y = element_text(margin = margin(l = 20, r = 20)),
        axis.title.x = element_text(margin = margin(t = 20, b = 40)),
        strip.background=element_rect(color = NA),
        panel.spacing=unit(2, &quot;lines&quot;),
        strip.placement = &quot;top&quot;,
        legend.position = &quot;none&quot;,
        strip.text.y.right = element_text(angle = 0)) +
  scale_fill_manual(values=c(&quot;#004766&quot;, &quot;#77c8dd&quot;)) +
  scale_colour_manual(values=c(&quot;#004766&quot;, &quot;#77c8dd&quot;)) +
  ylab(&quot;&quot;) +
  xlab(&quot;Position in fragment&quot;)</code></pre>
</div>
<div id="figure-2b-and-2c" class="section level2">
<h2>Figure 2B and 2C</h2>
<pre class="r"><code>t &lt;- readRDS(here(&quot;data/Moss_ratios.rds&quot;))
t &lt;- t[-which(t$chr %in% c(&quot;chrX&quot;, &quot;chrY&quot;)),]

acg_fwd &lt;- t[which(substr(t$seq, 2, 4) == &quot;ACG&quot;),c(2,3,4,5,8,9)]
acg_bwd &lt;- t[which(substr(t$seq, 3, 5) == &quot;CGT&quot;),c(2,3,4,5,10,11)]
colnames(acg_bwd) &lt;- colnames(acg_fwd)
acg &lt;- rbind(acg_fwd, acg_bwd)
acg$group &lt;- &quot;A | CG&quot;
ccg_fwd &lt;- t[which(substr(t$seq, 2, 4) == &quot;CCG&quot;),c(2,3,4,5,8,9)]
ccg_bwd &lt;- t[which(substr(t$seq, 3, 5) == &quot;CGG&quot;),c(2,3,4,5,10,11)]
colnames(ccg_bwd) &lt;- colnames(ccg_fwd)
ccg &lt;- rbind(ccg_fwd, ccg_bwd)
ccg$group &lt;- &quot;C | CG&quot;
gcg_fwd &lt;- t[which(substr(t$seq, 2, 4) == &quot;GCG&quot;),c(2,3,4,5,8,9)]
gcg_bwd &lt;- t[which(substr(t$seq, 3, 5) == &quot;CGC&quot;),c(2,3,4,5,10,11)]
colnames(gcg_bwd) &lt;- colnames(gcg_fwd)
gcg &lt;- rbind(gcg_fwd, gcg_bwd)
gcg$group &lt;- &quot;G | CG&quot;
tcg_fwd &lt;- t[which(substr(t$seq, 2, 4) == &quot;TCG&quot;),c(2,3,4,5,8,9)]
tcg_bwd &lt;- t[which(substr(t$seq, 3, 5) == &quot;CGA&quot;),c(2,3,4,5,10,11)]
colnames(tcg_bwd) &lt;- colnames(gcg_fwd)
tcg &lt;- rbind(tcg_fwd, tcg_bwd)
tcg$group &lt;- &quot;T | CG&quot;
t &lt;- rbind(acg, ccg, gcg, tcg)

all10bars &lt;- t
all10bars$subgroup &lt;- round(all10bars$beta, digits = 1)
all10bars$group &lt;- factor(all10bars$group, levels=c(&quot;C | CG&quot;, &quot;G | CG&quot;, &quot;A | CG&quot;, &quot;T | CG&quot;))

# Compute summary stats for the paper
all10bars %&gt;%
  group_by(group, subgroup) %&gt;%
  summarise(ratio = sum(start_cg) / sum(start_cg_over)) %&gt;%
  group_by(group) %&gt;%
  summarise(min = min(ratio), max = max(ratio), ratio = max(ratio) / min(ratio))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;group&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 4 × 4
  group      min     max ratio
  &lt;fct&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;
1 C | CG 0.00248 0.00553  2.22
2 G | CG 0.00212 0.00514  2.42
3 A | CG 0.00426 0.00960  2.25
4 T | CG 0.00479 0.00990  2.07</code></pre>
<pre class="r"><code>len.samp &lt;- 1e3

# Resample from each listed group keeping rows intact
resample &lt;- function(x, ...) {
    dots &lt;- list(...)
    xpop &lt;- nest(x, data=-unlist(dots))
    xsmp &lt;- bind_cols(select(xpop, unlist(dots)), 
            tibble(data = lapply(xpop$data, function(x) x[sample.int(nrow(x), nrow(x), replace=TRUE),]))) 
    return(unnest(xsmp, data))                         
}

# Compute the &quot;ratio&quot;
compute_ratio &lt;- function(x, ...) {
    options(warn=-1)
    dots &lt;- list(...)
    numer &lt;- colnames(x)[str_detect(colnames(x), &quot;start_.*cg$&quot;)] 
    denom &lt;- colnames(x)[str_detect(colnames(x), &quot;start_.*cg_over&quot;)]
    ratio_fc &lt;- x %&gt;%
        group_by_(.dots = unlist(dots)) %&gt;%
        summarize(ratio = sum(eval(parse(text=numer)))/sum(eval(parse(text=denom))), .groups=&quot;drop&quot;) %&gt;%
        group_by_(.dots = unlist(dots)[1]) %&gt;%
        summarize(min = min(ratio), max = max(ratio), ratio = max(ratio)/min(ratio), .groups=&quot;drop&quot;)
    ratio_vec &lt;- ratio_fc$ratio
    names(ratio_vec) &lt;- ratio_fc$group 
    return(ratio_vec)
}

# Extremely slow and falls out of memory with data.frame or list of data.frame output!!! Poor mem mgmt of R?
#rep_all10bars &lt;- replicate(len.samp, compute_ratio(resample(all10bars, &quot;group&quot;, &quot;subgroup&quot;), &quot;group&quot;, &quot;subgroup&quot;)) 

# Try parallel
my.cluster &lt;- makeCluster(spec=10, type = &quot;FORK&quot;)
doParallel::registerDoParallel(cl = my.cluster)

set.seed(1)
if(!file.exists(here(&quot;data/Figure_2B_stats.rds&quot;))) {
    all10bars_rep &lt;- foreach(repnum=seq_len(len.samp), combine = list) %dorng% {
        all10bars_samp &lt;- resample(all10bars, &quot;group&quot;, &quot;subgroup&quot;)
        all10bars_ratio &lt;- compute_ratio(all10bars_samp, &quot;group&quot;, &quot;subgroup&quot;)
        all10bars_ratio 
    }
    saveRDS(all10bars_rep, here(&quot;data/Figure_2B_stats.rds&quot;))
} else {
    all10bars_rep &lt;- readRDS(here(&quot;data/Figure_2B_stats.rds&quot;))
}

stopCluster(cl = my.cluster)

# For each motif, compute the quantiles of interest
all10bars_rep %&gt;%
    bind_rows() %&gt;%
    apply(., 2, function(x) { 
              y &lt;- quantile(x, p = c(0.5, 0.025, 0.975))
              names(y) &lt;- c(&quot;median&quot;, &quot;conf.low&quot;, &quot;conf.high&quot;)
              round(y,2)
            })</code></pre>
<pre><code>          C | CG G | CG A | CG T | CG
median      2.23   2.42   2.25   2.08
conf.low    2.20   2.39   2.22   2.04
conf.high   2.26   2.45   2.29   2.22</code></pre>
<pre class="r"><code>plot2b &lt;- all10bars %&gt;%
  group_by(group, subgroup) %&gt;%
  summarise(ratio = sum(start_cg) / sum(start_cg_over)) %&gt;%
  ggplot(aes(x=factor(subgroup), y=ratio, fill=subgroup)) +
  geom_bar(stat=&quot;identity&quot;, position = &quot;dodge&quot;) +
  facet_wrap(~group, nrow=1) +
  theme_classic(base_size = 20) +
  #xlab(&quot;Methylation (beta-value)&quot;) +
  #ylab(expression(atop(&quot;Fraction of cfDNA fragments&quot;, paste(&quot;starting or ending at CGs&quot;)))) +
  ylab(&quot;Fraction of cfDNA fragments\nstarting or ending at CGs&quot;) + 
  scale_y_continuous(labels = scales::percent, limits = c(0,0.0105),
                     breaks = c(0.01,0.0075,0.005,0.0025,0)) +
  scale_x_discrete(labels=c(&quot;0&quot;, &quot;1&quot;),
                   breaks = c(0,1)) +
  theme(text = element_text(size=16),
        axis.text.x = element_text(angle = 0, hjust=0.5),
        #axis.text.x = element_blank(),
        #axis.title.y = element_text(margin = margin(r = 5)),
        axis.title.x = element_blank(),
        strip.background=element_rect(color = NA),
        panel.spacing=unit(2, &quot;lines&quot;),
        strip.placement = &quot;outside&quot;,
        legend.position = &quot;none&quot;,
        axis.title.y = element_text(margin = margin(r = 20, l = 20))) +
  scale_fill_gradient(low=&quot;#77C8DD&quot;, high=&quot;#004766&quot;)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;group&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>t &lt;- readRDS(here(&quot;data/Moss_ratios.rds&quot;))
t &lt;- t[-which(t$chr %in% c(&quot;chrX&quot;, &quot;chrY&quot;)),]

accg_fwd &lt;- t[which(substr(t$seq, 1, 4) == &quot;ACCG&quot;),c(2,3,4,5,6,7)]
accg_bwd &lt;- t[which(substr(t$seq, 3, 6) == &quot;CGGT&quot;),c(2,3,4,5,12,13)]
colnames(accg_bwd) &lt;- colnames(accg_fwd)
accg &lt;- rbind(accg_fwd, accg_bwd)
accg$group &lt;- &quot;A | CCG&quot;
cccg_fwd &lt;- t[which(substr(t$seq, 1, 4) == &quot;CCCG&quot;),c(2,3,4,5,6,7)]
cccg_bwd &lt;- t[which(substr(t$seq, 3, 6) == &quot;CGGG&quot;),c(2,3,4,5,12,13)]
colnames(cccg_bwd) &lt;- colnames(cccg_fwd)
cccg &lt;- rbind(cccg_fwd, cccg_bwd)
cccg$group &lt;- &quot;C | CCG&quot;
gccg_fwd &lt;- t[which(substr(t$seq, 1, 4) == &quot;GCCG&quot;),c(2,3,4,5,6,7)]
gccg_bwd &lt;- t[which(substr(t$seq, 3, 6) == &quot;CGGC&quot;),c(2,3,4,5,12,13)]
colnames(gccg_bwd) &lt;- colnames(gccg_fwd)
gccg &lt;- rbind(gccg_fwd, gccg_bwd)
gccg$group &lt;- &quot;G | CCG&quot;
tccg_fwd &lt;- t[which(substr(t$seq, 1, 4) == &quot;TCCG&quot;),c(2,3,4,5,6,7)]
tccg_bwd &lt;- t[which(substr(t$seq, 3, 6) == &quot;CGGA&quot;),c(2,3,4,5,12,13)]
colnames(tccg_bwd) &lt;- colnames(gccg_fwd)
tccg &lt;- rbind(tccg_fwd, tccg_bwd)
tccg$group &lt;- &quot;T | CCG&quot;
t &lt;- rbind(accg, cccg, gccg, tccg)

all10bars &lt;- t
all10bars$subgroup &lt;- round(all10bars$beta, digits = 1)
all10bars$group &lt;- factor(all10bars$group, levels=c(&quot;C | CCG&quot;, &quot;G | CCG&quot;, &quot;A | CCG&quot;, &quot;T | CCG&quot;))

all10bars %&gt;%
  group_by(group, subgroup) %&gt;%
  summarise(ratio = sum(start_ncg) / sum(start_ncg_over)) %&gt;%
  group_by(group) %&gt;%
  summarise(min = min(ratio), max = max(ratio), ratio = max(ratio) / min(ratio))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;group&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 4 × 4
  group       min     max ratio
  &lt;fct&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;
1 C | CCG 0.00238 0.00547  2.30
2 G | CCG 0.00181 0.00452  2.49
3 A | CCG 0.00413 0.00838  2.03
4 T | CCG 0.00321 0.0118   3.68</code></pre>
<pre class="r"><code># Try parallel
my.cluster &lt;- makeCluster(spec=10, type = &quot;FORK&quot;)
doParallel::registerDoParallel(cl = my.cluster)

set.seed(1)
if(!file.exists(here(&quot;data/Figure_2C_stats.rds&quot;))) {
    all10bars_rep &lt;- foreach(repnum=seq_len(len.samp), combine = list) %dorng% {
        all10bars_samp &lt;- resample(all10bars, &quot;group&quot;, &quot;subgroup&quot;)
        all10bars_ratio &lt;- compute_ratio(all10bars_samp, &quot;group&quot;, &quot;subgroup&quot;)
        all10bars_ratio 
    }
    saveRDS(all10bars_rep, here(&quot;data/Figure_2C_stats.rds&quot;))
} else {
    all10bars_rep &lt;- readRDS(here(&quot;data/Figure_2C_stats.rds&quot;))
}

stopCluster(cl = my.cluster)

# For each motif, compute the quantiles of interest
all10bars_rep %&gt;%
    bind_rows() %&gt;%
    apply(., 2, function(x) { 
              y &lt;- quantile(x, p = c(0.5, 0.025, 0.975))
              names(y) &lt;- c(&quot;median&quot;, &quot;conf.low&quot;, &quot;conf.high&quot;)
              round(y,2)
            })</code></pre>
<pre><code>          C | CCG G | CCG A | CCG T | CCG
median       2.30    2.51    2.03    3.69
conf.low     2.21    2.38    1.92    3.46
conf.high    2.43    2.65    2.19    3.92</code></pre>
<pre class="r"><code>plot2c &lt;- all10bars %&gt;%
  group_by(group, subgroup) %&gt;%
  summarise(ratio = sum(start_ncg) / sum(start_ncg_over)) %&gt;%
  ggplot(aes(x=factor(subgroup), y=ratio, fill=subgroup)) +
  geom_bar(stat=&quot;identity&quot;, position = &quot;dodge&quot;) +
  facet_wrap(~group, nrow=1) +
  theme_classic(base_size = 20) +
  #xlab(&quot;Methylation (beta-value)&quot;) +
  #ylab(expression(atop(&quot;Fraction of cfDNA fragments&quot;, paste(&quot;starting or ending at CGs&quot;)))) +
  ylab(&quot;Fraction of cfDNA fragments\nstarting or ending at CCGs&quot;) + 
  scale_y_continuous(labels = scales::percent, limits = c(0,0.0125),
                     breaks = c(0.0125, 0.01,0.0075,0.005,0.0025,0)) +
  scale_x_discrete(labels=c(&quot;0&quot;, &quot;1&quot;),
                   breaks = c(0,1)) +
  theme(text = element_text(size=16),
        axis.text.x = element_text(angle = 0, hjust=0.5),
        #axis.text.x = element_blank(),
        #axis.title.y = element_text(margin = margin(r = 5)),
        axis.title.x = element_blank(),
        strip.background=element_rect(color = NA),
        panel.spacing=unit(2, &quot;lines&quot;),
        strip.placement = &quot;outside&quot;,
        legend.position = &quot;none&quot;,
        axis.title.y = element_text(margin = margin(r = 20, l = 20))) +
  scale_fill_gradient(low=&quot;#77C8DD&quot;, high=&quot;#004766&quot;)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;group&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>title &lt;- textGrob(&quot;window: +/- 50 bps (total: 101 bps)&quot;, gp = gpar(col = &quot;black&quot;, fontsize = 19))
#plot2bc_50 &lt;- plot_grid(title, plot2b, plot2c, ncol = 1, align=&quot;v&quot;, axis=&quot;bt&quot;, rel_heights = c(1, 5, 5), labels = c(&quot;a&quot;, &quot;&quot;, &quot;&quot;), label_size = 20)
#plot2BC &lt;- plot_grid(plot2b, plot2c, ncol = 1, align=&quot;v&quot;, axis=&quot;bt&quot;, rel_heights = c(5, 5), labels = c(&quot;b&quot;, &quot;c&quot;), label_size = 20)
#ggsave(here(&quot;docs/figure/figure2bc_window50bp.jpg&quot;), plot = plot2bc_50, device = &quot;jpeg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=300)</code></pre>
</div>
<div id="figure-2d-and-2e" class="section level2">
<h2>Figure 2D and 2E</h2>
<pre class="r"><code>sum2 &lt;- readRDS(here(&quot;data/som_chrx.rds&quot;))

sum2[which(nchar(sum2$motif) == 4),]$motif &lt;- sub(&quot;CCG&quot;, &quot; | CCG&quot;, sum2[which(nchar(sum2$motif) == 4),]$motif)
sum2[which(nchar(sum2$motif) == 3),]$motif &lt;- sub(&quot;CG&quot;, &quot; | CG&quot;, sum2[which(nchar(sum2$motif) == 3),]$motif)
sum2$motif &lt;- factor(sum2$motif, levels=c(&quot;C | CCG&quot;, &quot;G | CCG&quot;, &quot;A | CCG&quot;, &quot;T | CCG&quot;, &quot;C | CG&quot;, &quot;G | CG&quot;, &quot;A | CG&quot;, &quot;T | CG&quot;))
sum2$sex &lt;- factor(sum2$sex, levels = c(&quot;male&quot;, &quot;female&quot;))
sum &lt;- sum2

plot2D2 &lt;- ggplot(data=sum[which((nchar(as.character(sum$motif)) == 7) &amp; (sum$group == &quot;islands&quot;) &amp; (sum$chromosome == &quot;chrX&quot;)) ,], aes(x = sex, y=ratio, fill=sex)) +
  geom_jitter(size=0.4, alpha=1, aes(colour=sex)) +
  geom_boxplot(outlier.shape = NA, aes(alpha = 0.1)) +
  scale_y_continuous(labels = scales::percent,
                     limits = c(0,0.013),
                     breaks=c(0.0125,0.01,0.0075,0.005,0.0025,0)) +
  ylab(&quot;Fraction of cfDNA fragments\nstarting or ending at CCGs&quot;) + 
  facet_grid(~motif) +
  theme_classic(base_size = 20) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        text = element_text(size=16),
        #axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        strip.background=element_rect(color = NA,  fill=NA),
        legend.position = &quot;none&quot;,
        axis.title.y = element_text(margin = margin(r = 20, l = 20))) +
  scale_fill_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;)) +
  scale_color_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;))  + 
  stat_compare_means(method = &quot;t.test&quot;, size = 3, aes(label = paste0(&quot;p &quot;, after_stat(p.format))))

plot2D1 &lt;- ggplot(data=sum[which((nchar(as.character(sum$motif)) == 6) &amp; (sum$group == &quot;islands&quot;) &amp; (sum$chromosome == &quot;chrX&quot;)),], aes(x = sex, y=ratio, fill=sex)) +
  geom_jitter(size=0.4, alpha=1, aes(colour=sex)) +
  geom_boxplot(outlier.shape = NA, aes(alpha = 0.1)) +
  scale_y_continuous(labels = scales::percent,
                     breaks = c(0.01,0.0075,0.005,0.0025,0),
                     limits = c(0,0.0105)) +
  ylab(&quot;Fraction of cfDNA fragments\nstarting or ending at CGs&quot;) + 
  facet_grid(~motif) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=16),
        #axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x.bottom = element_blank(),
        strip.background=element_rect(color = NA,  fill=NA),
        legend.position = &quot;none&quot;,
        axis.title.y = element_text(margin = margin(r = 20, l = 20)),
        plot.title = element_text(hjust = 0.5, vjust=0.1, size = 16)) +
  scale_fill_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;)) +
  scale_color_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;)) +
  ggtitle(&quot;Chromosome X&quot;) + 
  stat_compare_means(method = &quot;t.test&quot;, size = 3, aes(label = paste0(&quot;p &quot;, after_stat(p.format))))

plot2E2 &lt;- ggplot(data=sum2[which((nchar(as.character(sum2$motif)) == 7) &amp; (sum2$group == &quot;islands&quot;) &amp; (sum2$chromosome == &quot;somatic&quot;)) ,], aes(x = sex, y=ratio, fill=sex)) +
  geom_jitter(size=0.4, alpha=1, aes(colour=sex)) +
  geom_boxplot(outlier.shape = NA, aes(alpha = 0.1)) +
  scale_y_continuous(labels = scales::percent,
                     limits = c(0,0.013),
                     breaks=c(0.0125,0.01,0.0075,0.005,0.0025,0)) +
  ylab(&quot;Fraction of cfDNA fragments\nstarting or ending at CCGs&quot;) + 
  facet_grid(~motif) +
  theme_classic(base_size = 20) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        text = element_text(size=16),
        #axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        strip.background=element_rect(color = NA,  fill=NA),
        legend.position = &quot;none&quot;,
        axis.title.y = element_text(margin = margin(r = 20, l = 20))) +
  scale_fill_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;)) +
  scale_color_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;)) + 
  stat_compare_means(method = &quot;t.test&quot;, size = 3, aes(label = paste0(&quot;p = &quot;, after_stat(p.format))))

plot2E1 &lt;- ggplot(data=sum2[which((nchar(as.character(sum2$motif)) == 6) &amp; (sum2$group == &quot;islands&quot;)  &amp; (sum2$chromosome == &quot;somatic&quot;)),], aes(x = sex, y=ratio, fill=sex)) +
  geom_jitter(size=0.4, alpha=1, aes(colour=sex)) +
  geom_boxplot(outlier.shape = NA, aes(alpha = 0.1)) +
  scale_y_continuous(labels = scales::percent,
                     breaks = c(0.01,0.0075,0.005,0.0025,0),
                     limits = c(0,0.0105)) +
  ggtitle(&quot;Autosomes&quot;) +
  ylab(&quot;Fraction of cfDNA fragments\nstarting or ending at CGs&quot;) + 
  facet_grid(~motif) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=16),
        #axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x.bottom = element_blank(),
        strip.background=element_rect(color = NA,  fill=NA),
        legend.position = &quot;none&quot;,
        axis.title.y = element_text(margin = margin(r = 20, l = 20)),
        plot.title = element_text(hjust = 0.5, vjust=0.1, size = 16)) +
  scale_fill_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;)) +
  scale_color_manual(values = c(&quot;#99CCFF&quot;, &quot;#FF99BB&quot;)) + 
  stat_compare_means(size = 3, method = &quot;t.test&quot;, aes(label = paste0(&quot;p = &quot;, after_stat(p.format))))</code></pre>
<pre class="r"><code>get_ratio &lt;- function(x, sex_cat) {
    return(filter(x, sex == sex_cat) %&gt;% pull(ratio))
}

apply_test &lt;- function(x, func, ...) {
    m &lt;- get_ratio(x, &quot;male&quot;) 
    f &lt;- get_ratio(x, &quot;female&quot;)
    return(tidy(func(f,m, ...)))    
}

apply_fc &lt;- function(x) {
    len.samp &lt;- 1e3
    m &lt;- get_ratio(x, &quot;male&quot;)
    f &lt;- get_ratio(x, &quot;female&quot;)

    # compute median on resample 1000 times
    mrep_median &lt;- apply(replicate(len.samp, sample(m, length(m), replace=TRUE)),2,median)
    frep_median &lt;- apply(replicate(len.samp, sample(f, length(f), replace=TRUE)),2,median)

    # compute statistics
    fc &lt;- frep_median/mrep_median
    statistic &lt;- median(fc)
    conf.int &lt;- quantile(fc, prob=c(0.025, 0.975))
    if(statistic &gt;= 1) p.value &lt;- mean(fc &lt; 1) else p.value &lt;- mean(fc &gt;= 1)

    # return stat data
    tibble(statistic = statistic, conf.low = conf.int[1], conf.high=conf.int[2], p.value = p.value, method = &quot;Fold-change by sampling&quot;, .fc_rep = list(fc))
}

set.seed(1)
stat_table &lt;- sum2 %&gt;%
              filter(group == &quot;islands&quot;) %&gt;%
              mutate(motif_width = nchar(as.character(motif))) %&gt;%
              nest(data = -c(chromosome, motif, motif_width)) %&gt;%
              mutate(ttest = lapply(data, apply_test, t.test),
                     wilcox = lapply(data, apply_test, wilcox.test, conf.int=TRUE),
                     ks = lapply(data, apply_test, ks.test), 
                     fc = lapply(data, apply_fc)) %&gt;%
              pivot_longer(cols = c(ttest, wilcox, ks, fc), names_to=&quot;test&quot;, values_to=&quot;test_results&quot;) %&gt;%
              select(-data) %&gt;%
              unnest(test_results) %&gt;%
              mutate(p.value = as.character(signif(p.value, 5))) %&gt;%
              arrange(chromosome, motif) </code></pre>
<pre><code>Warning: There were 7 warnings in `mutate()`.
The first warning was:
ℹ In argument: `ks = lapply(data, apply_test, ks.test)`.
Caused by warning in `ks.test.default()`:
! p-value will be approximate in the presence of ties
ℹ Run `dplyr::last_dplyr_warnings()` to see the 6 remaining warnings.</code></pre>
<pre class="r"><code>kable(stat_table %&gt;% select(-.fc_rep))</code></pre>
<table>
<colgroup>
<col width="6%" />
<col width="5%" />
<col width="6%" />
<col width="3%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
<col width="25%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">motif</th>
<th align="left">chromosome</th>
<th align="right">motif_width</th>
<th align="left">test</th>
<th align="right">estimate</th>
<th align="right">estimate1</th>
<th align="right">estimate2</th>
<th align="right">statistic</th>
<th align="left">p.value</th>
<th align="right">parameter</th>
<th align="right">conf.low</th>
<th align="right">conf.high</th>
<th align="left">method</th>
<th align="left">alternative</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">-0.0008183</td>
<td align="right">0.0041963</td>
<td align="right">0.0050146</td>
<td align="right">-16.5722218</td>
<td align="left">2.8534e-49</td>
<td align="right">488.5832</td>
<td align="right">-0.0009154</td>
<td align="right">-0.0007213</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">C | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">-0.0008347</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">11597.5000000</td>
<td align="left">2.2282e-43</td>
<td align="right">NA</td>
<td align="right">-0.0009215</td>
<td align="right">-0.0007217</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">C | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.5709108</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">C | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.8330702</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">0.8096351</td>
<td align="right">0.8600672</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">G | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">-0.0008348</td>
<td align="right">0.0033582</td>
<td align="right">0.0041930</td>
<td align="right">-18.0341592</td>
<td align="left">1.2988e-55</td>
<td align="right">468.9082</td>
<td align="right">-0.0009258</td>
<td align="right">-0.0007438</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">G | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">-0.0008327</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">9477.0000000</td>
<td align="left">1.1529e-50</td>
<td align="right">NA</td>
<td align="right">-0.0009425</td>
<td align="right">-0.0007640</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">G | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.6165468</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">G | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.7990877</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">0.7780157</td>
<td align="right">0.8226999</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">A | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">-0.0010669</td>
<td align="right">0.0062459</td>
<td align="right">0.0073128</td>
<td align="right">-9.9102599</td>
<td align="left">4.9279e-21</td>
<td align="right">440.4801</td>
<td align="right">-0.0012784</td>
<td align="right">-0.0008553</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">-0.0010600</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">20029.0000000</td>
<td align="left">3.7164e-20</td>
<td align="right">NA</td>
<td align="right">-0.0012906</td>
<td align="right">-0.0008410</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">A | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.4182028</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">A | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.8492628</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">0.8222883</td>
<td align="right">0.8747114</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">T | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">-0.0025614</td>
<td align="right">0.0081934</td>
<td align="right">0.0107549</td>
<td align="right">-22.9787233</td>
<td align="left">1.3239e-77</td>
<td align="right">444.8172</td>
<td align="right">-0.0027805</td>
<td align="right">-0.0023423</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">T | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">-0.0025730</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">5862.0000000</td>
<td align="left">1.9863e-64</td>
<td align="right">NA</td>
<td align="right">-0.0028125</td>
<td align="right">-0.0023962</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">T | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.7271074</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">T | CCG</td>
<td align="left">chrX</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.7561400</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">0.7373494</td>
<td align="right">0.7740291</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">C | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">0.0007060</td>
<td align="right">0.0033631</td>
<td align="right">0.0026572</td>
<td align="right">25.7026867</td>
<td align="left">7.4315e-94</td>
<td align="right">506.2998</td>
<td align="right">0.0006520</td>
<td align="right">0.0007599</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">C | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">0.0007096</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">69259.0000000</td>
<td align="left">1.9828e-70</td>
<td align="right">NA</td>
<td align="right">0.0006666</td>
<td align="right">0.0007905</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">C | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.7553550</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">C | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.2828704</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">1.2480281</td>
<td align="right">1.3090864</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">G | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">0.0006017</td>
<td align="right">0.0030563</td>
<td align="right">0.0024545</td>
<td align="right">22.7048855</td>
<td align="left">1.3754e-79</td>
<td align="right">514.6727</td>
<td align="right">0.0005497</td>
<td align="right">0.0006538</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">G | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">0.0005918</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">67878.0000000</td>
<td align="left">1.0347e-64</td>
<td align="right">NA</td>
<td align="right">0.0005409</td>
<td align="right">0.0006503</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">G | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.6955070</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">G | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.2360335</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">1.2071220</td>
<td align="right">1.2619465</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">A | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">0.0009838</td>
<td align="right">0.0060002</td>
<td align="right">0.0050165</td>
<td align="right">16.7221270</td>
<td align="left">1.3e-50</td>
<td align="right">523.7699</td>
<td align="right">0.0008682</td>
<td align="right">0.0010993</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">0.0009919</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">62097.0000000</td>
<td align="left">1.8497e-43</td>
<td align="right">NA</td>
<td align="right">0.0008600</td>
<td align="right">0.0011077</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">A | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.5423103</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">A | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.1970748</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">1.1706109</td>
<td align="right">1.2250318</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">T | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">0.0012174</td>
<td align="right">0.0069466</td>
<td align="right">0.0057293</td>
<td align="right">21.2281826</td>
<td align="left">8.4914e-73</td>
<td align="right">528.8863</td>
<td align="right">0.0011047</td>
<td align="right">0.0013300</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">T | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">0.0012361</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">66377.0000000</td>
<td align="left">8.8894e-59</td>
<td align="right">NA</td>
<td align="right">0.0011209</td>
<td align="right">0.0013670</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">T | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.6690240</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">T | CG</td>
<td align="left">chrX</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.2157725</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">1.1901867</td>
<td align="right">1.2474226</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">C | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">0.0000430</td>
<td align="right">0.0048899</td>
<td align="right">0.0048468</td>
<td align="right">1.7208865</td>
<td align="left">0.085843</td>
<td align="right">540.8771</td>
<td align="right">-0.0000061</td>
<td align="right">0.0000922</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">C | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">0.0000481</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">39644.0000000</td>
<td align="left">0.12434</td>
<td align="right">NA</td>
<td align="right">-0.0000167</td>
<td align="right">0.0000932</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">C | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.0709108</td>
<td align="left">0.50254</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">C | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.0048669</td>
<td align="left">0.191</td>
<td align="right">NA</td>
<td align="right">0.9940504</td>
<td align="right">1.0172160</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">G | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">0.0000085</td>
<td align="right">0.0040246</td>
<td align="right">0.0040160</td>
<td align="right">0.3816294</td>
<td align="left">0.70289</td>
<td align="right">533.8919</td>
<td align="right">-0.0000354</td>
<td align="right">0.0000525</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">G | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">0.0000116</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">37428.0000000</td>
<td align="left">0.74577</td>
<td align="right">NA</td>
<td align="right">-0.0000468</td>
<td align="right">0.0000660</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">G | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.0500204</td>
<td align="left">0.88641</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">G | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.0019129</td>
<td align="left">0.349</td>
<td align="right">NA</td>
<td align="right">0.9919317</td>
<td align="right">1.0136983</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">A | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">0.0000079</td>
<td align="right">0.0071118</td>
<td align="right">0.0071039</td>
<td align="right">0.2142483</td>
<td align="left">0.83043</td>
<td align="right">540.7156</td>
<td align="right">-0.0000645</td>
<td align="right">0.0000803</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">-0.0000052</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">37360.0000000</td>
<td align="left">0.77411</td>
<td align="right">NA</td>
<td align="right">-0.0000577</td>
<td align="right">0.0000943</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">A | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.0467354</td>
<td align="left">0.92836</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">A | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.0030376</td>
<td align="left">0.324</td>
<td align="right">NA</td>
<td align="right">0.9922318</td>
<td align="right">1.0148320</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">T | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ttest</td>
<td align="right">-0.0001708</td>
<td align="right">0.0103425</td>
<td align="right">0.0105133</td>
<td align="right">-3.7078104</td>
<td align="left">0.00023066</td>
<td align="right">540.0889</td>
<td align="right">-0.0002613</td>
<td align="right">-0.0000803</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">T | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">wilcox</td>
<td align="right">-0.0001675</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">30384.0000000</td>
<td align="left">0.00041601</td>
<td align="right">NA</td>
<td align="right">-0.0002214</td>
<td align="right">-0.0000680</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">T | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.1468712</td>
<td align="left">0.0057413</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">T | CCG</td>
<td align="left">somatic</td>
<td align="right">7</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.9865114</td>
<td align="left">0.002</td>
<td align="right">NA</td>
<td align="right">0.9770599</td>
<td align="right">0.9959931</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">C | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">0.0000785</td>
<td align="right">0.0028872</td>
<td align="right">0.0028087</td>
<td align="right">5.3022026</td>
<td align="left">1.6722e-07</td>
<td align="right">538.8709</td>
<td align="right">0.0000494</td>
<td align="right">0.0001076</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">C | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">0.0000869</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">45853.0000000</td>
<td align="left">8.0409e-07</td>
<td align="right">NA</td>
<td align="right">0.0000399</td>
<td align="right">0.0000975</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">C | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.2012488</td>
<td align="left">3.3748e-05</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">C | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.0256265</td>
<td align="left">0</td>
<td align="right">NA</td>
<td align="right">1.0150666</td>
<td align="right">1.0354111</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">G | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">0.0000504</td>
<td align="right">0.0025883</td>
<td align="right">0.0025379</td>
<td align="right">3.5404947</td>
<td align="left">0.00043387</td>
<td align="right">539.9168</td>
<td align="right">0.0000224</td>
<td align="right">0.0000784</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">G | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">0.0000664</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">42731.0000000</td>
<td align="left">0.0012553</td>
<td align="right">NA</td>
<td align="right">-0.0000040</td>
<td align="right">0.0000530</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">G | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.1687390</td>
<td align="left">0.00088246</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">G | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.0142573</td>
<td align="left">0.011</td>
<td align="right">NA</td>
<td align="right">1.0020890</td>
<td align="right">1.0283634</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">A | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">0.0000470</td>
<td align="right">0.0054591</td>
<td align="right">0.0054121</td>
<td align="right">1.8093518</td>
<td align="left">0.070952</td>
<td align="right">540.6403</td>
<td align="right">-0.0000040</td>
<td align="right">0.0000981</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">A | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">0.0000498</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">40505.0000000</td>
<td align="left">0.044649</td>
<td align="right">NA</td>
<td align="right">0.0000162</td>
<td align="right">0.0000742</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">A | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.1028234</td>
<td align="left">0.11351</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">A | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.0088788</td>
<td align="left">0.105</td>
<td align="right">NA</td>
<td align="right">0.9959093</td>
<td align="right">1.0223522</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">T | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ttest</td>
<td align="right">-0.0000342</td>
<td align="right">0.0058221</td>
<td align="right">0.0058563</td>
<td align="right">-1.5330813</td>
<td align="left">0.12584</td>
<td align="right">540.9718</td>
<td align="right">-0.0000781</td>
<td align="right">0.0000096</td>
<td align="left">Welch Two Sample t-test</td>
<td align="left">two.sided</td>
</tr>
<tr class="even">
<td align="left">T | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">wilcox</td>
<td align="right">-0.0000471</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">34365.0000000</td>
<td align="left">0.1766</td>
<td align="right">NA</td>
<td align="right">-0.0000666</td>
<td align="right">0.0000243</td>
<td align="left">Wilcoxon rank sum test with continuity correction</td>
<td align="left">two.sided</td>
</tr>
<tr class="odd">
<td align="left">T | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">ks</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.0804262</td>
<td align="left">0.34398</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">Asymptotic two-sample Kolmogorov-Smirnov test</td>
<td align="left">two-sided</td>
</tr>
<tr class="even">
<td align="left">T | CG</td>
<td align="left">somatic</td>
<td align="right">6</td>
<td align="left">fc</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.9931205</td>
<td align="left">0.046</td>
<td align="right">NA</td>
<td align="right">0.9860432</td>
<td align="right">1.0013421</td>
<td align="left">Fold-change by sampling</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<pre class="r"><code>write.csv(stat_table %&gt;% select(-.fc_rep), here(&quot;docs/figure/Figure_2DE.csv&quot;), row.names=FALSE, quote=FALSE)

# Manuscript text stats
stat_table %&gt;%
    filter(test == &quot;fc&quot;) %&gt;%
    group_by(chromosome, motif_width) %&gt;%
    summarize(statistic = median(unlist(.fc_rep)),
              conf.low = quantile(unlist(.fc_rep), p=0.025),
              conf.high = quantile(unlist(.fc_rep), p=0.975))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;chromosome&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code># A tibble: 4 × 5
# Groups:   chromosome [2]
  chromosome motif_width statistic conf.low conf.high
  &lt;chr&gt;            &lt;int&gt;     &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;
1 chrX                 6     1.23     1.18      1.30 
2 chrX                 7     0.815    0.743     0.866
3 somatic              6     1.01     0.988     1.03 
4 somatic              7     1.00     0.980     1.01 </code></pre>
</div>
<div id="figure-2-combine-figure-2a-figure-2b-and-2c-figure-2d-and-2e"
class="section level2">
<h2>Figure 2: combine Figure 2A, Figure 2B and 2C, Figure 2D and 2E</h2>
<pre class="r"><code>p1 &lt;- plot_grid(plot2b, NULL, plot2D1, NULL, plot2E1, ncol=5, rel_widths = c(12,1,7, 1, 7), align=&quot;h&quot;, axis=&quot;bt&quot;, labels = c(&quot;b&quot;, &quot;&quot;, &quot;d&quot;, &quot;&quot;, &quot;e&quot;), label_size = 20)
p2 &lt;- plot_grid(plot2c, NULL, plot2D2, NULL, plot2E2, ncol=5, rel_widths = c(12,1,7, 1, 7), align=&quot;h&quot;, axis=&quot;bt&quot;, labels = c(&quot;c&quot;, &quot;&quot;, &quot;&quot;, &quot;&quot;, &quot;&quot;),  label_size = 20)</code></pre>
<pre><code>Warning: Removed 15 rows containing non-finite values (`stat_boxplot()`).</code></pre>
<pre><code>Warning: Removed 15 rows containing non-finite values (`stat_compare_means()`).</code></pre>
<pre><code>Warning: Removed 15 rows containing missing values (`geom_point()`).</code></pre>
<pre class="r"><code>p12 &lt;- plot_grid(p1,NULL, p2,ncol=1,rel_heights = c(0.9,0.05,1.1), align=&quot;v&quot;)
plot2 &lt;- plot_grid(plot2A, p12, ncol=1,rel_heights= c(1,2), labels = c(&quot;a&quot;, &quot;&quot;), label_size = 20)

dir.create(here(&quot;docs/figure&quot;), showWarnings=FALSE)
ggsave(here(&quot;docs/figure/Fig_2.jpg&quot;), plot = plot2, device = &quot;jpeg&quot;,width = 180, height=108,units = &quot;mm&quot;, dpi=600, bg = &quot;white&quot;, scale = 3)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] parallel  grid      stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] here_1.0.1           doRNG_1.8.6          rngtools_1.5.2      
 [4] foreach_1.5.2        knitr_1.45           yardstick_1.3.0     
 [7] workflowsets_1.0.1   workflows_1.1.3      tune_1.1.2          
[10] rsample_1.2.0        recipes_1.0.9        parsnip_1.1.1       
[13] modeldata_1.3.0      infer_1.0.6          dials_1.2.0         
[16] scales_1.3.0         broom_1.0.5          tidymodels_1.1.1    
[19] ggpubr_0.6.0         GenomicRanges_1.50.2 GenomeInfoDb_1.34.9 
[22] IRanges_2.32.0       S4Vectors_0.36.2     BiocGenerics_0.44.0 
[25] data.table_1.15.0    cowplot_1.1.3        annotatr_1.24.0     
[28] lubridate_1.9.3      forcats_1.0.0        stringr_1.5.1       
[31] dplyr_1.1.4          purrr_1.0.2          readr_2.1.5         
[34] tidyr_1.3.1          tibble_3.2.1         ggplot2_3.4.4       
[37] tidyverse_2.0.0      workflowr_1.7.1     

loaded via a namespace (and not attached):
  [1] utf8_1.2.4                    tidyselect_1.2.0             
  [3] RSQLite_2.3.5                 AnnotationDbi_1.60.2         
  [5] BiocParallel_1.32.6           munsell_0.5.0                
  [7] ragg_1.2.7                    codetools_0.2-19             
  [9] future_1.33.1                 withr_3.0.0                  
 [11] colorspace_2.1-0              Biobase_2.58.0               
 [13] filelock_1.0.3                rstudioapi_0.15.0            
 [15] ggsignif_0.6.4                listenv_0.9.1                
 [17] labeling_0.4.3                MatrixGenerics_1.10.0        
 [19] git2r_0.33.0                  GenomeInfoDbData_1.2.9       
 [21] farver_2.1.1                  bit64_4.0.5                  
 [23] DiceDesign_1.10               rprojroot_2.0.4              
 [25] parallelly_1.36.0             vctrs_0.6.5                  
 [27] generics_0.1.3                ipred_0.9-14                 
 [29] xfun_0.42                     timechange_0.3.0             
 [31] BiocFileCache_2.11.1          doParallel_1.0.17            
 [33] regioneR_1.30.0               R6_2.5.1                     
 [35] bitops_1.0-7                  lhs_1.1.6                    
 [37] cachem_1.0.8                  DelayedArray_0.24.0          
 [39] promises_1.2.1                BiocIO_1.8.0                 
 [41] nnet_7.3-19                   gtable_0.3.4                 
 [43] globals_0.16.2                processx_3.8.3               
 [45] timeDate_4032.109             rlang_1.1.3                  
 [47] systemfonts_1.0.5             splines_4.2.3                
 [49] rtracklayer_1.58.0            rstatix_0.7.2                
 [51] BiocManager_1.30.22           yaml_2.3.8                   
 [53] reshape2_1.4.4                abind_1.4-5                  
 [55] GenomicFeatures_1.50.4        backports_1.4.1              
 [57] httpuv_1.6.14                 tools_4.2.3                  
 [59] lava_1.7.3                    ellipsis_0.3.2               
 [61] jquerylib_0.1.4               Rcpp_1.0.12                  
 [63] plyr_1.8.9                    progress_1.2.3               
 [65] zlibbioc_1.44.0               RCurl_1.98-1.14              
 [67] ps_1.7.6                      prettyunits_1.2.0            
 [69] rpart_4.1.23                  SummarizedExperiment_1.28.0  
 [71] fs_1.6.3                      furrr_0.3.1                  
 [73] magrittr_2.0.3                GPfit_1.0-8                  
 [75] whisker_0.4.1                 matrixStats_1.1.0            
 [77] hms_1.1.3                     mime_0.12                    
 [79] evaluate_0.23                 xtable_1.8-4                 
 [81] XML_3.99-0.16.1               compiler_4.2.3               
 [83] biomaRt_2.54.1                crayon_1.5.2                 
 [85] htmltools_0.5.7               later_1.3.2                  
 [87] tzdb_0.4.0                    DBI_1.2.1                    
 [89] dbplyr_2.4.0                  MASS_7.3-60.0.1              
 [91] rappdirs_0.3.3                Matrix_1.6-5                 
 [93] car_3.1-2                     cli_3.6.2                    
 [95] gower_1.0.1                   pkgconfig_2.0.3              
 [97] getPass_0.2-4                 GenomicAlignments_1.34.1     
 [99] xml2_1.3.6                    bslib_0.6.1                  
[101] hardhat_1.3.1                 XVector_0.38.0               
[103] prodlim_2023.08.28            callr_3.7.3                  
[105] digest_0.6.34                 Biostrings_2.66.0            
[107] rmarkdown_2.25                restfulr_0.0.15              
[109] curl_5.2.0                    shiny_1.8.0                  
[111] Rsamtools_2.14.0              rjson_0.2.21                 
[113] lifecycle_1.0.4               jsonlite_1.8.8               
[115] carData_3.0-5                 BSgenome_1.66.3              
[117] fansi_1.0.6                   pillar_1.9.0                 
[119] lattice_0.22-5                KEGGREST_1.38.0              
[121] fastmap_1.1.1                 httr_1.4.7                   
[123] survival_3.5-7                interactiveDisplayBase_1.36.0
[125] glue_1.7.0                    iterators_1.0.14             
[127] png_0.1-8                     BiocVersion_3.16.0           
[129] bit_4.0.5                     class_7.3-22                 
[131] stringi_1.8.3                 sass_0.4.8                   
[133] blob_1.2.4                    textshaping_0.3.7            
[135] AnnotationHub_3.6.0           memoise_2.0.1                
[137] future.apply_1.11.1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
