<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Michael Noe" />

<meta name="date" content="2024-03-12" />

<title>Figure3.Rmd</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">noe2024</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure3.Rmd</h1>
<h4 class="author">Michael Noe</h4>
<h4 class="date">2024-03-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>noe2024/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024tree65ac5a530490a0f31dfd2a3aef73903ed07b27bdtargetblank65ac5a5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/cancer-genomics/noe2024/tree/65ac5a530490a0f31dfd2a3aef73903ed07b27bd" target="_blank">65ac5a5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024tree65ac5a530490a0f31dfd2a3aef73903ed07b27bdtargetblank65ac5a5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/cancer-genomics/noe2024/tree/65ac5a530490a0f31dfd2a3aef73903ed07b27bd" target="_blank">65ac5a5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/results/

Unstaged changes:
    Deleted:    analysis/.nfs00000000003fc5760000002d
    Deleted:    analysis/.nfs00000000003fc81300000027
    Deleted:    analysis/.nfs00000000003fc8ee00000029
    Deleted:    analysis/.nfs00000000003fc8f30000002a
    Deleted:    analysis/.nfs00000000003fc8f80000002b
    Deleted:    analysis/.nfs00000000003fc97f0000002c
    Deleted:    analysis/.nfs00000000003fca2d00000028
    Deleted:    analysis/.nfs00000000003fcac40000002e
    Modified:   analysis/Figure3.Rmd
    Modified:   analysis/SessionInfo.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Figure3.Rmd</code>) and HTML
(<code>docs/Figure3.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/65ac5a530490a0f31dfd2a3aef73903ed07b27bd/docs/Figure3.html" target="_blank">65ac5a5</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-11
</td>
<td>
Update supplementary tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/72505ccf42fe6c58cb1c096f189ce60425d14c54/analysis/Figure3.Rmd" target="_blank">72505cc</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-10
</td>
<td>
Update index, SessionInfo, supplementary tables, Fig 3
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/e2712c2174370e09a641030a5487b6a18338704f/analysis/Figure3.Rmd" target="_blank">e2712c2</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-06
</td>
<td>
Reorder figures: add S4,S6,S13
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/e2712c2174370e09a641030a5487b6a18338704f/docs/Figure3.html" target="_blank">e2712c2</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-06
</td>
<td>
Reorder figures: add S4,S6,S13
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/9d3f606bb5ef1c1a9c3c587cddd0f2250b76198f/analysis/Figure3.Rmd" target="_blank">9d3f606</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-24
</td>
<td>
Add Supp Fig 9, reorder Fig 3 figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/9d3f606bb5ef1c1a9c3c587cddd0f2250b76198f/docs/Figure3.html" target="_blank">9d3f606</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-24
</td>
<td>
Add Supp Fig 9, reorder Fig 3 figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/dc39d3fb10aa7b5335c6182588b4ef4a906fb458/analysis/Figure3.Rmd" target="_blank">dc39d3f</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-21
</td>
<td>
Add supplementary figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/dc39d3fb10aa7b5335c6182588b4ef4a906fb458/docs/Figure3.html" target="_blank">dc39d3f</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-21
</td>
<td>
Add supplementary figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/e1f36dcdb578c47e206926e77af40fe0d2c7053d/analysis/Figure3.Rmd" target="_blank">e1f36dc</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-15
</td>
<td>
Add supp. figs and tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/e1f36dcdb578c47e206926e77af40fe0d2c7053d/docs/Figure3.html" target="_blank">e1f36dc</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-15
</td>
<td>
Add supp. figs and tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/de3561665de4eb4a67334ac71cee69954c0f14f5/analysis/Figure3.Rmd" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/de3561665de4eb4a67334ac71cee69954c0f14f5/docs/Figure3.html" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="pre-requisites" class="section level2">
<h2>Pre-requisites</h2>
<p>These packages are necessary to make the plots that make up Figure
3.</p>
<pre class="r"><code>library(&quot;reshape2&quot;)
library(&quot;cowplot&quot;)
library(&quot;tidyverse&quot;)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter()    masks stats::filter()
✖ dplyr::lag()       masks stats::lag()
✖ lubridate::stamp() masks cowplot::stamp()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(&quot;GenomicRanges&quot;)</code></pre>
<pre><code>Loading required package: stats4
Loading required package: BiocGenerics

Attaching package: &#39;BiocGenerics&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    intersect, setdiff, union

The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union

The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs

The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors

Attaching package: &#39;S4Vectors&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    second, second&lt;-

The following objects are masked from &#39;package:dplyr&#39;:

    first, rename

The following object is masked from &#39;package:tidyr&#39;:

    expand

The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: &#39;IRanges&#39;

The following object is masked from &#39;package:lubridate&#39;:

    %within%

The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice

The following object is masked from &#39;package:purrr&#39;:

    reduce

Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>library(&quot;gtools&quot;)
library(&quot;here&quot;)</code></pre>
<pre><code>here() starts at /Users/skoul/Documents/Projects/cfepi_wflow/noe2024</code></pre>
</div>
<div id="figure-3a-cpg-islands-coverage-mononucleosomal-reads"
class="section level2">
<h2>Figure 3A: CpG-islands coverage (mononucleosomal reads)</h2>
<pre class="r"><code>matrix &lt;- readRDS(here(&quot;data/cpg_cov_smallfragments.rds&quot;))
cpg &lt;- readRDS(here(&quot;data/cpgIslands.hg19.20001.rds&quot;))
geneid_ordered &lt;- order(cpg$beta)
cpg &lt;- cpg[geneid_ordered]
remove &lt;- which(cpg$amount &lt;=2)
cpg &lt;- cpg[-remove]
tracker &lt;- tibble(&quot;index&quot; = c(1:nrow(matrix)))
tracker &lt;- tracker[-remove,]

matrix_ordered &lt;- matrix
matrix_ordered &lt;- matrix_ordered[-remove,]

included_genes &lt;- which(rowSums(matrix_ordered) &gt;= 130000)
tracker &lt;- tracker[included_genes,]
cpg &lt;- cpg[included_genes]
matrix_ordered &lt;- matrix_ordered[which(rowSums(matrix_ordered) &gt;= 130000),]
matrix_ordered_long &lt;- reshape2::melt(matrix_ordered)
colnames(matrix_ordered_long) &lt;- c(&quot;gene_index&quot;, &quot;relative_position&quot;, &quot;coverage&quot;)
matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position - 251
matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position - 250
matrix_ordered_long$relative_position &lt;- matrix_ordered_long$relative_position * 10
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index * -1
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index - min(matrix_ordered_long$gene_index) +1
m &lt;- as_tibble(matrix_ordered_long) %&gt;%
    pivot_wider(id_cols=gene_index,
                names_from=relative_position,
                values_from=coverage)
m2 &lt;- m %&gt;% 
  dplyr::select(-gene_index) %&gt;%
  as.matrix()
sfun &lt;- function(x, k){
    s &lt;- runmean(Rle(x), k, endrule=&quot;constant&quot;)
    as.numeric(s)
}
m3 &lt;- apply(m2, 2, sfun, k=5)
m[, -1] &lt;- m3
x2 &lt;- m %&gt;%
    pivot_longer(cols=-1, names_to=&quot;relative_position&quot;,
                 values_to=&quot;coverage&quot;) %&gt;%
    mutate(relative_position=as.integer(relative_position))  %&gt;%
    mutate(method=&quot;smoothed&quot;)
pos &lt;- sort(c(seq(-2500, 2500, 1000), -10))
brks &lt;- match(pos, sort(unique(x2$relative_position)))
pos[pos == -10] &lt;- 0
downsample &lt;- sample(unique(x2$gene_index), 1000)
downsample &lt;- unique(x2$gene_index)
width &lt;- unit(10, &quot;cm&quot;)
#p1 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
p1 &lt;- x2 %&gt;%
    mutate(coverage=ifelse(coverage &gt; 600, 600, coverage),
           coverage=ifelse(coverage &lt; 100, 100, coverage),
           gene_index=as.integer(factor(gene_index)),
           x=as.integer(factor(relative_position))) %&gt;%
    ggplot(aes(x, gene_index)) +
    geom_tile(aes(fill=coverage), size=0) +
    scale_fill_gradientn(colours = c(&quot;#DD6627&quot;, &quot;#77C8DD&quot;, &quot;#004766&quot;)) +
    scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
    scale_y_continuous(expand = c(0,0)) +
    labs(x=&quot;Position relative to the middle of the CpG-island&quot;) +
    theme_classic(base_size=10) +
    theme(text = element_text(size=20),
          plot.title=element_blank(),
          panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=0.5),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.position=&quot;bottom&quot;,
          legend.box=&quot;horizontal&quot;,
          legend.spacing.x = unit(0, &quot;cm&quot;),
          legend.key.width = unit(0.217, &quot;snpc&quot;),
          plot.margin = margin(0,0.7,0.7,0.7, &quot;cm&quot;)) + 
  guides(fill = guide_colourbar(title.position=&quot;bottom&quot;, title.hjust = 0.5, title.vjust = 3, barheight = 0.5, frame.colour = &quot;black&quot;, frame.linewidth = 0.5, title=&quot;cfDNA coverage&quot;))</code></pre>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>Warning: The `size` argument of `element_rect()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre class="r"><code>values &lt;- data.frame(&quot;var_x&quot; = 1,
                     &quot;var_y&quot; = max(matrix_ordered_long$gene_index):1,
                     &quot;value&quot; = cpg$beta,
                     &quot;loc&quot; = paste0(seqnames(cpg), &quot;:&quot;, start(cpg), &quot;-&quot;, end(cpg)))
values$value &lt;- as.numeric(values$value)
matrix_ordered_long$cpg_loc &lt;- values[match(matrix_ordered_long$gene_index, values$var_y),]$loc
matrix_ordered_long$cpg_beta &lt;- values[match(matrix_ordered_long$gene_index, values$var_y),]$value
#saveRDS(matrix_ordered_long, &quot;/Users/mnoe/Documents/cpg_coverage.rds&quot;)
p2 &lt;- ggplot(values, aes(x = var_y, y = value)) +
  geom_bar(stat=&quot;identity&quot;, fill = &quot;black&quot;) +
  labs(x=&quot;CpG-islands in order of methylation (beta-value)&quot;, y=&quot;&quot;, title=&quot;&quot;) +
  theme_classic(base_size=10) + 
  theme(text = element_text(size=20),
        axis.title.y.right =  element_text(angle=90, vjust = 2),
        axis.line.x.bottom = element_blank(),
        axis.line.y.right = element_blank(),
        plot.title=element_blank(),
        axis.ticks.y = element_blank(),
        axis.ticks.x.bottom = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x.bottom = element_blank(),
        legend.position=&quot;bottom&quot;,
        plot.margin = margin(0,0,0,0, &quot;cm&quot;)) + 
  coord_flip() +
  guides(x.sec = &quot;axis&quot;, y.sec = &quot;axis&quot;) +
  scale_y_continuous(expand=c(0,0), limits = c(0, 1), breaks = seq(0, 1, by=1), labels=c(&quot;0.000000&quot; = &quot;0&quot;, &quot;1&quot; = &quot;1&quot;)) +
  scale_x_continuous(expand=c(0,0), position = &quot;top&quot;)


p3 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
  group_by(relative_position) %&gt;%
  summarise(ncoverage = sum(coverage)) %&gt;%
  ggplot(aes(relative_position, ncoverage)) +
  labs(x=&quot;&quot;, y=&quot;&quot;, title=&quot;Cumulative cfDNA coverage&quot;) +
  geom_line() +
  theme_classic(base_size=10) +
  theme(text = element_text(size=20),
        #axis.title.x.bottom = element_text(angle=0, vjust = 10),
        plot.title=element_text(hjust=0.5, vjust=0, size = 16),
        axis.ticks = element_blank(),
        axis.text = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        legend.position=&quot;bottom&quot;,
        legend.box=&quot;horizontal&quot;,
        legend.spacing.x = unit(0, &quot;cm&quot;),
        plot.margin = margin(0.2,0.7,0,0.7, &quot;cm&quot;)) +
  scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
  ggtitle(&quot;Cumulative cfDNA coverage&quot;)


p1_legend &lt;- get_legend(p1)
p1 &lt;- p1 + theme(legend.position=&#39;none&#39;)
plot_combi1 &lt;- plot_grid(p3, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi2 &lt;- plot_grid(p1, p2, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi3 &lt;- plot_grid(p1_legend, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_cpg_cov &lt;- plot_grid(plot_combi1, plot_combi2, plot_combi3, NULL, ncol=1, rel_heights = c(2,15,1,0.5))
title &lt;- ggdraw() + 
  draw_label(
    &quot;Coverage over CpG-islands&quot;,
    hjust = 0.5,
    size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
title &lt;- plot_grid(title, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_cpg_cov_title &lt;- plot_grid(title, plot_combi_cpg_cov, ncol = 1, rel_heights = c(1,15), align = &quot;h&quot;)</code></pre>
</div>
<div id="figure-3b-tss-coverage-mononucleosomal-reads"
class="section level2">
<h2>Figure 3B: TSS coverage (mononucleosomal reads)</h2>
<pre class="r"><code>matrix &lt;- readRDS(here(&quot;data/tss_cov_smallfragments.rds&quot;))
tss &lt;- read.table(file = here(&quot;data/transcriptAnno-GRCh37.75.tsv&quot;))
tss &lt;- tss[which(tss$V2 %in% c(1:22)),]
tss$V2 &lt;- paste0(&quot;chr&quot;, tss$V2)
tissue_key &lt;- read.table(here(&quot;data/tissue_key.tsv&quot;),as.is=T,header=T,sep=&quot;\t&quot;,quote=&quot;\&quot;&quot;)
conv &lt;- read.table(here(&#39;data/labels.txt&#39;),as.is=T,header=T,sep=&quot;\t&quot;,quote=&quot;\&quot;&quot;)
rna &lt;- read.table(here(&#39;data/RNAtable.tsv.gz&#39;),as.is=T,header=T)
express &lt;- conv[which(conv$Category %in% c(&quot;Myeloid&quot;)),]$RName
express_df &lt;- rna[c(&quot;GeneID&quot;, express)]
express_df$mean &lt;- rowMeans(express_df[,c(2:(length(express) + 1))], na.rm=T)
express_df &lt;- express_df[order(-express_df$mean),]
geneid_ordered &lt;- express_df$GeneID
tss$expression &lt;- 0
tss &lt;- tss[which(tss$V1 %in% express_df$GeneID),]
tss[match(express_df[which(express_df$GeneID %in% tss$V1),]$GeneID, tss$V1),]$expression &lt;- express_df[which(express_df$GeneID %in% tss$V1),]$mean
tss &lt;- tss[order(-tss$expression),]
colnames(tss) &lt;- c(&quot;transcript&quot;, &quot;seqnames&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;expression&quot;)

matrix_ordered &lt;- matrix
included_genes &lt;- tss$transcript
included_genes &lt;- included_genes[which(rowSums(matrix_ordered) &gt;= 130000)]
matrix_ordered &lt;- matrix_ordered[which(rowSums(matrix_ordered) &gt;= 130000),]
matrix_ordered_long &lt;- melt(matrix_ordered)
colnames(matrix_ordered_long) &lt;- c(&quot;gene_index&quot;, &quot;relative_position&quot;, &quot;coverage&quot;)
matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position - 251
matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position - 250
matrix_ordered_long$relative_position &lt;- matrix_ordered_long$relative_position * 10
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index * -1
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index - min(matrix_ordered_long$gene_index) +1
m &lt;- as_tibble(matrix_ordered_long) %&gt;%
    pivot_wider(id_cols=gene_index,
                names_from=relative_position,
                values_from=coverage)
m2 &lt;- m %&gt;% 
  dplyr::select(-gene_index) %&gt;%
  as.matrix()
sfun &lt;- function(x, k){
    s &lt;- runmean(Rle(x), k, endrule=&quot;constant&quot;)
    as.numeric(s)
}
m3 &lt;- apply(m2, 2, sfun, k=5)
m[, -1] &lt;- m3
x2 &lt;- m %&gt;%
    pivot_longer(cols=-1, names_to=&quot;relative_position&quot;,
                 values_to=&quot;coverage&quot;) %&gt;%
    mutate(relative_position=as.integer(relative_position))  %&gt;%
    mutate(method=&quot;smoothed&quot;)
pos &lt;- sort(c(seq(-2500, 2500, 1000), -10))
brks &lt;- match(pos, sort(unique(x2$relative_position)))
pos[pos == -10] &lt;- 0
downsample &lt;- sample(unique(x2$gene_index), 1000)
downsample &lt;- unique(x2$gene_index)
width &lt;- unit(10, &quot;cm&quot;)
#p1 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
p1 &lt;- x2 %&gt;%
    mutate(coverage=ifelse(coverage &gt; 600, 600, coverage),
           coverage=ifelse(coverage &lt; 100, 100, coverage),
           gene_index=as.integer(factor(gene_index)),
           x=as.integer(factor(relative_position))) %&gt;%
    ggplot(aes(x, gene_index)) +
    geom_tile(aes(fill=coverage), size=0) +
    scale_fill_gradientn(colours = c(&quot;#DD6627&quot;, &quot;#77C8DD&quot;, &quot;#004766&quot;)) +
    scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
    scale_y_continuous(expand = c(0,0)) +
    labs(x=&quot;Position relative to TSS&quot;) +
    theme_classic(base_size=10) +
    theme(text = element_text(size=20),
          plot.title=element_blank(),
          panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=0.5),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.position=&quot;bottom&quot;,
          legend.box=&quot;horizontal&quot;,
          legend.spacing.x = unit(0, &quot;cm&quot;),
          legend.key.width = unit(0.217, &quot;snpc&quot;),
          plot.margin = margin(0,0.7,0.7,0.7, &quot;cm&quot;)) + 
  guides(fill = guide_colourbar(title.position=&quot;bottom&quot;, title.hjust = 0.5, title.vjust = 3, barheight = 0.5, frame.colour = &quot;black&quot;, frame.linewidth = 0.5, title=&quot;cfDNA coverage&quot;))
values &lt;- data.frame(&quot;var_x&quot; = 1,
                     &quot;var_y&quot; = c(max(matrix_ordered_long$gene_index):1),
                     &quot;value&quot; = express_df[which(express_df$GeneID %in% included_genes),]$mean,
                     &quot;gene&quot; = express_df[which(express_df$GeneID %in% included_genes),]$GeneID)
matrix_ordered_long$transcript_id &lt;- values[match(matrix_ordered_long$gene_index, values$var_y),]$gene
matrix_ordered_long$rna_expression &lt;- values[match(matrix_ordered_long$gene_index, values$var_y),]$value
#saveRDS(matrix_ordered_long, &quot;/Users/mnoe/Documents/expression_coverage.rds&quot;)
p2 &lt;- ggplot(values[which(values$var_y %in% downsample),], aes(x = var_y, y = log(value + 1))) +
  geom_bar(stat=&quot;identity&quot;, fill = &quot;black&quot;) +
  labs(x=&quot;Genes in order of expression (TPM)&quot;, y=&quot;&quot;, title=&quot;&quot;) +
  theme_classic(base_size=10) + 
  theme(text = element_text(size=20),
        axis.title.y.right =  element_text(angle=90, vjust = 4),
        axis.line.x.bottom = element_blank(),
        axis.line.y.right = element_blank(),
        plot.title=element_blank(),
        axis.ticks.y = element_blank(),
        axis.ticks.x.bottom = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x.bottom = element_blank(),
        legend.position=&quot;bottom&quot;,
        plot.margin = margin(0,0,0,0, &quot;cm&quot;)) + 
  coord_flip() +
  guides(x.sec = &quot;axis&quot;, y.sec = &quot;axis&quot;) +
  scale_y_continuous(expand=c(0,0), limits = c(0, 7.835816), breaks = seq(0, 7.835816, by=7.835816), labels=c(&quot;0.000000&quot; = &quot;0&quot;, &quot;7.835816&quot; = &quot;2528&quot;)) +
  scale_x_continuous(expand=c(0,0), position = &quot;top&quot;)


p3 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
  group_by(relative_position) %&gt;%
  summarise(ncoverage = sum(coverage)) %&gt;%
  ggplot(aes(relative_position, ncoverage)) +
  labs(x=&quot;&quot;, y=&quot;&quot;, title=&quot;Cumulative cfDNA coverage&quot;) +
  geom_line() +
  theme_classic(base_size=10) +
  theme(text = element_text(size=20),
        #axis.title.x.bottom = element_text(angle=0, vjust = 10),
        plot.title=element_text(hjust=0.5, vjust=0, size = 16),
        axis.ticks = element_blank(),
        axis.text = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        legend.position=&quot;bottom&quot;,
        legend.box=&quot;horizontal&quot;,
        legend.spacing.x = unit(0, &quot;cm&quot;),
        plot.margin = margin(0.2,0.7,0,0.7, &quot;cm&quot;)) +
  scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
  ggtitle(&quot;Cumulative cfDNA coverage&quot;)


p1_legend &lt;- get_legend(p1)
p1 &lt;- p1 + theme(legend.position=&#39;none&#39;)
#p2_legend &lt;- get_legend(p2)
#p2 &lt;- p2 + theme(legend.position=&#39;none&#39;)
#plot_combi &lt;- plot_grid(plot_grid(p1, p2, ncol=2, nrow=1,align = &quot;h&quot;, rel_widths = c(10,2)),
#          plot_grid(NULL, p1_legend, p2_legend, NULL, ncol = 1, nrow = 4, align=&quot;v&quot;, axis=&quot;b&quot;), #rel_widths = c(10,1))
plot_combi1 &lt;- plot_grid(p3, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi2 &lt;- plot_grid(p1, p2, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (`position_stack()`).</code></pre>
<pre class="r"><code>plot_combi3 &lt;- plot_grid(p1_legend, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_tss_cov &lt;- plot_grid(plot_combi1, plot_combi2, plot_combi3, NULL, ncol=1, rel_heights = c(2,15,1,0.5))
title &lt;- ggdraw() + 
  draw_label(
    &quot;Coverage over transcription start sites&quot;,
    hjust = 0.5,
    size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
title &lt;- plot_grid(title, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_tss_cov_title &lt;- plot_grid(title, plot_combi_tss_cov, ncol = 1, rel_heights = c(1,15))

t &lt;- readRDS(here(&quot;data/tss_cpg_connect.rds&quot;))
cvs &lt;- c()
for (i in c(1:501)) {
  data &lt;- matrix_ordered[,i]
  cv &lt;- cor(data, c(1:length(data)))
  cvs &lt;- c(cvs, cv)
}
#plot(c(1:501), cvs)
col &lt;- which(cvs==max(cvs))
t$cov &lt;- NA
t$cov &lt;- matrix_ordered[match(t$trans, values$gene),col]</code></pre>
</div>
<div id="figure-3-combine-figure-3a-figure-3b" class="section level2">
<h2>Figure 3: combine Figure 3A, Figure 3B</h2>
<pre class="r"><code>small_cov &lt;- plot_grid(plot_combi_cpg_cov_title, NULL, plot_combi_tss_cov_title , rel_widths = c(1,0.1,1), labels=c(&quot;a&quot;, &quot;&quot;, &quot;b&quot;), nrow=1, label_size = 20)
#ggsave(here(&quot;docs/figure/Fig_3AB.pdf&quot;), plot = small_cov, device = &quot;pdf&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=600, bg = &quot;white&quot;, scale = 0.78)
ggsave(here(&quot;docs/figure/Fig_3AB.jpg&quot;), plot = small_cov, device = &quot;jpg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=600, bg = &quot;white&quot;, scale = 0.82)</code></pre>
</div>
<div
id="figure-3c-cpg-islands-cfdna-fragment-size-mononucleosomal-reads"
class="section level2">
<h2>Figure 3C: CpG-islands cfDNA fragment size (mononucleosomal
reads)</h2>
<pre class="r"><code>colorz &lt;- c(&quot;#DD6627&quot;, &quot;#77C8DD&quot;, &quot;#004766&quot;)
matrix &lt;- readRDS(here(&quot;data/cpg_size_smallfragments.rds&quot;))
cpg &lt;- readRDS(here(&quot;data/cpgIslands.hg19.20001.rds&quot;))
geneid_ordered &lt;- order(cpg$beta)
cpg &lt;- cpg[geneid_ordered]
remove &lt;- which(cpg$amount &lt;=2)
cpg &lt;- cpg[-remove]
tracker &lt;- tibble(&quot;index&quot; = c(1:nrow(matrix)))
tracker &lt;- tracker[-remove,]

matrix_ordered &lt;- matrix
matrix_ordered &lt;- matrix_ordered[-remove,]

included_genes &lt;- which(rowSums(matrix_ordered) &gt;= 75000)
tracker &lt;- tracker[included_genes,]
cpg &lt;- cpg[included_genes]
matrix_ordered &lt;- matrix_ordered[which(rowSums(matrix_ordered) &gt;= 75000),]
matrix_ordered_long &lt;- melt(matrix_ordered)
colnames(matrix_ordered_long) &lt;- c(&quot;gene_index&quot;, &quot;relative_position&quot;, &quot;size&quot;)
matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position - 251
matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position - 250
matrix_ordered_long$relative_position &lt;- matrix_ordered_long$relative_position * 10
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index * -1
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index - min(matrix_ordered_long$gene_index) +1
m &lt;- as_tibble(matrix_ordered_long) %&gt;%
    pivot_wider(id_cols=gene_index,
                names_from=relative_position,
                values_from=size)
m2 &lt;- m %&gt;% 
  dplyr::select(-gene_index) %&gt;%
  as.matrix()
sfun &lt;- function(x, k){
    s &lt;- runmean(Rle(x), k, endrule=&quot;constant&quot;)
    as.numeric(s)
}
m3 &lt;- apply(m2, 2, sfun, k=5)
m[, -1] &lt;- m3
x2 &lt;- m %&gt;%
    pivot_longer(cols=-1, names_to=&quot;relative_position&quot;,
                 values_to=&quot;size&quot;) %&gt;%
    mutate(relative_position=as.integer(relative_position))  %&gt;%
    mutate(method=&quot;smoothed&quot;)
pos &lt;- sort(c(seq(-2500, 2500, 1000), -10))
brks &lt;- match(pos, sort(unique(x2$relative_position)))
pos[pos == -10] &lt;- 0
downsample &lt;- sample(unique(x2$gene_index), 1000)
downsample &lt;- unique(x2$gene_index)
width &lt;- unit(10, &quot;cm&quot;)
x2[which(is.na(x2$size)),]$size &lt;- mean(x2[-which(is.na(x2$size)),]$size)
#tibfinal &lt;- tibble()
#for(i in unique(x2$relative_position)) {
#  tib &lt;- tibble(&quot;diff&quot; = (mean(x2[which(x2$relative_position %in% c(i)),][c(1:1000),]$size) - mean(x2[which(x2$relative_position %in% c(i)),][c((length(unique(x2$gene_index))-1000):(length(unique(x2$gene_index)))),]$size)),
#         &quot;pos&quot; = i)
#  tibfinal &lt;- rbind(tibfinal, tib)
#}
p1 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
    mutate(size=ifelse(size &gt; 176, 176, size),
           size=ifelse(size &lt; 160, 160, size),
           gene_index=as.integer(factor(gene_index)),
           x=as.integer(factor(relative_position))) %&gt;%
    ggplot(aes(x, gene_index)) +
    geom_tile(aes(fill=size), size=0) +
    scale_fill_gradientn(colours = colorz) +
    scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
    scale_y_continuous(expand = c(0,0)) +
    labs(x=&quot;Position relative to the middle of the CpG-island&quot;) +
    theme_classic(base_size=10) +
    theme(text = element_text(size=20),
          plot.title=element_blank(),
          panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=0.5),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.position=&quot;bottom&quot;,
          legend.box=&quot;horizontal&quot;,
          legend.spacing.x = unit(0, &quot;cm&quot;),
          legend.key.width = unit(0.217, &quot;snpc&quot;),
          plot.margin = margin(0,0.7,0.7,0.7, &quot;cm&quot;)) + 
  guides(fill = guide_colourbar(title.position=&quot;bottom&quot;, title.hjust = 0.5, title.vjust = 3, barheight = 0.5, frame.colour = &quot;black&quot;, frame.linewidth = 0.5, title=&quot;cfDNA-fragment size&quot;))
values &lt;- data.frame(&quot;var_x&quot; = 1,
                     &quot;var_y&quot; = max(matrix_ordered_long$gene_index):1,
                     &quot;value&quot; = cpg$beta)

values$value &lt;- as.numeric(values$value)
p2 &lt;- ggplot(values, aes(x = var_y, y = value)) +
  geom_bar(stat=&quot;identity&quot;, fill = &quot;black&quot;) +
  labs(x=&quot;CpG-islands in order of methylation (beta-value)&quot;, y=&quot;&quot;, title=&quot;&quot;) +
  theme_classic(base_size=10) + 
  theme(text = element_text(size=20),
        axis.title.y.right =  element_text(angle=90, vjust = 2),
        axis.line.x.bottom = element_blank(),
        axis.line.y.right = element_blank(),
        plot.title=element_blank(),
        axis.ticks.y = element_blank(),
        axis.ticks.x.bottom = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x.bottom = element_blank(),
        legend.position=&quot;bottom&quot;,
        plot.margin = margin(0,0,0,0, &quot;cm&quot;)) + 
  coord_flip() +
  guides(x.sec = &quot;axis&quot;, y.sec = &quot;axis&quot;) +
  scale_y_continuous(expand=c(0,0), limits = c(0, 1), breaks = seq(0, 1, by=1), labels=c(&quot;0.000000&quot; = &quot;0&quot;, &quot;1&quot; = &quot;1&quot;)) +
  scale_x_continuous(expand=c(0,0), position = &quot;top&quot;)


p3 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
  group_by(relative_position) %&gt;%
  summarise(nsize = sum(size)) %&gt;%
  ggplot(aes(relative_position, nsize)) +
  labs(x=&quot;&quot;, y=&quot;&quot;, title=&quot;average cfDNA-fragment size&quot;) +
  geom_line() +
  theme_classic(base_size=10) +
  theme(text = element_text(size=20),
        #axis.title.x.bottom = element_text(angle=0, vjust = 10),
        plot.title=element_text(hjust=0.5, vjust=0, size = 16),
        axis.ticks = element_blank(),
        axis.text = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        legend.position=&quot;bottom&quot;,
        legend.box=&quot;horizontal&quot;,
        legend.spacing.x = unit(0, &quot;cm&quot;),
        plot.margin = margin(0.2,0.7,0,0.7, &quot;cm&quot;)) +
  scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
  ggtitle(&quot;cfDNA-fragment size&quot;)


p1_legend &lt;- get_legend(p1)
p1 &lt;- p1 + theme(legend.position=&#39;none&#39;)
#p2_legend &lt;- get_legend(p2)
#p2 &lt;- p2 + theme(legend.position=&#39;none&#39;)
#plot_combi &lt;- plot_grid(plot_grid(p1, p2, ncol=2, nrow=1,align = &quot;h&quot;, rel_widths = c(10,2)),
#          plot_grid(NULL, p1_legend, p2_legend, NULL, ncol = 1, nrow = 4, align=&quot;v&quot;, axis=&quot;b&quot;), #rel_widths = c(10,1))
plot_combi1 &lt;- plot_grid(p3, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi2 &lt;- plot_grid(p1, p2, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi3 &lt;- plot_grid(p1_legend, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_cpg_size &lt;- plot_grid(plot_combi1, plot_combi2, plot_combi3, NULL, ncol=1, rel_heights = c(2,15,1,0.5))
title &lt;- ggdraw() + 
  draw_label(
    &quot;cfDNA-fragment size over CpG-islands&quot;,
    hjust = 0.5,
    size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
title &lt;- plot_grid(title, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_cpg_size_title &lt;- plot_grid(title, plot_combi_cpg_size, ncol = 1, rel_heights = c(1,15))</code></pre>
</div>
<div id="figure-3d-tss-cfdna-fragment-size-mononucleosomal-reads"
class="section level2">
<h2>Figure 3D: TSS cfDNA fragment size (mononucleosomal reads)</h2>
<pre class="r"><code>colorz &lt;- c(&quot;#DD6627&quot;, &quot;#77C8DD&quot;, &quot;#004766&quot;)
matrix &lt;- readRDS(here(&quot;data/tss_size_smallfragments.rds&quot;))
tss &lt;- read.table(file = here(&quot;data/transcriptAnno-GRCh37.75.tsv&quot;))
tss &lt;- tss[which(tss$V2 %in% c(1:22)),]
tss$V2 &lt;- paste0(&quot;chr&quot;, tss$V2)
tissue_key &lt;- read.table(here(&quot;data/tissue_key.tsv&quot;),as.is=T,header=T,sep=&quot;\t&quot;,quote=&quot;\&quot;&quot;)
conv &lt;- read.table(here(&#39;data/labels.txt&#39;),as.is=T,header=T,sep=&quot;\t&quot;,quote=&quot;\&quot;&quot;)
rna &lt;- read.table(here(&#39;data/RNAtable.tsv.gz&#39;),as.is=T,header=T)
express &lt;- conv[which(conv$Category %in% c(&quot;Myeloid&quot;)),]$RName
express_df &lt;- rna[c(&quot;GeneID&quot;, express)]
express_df$mean &lt;- rowMeans(express_df[,c(2:(length(express) + 1))], na.rm=T)
express_df &lt;- express_df[order(-express_df$mean),]
geneid_ordered &lt;- express_df$GeneID
tss$expression &lt;- 0
tss &lt;- tss[which(tss$V1 %in% express_df$GeneID),]
tss[match(express_df[which(express_df$GeneID %in% tss$V1),]$GeneID, tss$V1),]$expression &lt;- express_df[which(express_df$GeneID %in% tss$V1),]$mean
tss &lt;- tss[order(-tss$expression),]
colnames(tss) &lt;- c(&quot;transcript&quot;, &quot;seqnames&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;expression&quot;)

matrix_ordered &lt;- matrix
included_genes &lt;- tss$transcript
included_genes &lt;- included_genes[which(rowSums(matrix_ordered) &gt;= 75000)]
matrix_ordered &lt;- matrix_ordered[which(rowSums(matrix_ordered) &gt;= 75000),]
matrix_ordered_long &lt;- melt(matrix_ordered)
colnames(matrix_ordered_long) &lt;- c(&quot;gene_index&quot;, &quot;relative_position&quot;, &quot;size&quot;)
matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &lt;= 250),]$relative_position - 251
matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position &lt;- matrix_ordered_long[which(matrix_ordered_long$relative_position &gt; 250),]$relative_position - 250
matrix_ordered_long$relative_position &lt;- matrix_ordered_long$relative_position * 10
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index * -1
matrix_ordered_long$gene_index &lt;- matrix_ordered_long$gene_index - min(matrix_ordered_long$gene_index) +1
m &lt;- as_tibble(matrix_ordered_long) %&gt;%
    pivot_wider(id_cols=gene_index,
                names_from=relative_position,
                values_from=size)
m2 &lt;- m %&gt;% 
  dplyr::select(-gene_index) %&gt;%
  as.matrix()
sfun &lt;- function(x, k){
    s &lt;- runmean(Rle(x), k, endrule=&quot;constant&quot;)
    as.numeric(s)
}
m3 &lt;- apply(m2, 2, sfun, k=5)
m[, -1] &lt;- m3
x2 &lt;- m %&gt;%
    pivot_longer(cols=-1, names_to=&quot;relative_position&quot;,
                 values_to=&quot;size&quot;) %&gt;%
    mutate(relative_position=as.integer(relative_position))  %&gt;%
    mutate(method=&quot;smoothed&quot;)
pos &lt;- sort(c(seq(-2500, 2500, 1000), -10))
brks &lt;- match(pos, sort(unique(x2$relative_position)))
pos[pos == -10] &lt;- 0
downsample &lt;- sample(unique(x2$gene_index), 1000)
downsample &lt;- unique(x2$gene_index)
width &lt;- unit(10, &quot;cm&quot;)
x2[which(is.na(x2$size)),]$size &lt;- mean(x2[-which(is.na(x2$size)),]$size)

mean(x2[which(abs(x2$relative_position) %in% c(800:1000) &amp; (x2$gene_index %in% unique(x2$gene_index)[1:1000])),]$size)</code></pre>
<pre><code>[1] 165.5414</code></pre>
<pre class="r"><code>mean(x2[which(abs(x2$relative_position) %in% c(800:1000) &amp; (x2$gene_index %in% unique(x2$gene_index)[(length(unique(x2$gene_index))-999):length(unique(x2$gene_index))])),]$size)</code></pre>
<pre><code>[1] 168.9282</code></pre>
<pre class="r"><code>mean(x2[which(x2$relative_position %in% c(-800:-1000) &amp; (x2$gene_index %in% unique(x2$gene_index)[1:1000])),]$size)</code></pre>
<pre><code>[1] 164.9494</code></pre>
<pre class="r"><code>mean(x2[which(x2$relative_position %in% c(-800:-1000) &amp; (x2$gene_index %in% unique(x2$gene_index)[(length(unique(x2$gene_index))-999):length(unique(x2$gene_index))])),]$size)</code></pre>
<pre><code>[1] 168.7399</code></pre>
<pre class="r"><code>p1 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
    mutate(size=ifelse(size &gt; 176, 176, size),
           size=ifelse(size &lt; 160, 160, size),
           gene_index=as.integer(factor(gene_index)),
           x=as.integer(factor(relative_position))) %&gt;%
    ggplot(aes(x, gene_index)) +
    geom_tile(aes(fill=size), size=0) +
    scale_fill_gradientn(colours = colorz) +
    scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
    scale_y_continuous(expand = c(0,0)) +
    labs(x=&quot;Position relative to TSS&quot;) +
    theme_classic(base_size=10) +
    theme(text = element_text(size=20),
          plot.title=element_blank(),
          panel.border = element_rect(colour = &quot;black&quot;, fill=NA, size=0.5),
          axis.ticks.y = element_blank(),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.position=&quot;bottom&quot;,
          legend.box=&quot;horizontal&quot;,
          legend.spacing.x = unit(0, &quot;cm&quot;),
          legend.key.width = unit(0.217, &quot;snpc&quot;),
          plot.margin = margin(0,0.7,0.7,0.7, &quot;cm&quot;)) + 
  guides(fill = guide_colourbar(title.position=&quot;bottom&quot;, title.hjust = 0.5, title.vjust = 3, barheight = 0.5, frame.colour = &quot;black&quot;, frame.linewidth = 0.5, title=&quot;cfDNA-fragment size&quot;))
values &lt;- data.frame(&quot;var_x&quot; = 1,
                     &quot;var_y&quot; = c(max(matrix_ordered_long$gene_index):1),
                     &quot;value&quot; = express_df[which(express_df$GeneID %in% included_genes),]$mean,
                     &quot;gene&quot; = express_df[which(express_df$GeneID %in% included_genes),]$GeneID)

p2 &lt;- ggplot(values[which(values$var_y %in% downsample),], aes(x = var_y, y = log(value + 1))) +
  geom_bar(stat=&quot;identity&quot;, fill = &quot;black&quot;) +
  labs(x=&quot;Genes in order of expression (TPM)&quot;, y=&quot;&quot;, title=&quot;&quot;) +
  theme_classic(base_size=10) + 
  theme(text = element_text(size=20),
        axis.title.y.right =  element_text(angle=90, vjust = 4),
        axis.line.x.bottom = element_blank(),
        axis.line.y.right = element_blank(),
        plot.title=element_blank(),
        axis.ticks.y = element_blank(),
        axis.ticks.x.bottom = element_blank(),
        axis.text.y = element_blank(),
        axis.text.x.bottom = element_blank(),
        legend.position=&quot;bottom&quot;,
        plot.margin = margin(0,0,0,0, &quot;cm&quot;)) + 
  coord_flip() +
  guides(x.sec = &quot;axis&quot;, y.sec = &quot;axis&quot;) +
  scale_y_continuous(expand=c(0,0), limits = c(0, 7.835816), breaks = seq(0, 7.835816, by=7.835816), labels=c(&quot;0.000000&quot; = &quot;0&quot;, &quot;7.835816&quot; = &quot;2528&quot;)) +
  scale_x_continuous(expand=c(0,0), position = &quot;top&quot;)


p3 &lt;- x2[which(x2$gene_index %in% downsample),] %&gt;%
  group_by(relative_position) %&gt;%
  summarise(nsize = sum(size)) %&gt;%
  ggplot(aes(relative_position, nsize)) +
  labs(x=&quot;&quot;, y=&quot;&quot;, title=&quot;average cfDNA-fragment size&quot;) +
  geom_line() +
  theme_classic(base_size=10) +
  theme(text = element_text(size=20),
        #axis.title.x.bottom = element_text(angle=0, vjust = 10),
        plot.title=element_text(hjust=0.5, vjust=0, size = 16),
        axis.ticks = element_blank(),
        axis.text = element_blank(),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        legend.position=&quot;bottom&quot;,
        legend.box=&quot;horizontal&quot;,
        legend.spacing.x = unit(0, &quot;cm&quot;),
        plot.margin = margin(0.2,0.7,0,0.7, &quot;cm&quot;)) +
  scale_x_continuous(breaks=brks,
                       labels=pos, 
                       expand = c(0,0)) +
  ggtitle(&quot;cfDNA-fragment size&quot;)


p1_legend &lt;- get_legend(p1)
p1 &lt;- p1 + theme(legend.position=&#39;none&#39;)
#p2_legend &lt;- get_legend(p2)
#p2 &lt;- p2 + theme(legend.position=&#39;none&#39;)
#plot_combi &lt;- plot_grid(plot_grid(p1, p2, ncol=2, nrow=1,align = &quot;h&quot;, rel_widths = c(10,2)),
#          plot_grid(NULL, p1_legend, p2_legend, NULL, ncol = 1, nrow = 4, align=&quot;v&quot;, axis=&quot;b&quot;), #rel_widths = c(10,1))
plot_combi1 &lt;- plot_grid(p3, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi2 &lt;- plot_grid(p1, p2, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (`position_stack()`).</code></pre>
<pre class="r"><code>plot_combi3 &lt;- plot_grid(p1_legend, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_tss_size &lt;- plot_grid(plot_combi1, plot_combi2, plot_combi3, NULL, ncol=1, rel_heights = c(2,15,1, 0.5))
title &lt;- ggdraw() + 
  draw_label(
    &quot;cfDNA-fragment size over transcription start sites&quot;,
    hjust = 0.5,
    size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
title &lt;- plot_grid(title, NULL, ncol=2, nrow=1, rel_widths = c(10,1), align = &quot;h&quot;)
plot_combi_tss_size_title &lt;- plot_grid(title, plot_combi_tss_size, ncol = 1, rel_heights = c(1,15))


cvs &lt;- c()
for (i in c(1:501)) {
  data &lt;- matrix_ordered[,i]
  cv &lt;- cor(data, c(1:length(data)))
  cvs &lt;- c(cvs, cv)
}
#plot(c(1:501), cvs)
col &lt;- which(cvs==max(cvs))
t$size &lt;- NA
t$size &lt;- matrix_ordered[match(t$trans, values$gene), col]</code></pre>
</div>
<div id="figure-3-combine-figure-3c-figure-3d" class="section level2">
<h2>Figure 3: combine Figure 3C, Figure 3D</h2>
<pre class="r"><code>small_size &lt;- plot_grid(plot_combi_cpg_size_title, NULL, plot_combi_tss_size_title, rel_widths = c(1,0.1,1), labels=c(&quot;c&quot;, &quot;&quot;, &quot;d&quot;), nrow=1,label_size = 20)
#ggsave(here(&quot;docs/figure/Fig_3CD.pdf&quot;), plot = small_size, device = &quot;pdf&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=600, bg = &quot;white&quot;, scale = 0.78)
ggsave(here(&quot;docs/figure/Fig_3CD.jpg&quot;), plot = small_size, device = &quot;jpg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=600, bg = &quot;white&quot;, scale = 0.82)</code></pre>
</div>
<div id="figure-3g-mouse-data" class="section level2">
<h2>Figure 3G: Mouse data</h2>
<pre class="r"><code>t &lt;- readRDS(here(&quot;data/mouse_frag.rds&quot;))
#t &lt;- t[which(t$top == 1000),]
t &lt;- t[-which(t$id == &quot;PGDX27350P&quot;),] ### remove a file with problems
t &lt;- t[-which(t$id == &quot;PGDX27354P&quot;),] ### remove a file with problems
t &lt;- t[-which(t$inplant == &quot;brain&quot;),]
t[which(t$mut == &quot;MT&quot;),]$mut &lt;- &quot;mutant&quot;
t[which(t$mut == &quot;WT&quot;),]$mut &lt;- &quot;wild-type&quot;
t2 &lt;- t %&gt;%
  group_by(mut,id,top) %&gt;%
  summarise(rna_min = (sum(rna_plus) / sum(all)) / rna_plus_length,
            rna_plus = (sum(rna_min) / sum(all)) / rna_min_length,
            met_min = (sum(met_min) / sum(all)) / met_min_length,
            met_plus = (sum(met_plus) / sum(all)) / met_plus_length,
            met_is_min = (sum(met_is_min) / sum(all)) / met_is_min_length,
            met_is_plus = (sum(met_is_plus) / sum(all)) / met_is_plus_length,
            ann_is_min = (sum(ann_cpg_is_min) / sum(all)) / ann_cpg_is_min_length,
            ann_is_plus = (sum(ann_cpg_is_plus) / sum(all)) / ann_cpg_is_plus_length,
            idh_all_min = (sum(idh_all_min) / sum(all)) / idh_all_min_length,
            idh_all_plus = (sum(idh_all_plus) / sum(all)) / idh_all_plus_length,
            idh_all_is_min = (sum(idh_all_is_plus) / sum(all)) / idh_all_is_plus_length,
            idh_all_is_plus = (sum(idh_all_is_min) / sum(all)) / idh_all_is_min_length,
            all = sum(all))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;mut&#39;, &#39;id&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>t2l &lt;- reshape2::melt(t2,id.vars = c(&quot;mut&quot;, &quot;id&quot;, &quot;top&quot;))

t2l_sum_met &lt;- t2l[which(t2l$variable %in% c(&#39;met_is_min&#39;, &#39;met_is_plus&#39;)),]
t2l_sum_met$group &lt;- &quot;Methylation&quot; 
t2l_sum_rna &lt;- t2l[which((t2l$variable %in% c(&#39;rna_min&#39;, &#39;rna_plus&#39;))),]
t2l_sum_rna$group &lt;- &quot;Expression&quot;
t2l_sum &lt;- rbind (t2l_sum_met, t2l_sum_rna)
t2l_sum$group &lt;- factor(t2l_sum$group, levels=c(&quot;Methylation&quot;, &quot;Expression&quot;))
t2l_sum_max &lt;- t2l_sum %&gt;%
  group_by(top, variable) %&gt;%
  summarise(max = max(value))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;top&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>t2l_sum$max &lt;- t2l_sum_max[match(paste0(t2l_sum$top, t2l_sum$variable), paste0(t2l_sum_max$top, t2l_sum_max$variable)),]$max
t2l_sum$norm_value &lt;- t2l_sum$value / t2l_sum$max

perms  &lt;- permutations(2,6,c(0,1),repeats=TRUE)
keep &lt;- perms[which(rowSums(perms) == 3),] 

#permutation using medians
all &lt;- tibble()
for (j in c(1:nrow(keep))) {
  print(j)
  P &lt;- j
  for (i in c(500, 1000, 2000, 3000, 4000, 5000)) {
    t2l_sum_temp &lt;- t2l_sum[which(t2l_sum$top == i),]
    for (q in unique(t2l_sum_temp$variable)) {
      t2l_sum_temp_temp &lt;-  t2l_sum_temp[which(t2l_sum_temp$variable == q),]
      pindex &lt;- keep[j,] + 1
      permutation &lt;- c(&quot;mutant&quot;, &quot;wild-type&quot;)[pindex]
      t2l_sum_temp_temp$mut &lt;- permutation
      #t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;mutant&quot;),]$value &lt;- sample(t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;mutant&quot;),]$value, size = 3, replace = T)
      #t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;wild-type&quot;),]$value &lt;- sample(t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;wild-type&quot;),]$value, size = 3, replace = T)
      sum &lt;- t2l_sum_temp_temp %&gt;%
        group_by(mut) %&gt;%
        summarise(med = median(norm_value))
      if (q %in% c(&quot;met_is_min&quot;, &quot;rna_min&quot;)) {
        sum_temp &lt;- tibble(&quot;P&quot; = P,
                           &quot;top&quot; = i,
                           &quot;group&quot; = q,
                           &quot;direction&quot; = sign(sum[which(sum$mut == &quot;wild-type&quot;),]$med - sum[which(sum$mut == &quot;mutant&quot;),]$med))
      }
      else {
          sum_temp &lt;- tibble(&quot;P&quot; = P,
                           &quot;top&quot; = i,
                           &quot;group&quot; = q,
                           &quot;direction&quot; = sign(sum[which(sum$mut == &quot;mutant&quot;),]$med - sum[which(sum$mut == &quot;wild-type&quot;),]$med))
        }
        all &lt;- rbind(all, sum_temp)
      }
    }
}</code></pre>
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20</code></pre>
<pre class="r"><code>all[which(all$direction == -1),]$direction &lt;- 0
all_sum &lt;- all %&gt;%
  group_by(P) %&gt;%
  summarise(count_positive = sum(direction))
hist(all_sum$count_positive, breaks = 30)</code></pre>
<p><img src="figure/Figure3.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/65ac5a530490a0f31dfd2a3aef73903ed07b27bd/docs/figure/Figure3.Rmd/unnamed-chunk-8-1.png" target="_blank">65ac5a5</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/e2712c2174370e09a641030a5487b6a18338704f/docs/figure/Figure3.Rmd/unnamed-chunk-8-1.png" target="_blank">e2712c2</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>all_sum_median &lt;- all_sum
hist(all_sum_median$count_positive, breaks = 30)</code></pre>
<p><img src="figure/Figure3.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/65ac5a530490a0f31dfd2a3aef73903ed07b27bd/docs/figure/Figure3.Rmd/unnamed-chunk-8-2.png" target="_blank">65ac5a5</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>max(all_sum_median$count_positive)</code></pre>
<pre><code>[1] 23</code></pre>
<pre class="r"><code>saveRDS(all_sum_median,here(&quot;data/mouse_permutations_median.rds&quot;))

#permutation using means
all &lt;- tibble()
for (j in c(1:nrow(keep))) {
  print(j)
  P &lt;- j
  for (i in c(500, 1000, 2000, 3000, 4000, 5000)) {
    t2l_sum_temp &lt;- t2l_sum[which(t2l_sum$top == i),]
    for (q in unique(t2l_sum_temp$variable)) {
      t2l_sum_temp_temp &lt;-  t2l_sum_temp[which( t2l_sum_temp$variable == q),]
      pindex &lt;- keep[j,] + 1
      permutation &lt;- c(&quot;mutant&quot;, &quot;wild-type&quot;)[pindex]
      t2l_sum_temp_temp$mut &lt;- permutation
      #t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;mutant&quot;),]$value &lt;- sample(t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;mutant&quot;),]$value, size = 3, replace = T)
      #t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;wild-type&quot;),]$value &lt;- sample(t2l_sum_temp_temp[which(t2l_sum_temp_temp$mut == &quot;wild-type&quot;),]$value, size = 3, replace = T)
      sum &lt;- t2l_sum_temp_temp %&gt;%
        group_by(mut) %&gt;%
        summarise(med = mean(norm_value))
      if (q %in% c(&quot;met_is_min&quot;, &quot;rna_min&quot;)) {
        sum_temp &lt;- tibble(&quot;P&quot; = P,
                           &quot;top&quot; = i,
                           &quot;group&quot; = q,
                           &quot;direction&quot; = sign(sum[which(sum$mut == &quot;wild-type&quot;),]$med - sum[which(sum$mut == &quot;mutant&quot;),]$med))
      }
      else {
          sum_temp &lt;- tibble(&quot;P&quot; = P,
                           &quot;top&quot; = i,
                           &quot;group&quot; = q,
                           &quot;direction&quot; = sign(sum[which(sum$mut == &quot;mutant&quot;),]$med - sum[which(sum$mut == &quot;wild-type&quot;),]$med))
        }
        all &lt;- rbind(all, sum_temp)
      }
    }
}</code></pre>
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5
[1] 6
[1] 7
[1] 8
[1] 9
[1] 10
[1] 11
[1] 12
[1] 13
[1] 14
[1] 15
[1] 16
[1] 17
[1] 18
[1] 19
[1] 20</code></pre>
<pre class="r"><code>all[which(all$direction == -1),]$direction &lt;- 0
all_sum &lt;- all %&gt;%
  group_by(P) %&gt;%
  summarise(count_positive = sum(direction))
all_sum_mean &lt;- all_sum
hist(all_sum_mean$count_positive, breaks = 30)</code></pre>
<p><img src="figure/Figure3.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-3">
Past versions of unnamed-chunk-8-3.png
</button>
</p>
<div id="fig-unnamed-chunk-8-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/65ac5a530490a0f31dfd2a3aef73903ed07b27bd/docs/figure/Figure3.Rmd/unnamed-chunk-8-3.png" target="_blank">65ac5a5</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>max(all_sum_mean$count_positive)</code></pre>
<pre><code>[1] 24</code></pre>
<pre class="r"><code>saveRDS(all_sum_mean,here(&quot;data/mouse_permutations_mean.rds&quot;))
  
#calculating the amount of times a metric is in the correct direction, when using correct labels
all_norm &lt;- tibble()
for (i in c(500, 1000, 2000, 3000, 4000, 5000)){
  t2l_sum_temp &lt;- t2l_sum[which(t2l_sum$top == i),]
  for (q in unique(t2l_sum$variable)) {
    t2l_sum_temp_temp &lt;-  t2l_sum_temp[which( t2l_sum_temp$variable == q),]
    sum &lt;- t2l_sum_temp_temp %&gt;%
      group_by(mut) %&gt;%
      summarise(med = mean(norm_value))
    if (q %in% c(&quot;met_is_min&quot;, &quot;rna_min&quot;)) {
        sum_temp &lt;- tibble(&quot;P&quot; = P,
                           &quot;top&quot; = i,
                           &quot;group&quot; = q,
                           &quot;direction&quot; = sign(sum[which(sum$mut == &quot;wild-type&quot;),]$med - sum[which(sum$mut == &quot;mutant&quot;),]$med))
      }
    else {
        sum_temp &lt;- tibble(&quot;P&quot; = P,
                           &quot;top&quot; = i,
                           &quot;group&quot; = q,
                           &quot;direction&quot; = sign(sum[which(sum$mut == &quot;mutant&quot;),]$med - sum[which(sum$mut == &quot;wild-type&quot;),]$med))
      }
    all_norm &lt;- rbind(all_norm, sum_temp)
  }
}
all_norm[which(all_norm$direction == -1),]$direction &lt;- 0
all_norm_sum &lt;- all_norm %&gt;%
  group_by(P) %&gt;%
  summarise(count_positive = sum(direction))

t2l_sum$variable &lt;- sub(&quot;rna_min&quot;, &quot;High expression regions\nIDH mutant&quot;, t2l_sum$variable)
t2l_sum$variable &lt;- sub(&quot;rna_plus&quot;, &quot;High expression regions\nIDH wild-type&quot;, t2l_sum$variable)
t2l_sum$variable &lt;- sub(&quot;met_is_min&quot;, &quot;High methylation regions\nIDH wild-type&quot;, t2l_sum$variable)
t2l_sum$variable &lt;- sub(&quot;met_is_plus&quot;, &quot;High methylation regions\nIDH mutant&quot;, t2l_sum$variable)
t2l_sum$group2 &lt;- &quot;lower&quot;
t2l_sum[which(t2l_sum$variable %in% c(&quot;High expression regions\nIDH mutant&quot;, &quot;High methylation regions\nIDH mutant&quot;)),]$group2 &lt;- &quot;higher&quot;
t2l_sum$variable &lt;- factor(t2l_sum$variable, levels = c(&quot;High methylation regions\nIDH mutant&quot;, &quot;High expression regions\nIDH mutant&quot;, &quot;High methylation regions\nIDH wild-type&quot;, &quot;High expression regions\nIDH wild-type&quot;))
mouse_plot &lt;- t2l_sum %&gt;%
  group_by(mut, top, variable, group, group2) %&gt;%
  summarise(mean_value = mean(norm_value)) %&gt;%
  ggplot(aes(x=factor(top), y=mean_value)) +
  geom_line(aes(group = mut, colour = mut), size =1) +
  facet_wrap(.~variable, scales = &quot;free&quot;) +
  theme_classic(base_size = 20) +
  theme(strip.background = element_blank(),
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x = element_text(margin = margin(t = 20, b = 0)),
        axis.title.y = element_text(margin = margin(r = 20, l = 0)))+
  #scale_x_continuous(breaks = c(500,1000,2000,3000,4000,5000)) +
  scale_y_continuous(labels = scales::percent, breaks = c(0,1), limits = c(0,1)) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;#004766&quot;)) +
  guides(fill=guide_legend(title=&quot;IDH status\nin isogenic\nxenografts&quot;),
         colour=guide_legend(title=&quot;IDH status\nin isogenic\nxenografts&quot;)) +
  ylab(&quot;Normalized human-derived cfDNA coverage&quot;) +
  xlab(&quot;Top n differentially methylated or expressed locations\nin isogenic IDH1 mutant or wild-type tumors&quot;)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;mut&#39;, &#39;top&#39;, &#39;variable&#39;, &#39;group&#39;. You can
override using the `.groups` argument.</code></pre>
<pre class="r"><code>mouse_plot_label &lt;- plot_grid(mouse_plot, labels=c(&quot;g&quot;), nrow=1, label_size = 20)
ggsave(here(&quot;docs/figure/Fig_3G.jpg&quot;), plot = mouse_plot_label, device = &quot;jpeg&quot;, width = 12 , height= 14, units = &quot;in&quot;, dpi=600, scale = 0.82)

#met_is_cases &lt;- ggplot(t2l_sum[which(t2l_sum$top == 1000),], aes(y=norm_value, x=factor(variable), fill=mut)) +
#  geom_boxplot(outlier.shape=NA, alpha=0.5) +
#  geom_point(position=position_jitterdodge(jitter.width=0.25), aes(colour=mut)) + 
#  facet_wrap(.~group,scales = &quot;free&quot;) +
#  theme_classic(base_size = 10) +
#  theme(#axis.title.x = element_blank(),
#        strip.background = element_blank(),
#        #legend.title=element_blank(),
#        text = element_text(size=20),
#        #axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
#        axis.title.x = element_text(margin = margin(t = 20, b = 0)),
#        axis.title.y = element_text(margin = margin(r = 0, l = 0))) +
#  ylab(&quot;Normalized human-derived cfDNA coverage&quot;) +
#  scale_fill_manual(values = c(&quot;red&quot;, &quot;#004766&quot;)) + 
#  scale_color_manual(values = c(&quot;red&quot;, &quot;#004766&quot;)) +
#  scale_y_continuous(labels = scales::percent, breaks = c(0,1)) +
#  scale_x_discrete(labels=c(&quot;rna_min&quot; = &quot;High\nexpression\nregions\nIDH mutant&quot;, 
#                            &quot;rna_plus&quot; = &quot;High\nexpression\nregions\nIDH WT&quot;,
#                            &quot;met_is_min&quot; = &quot;High\nmethylation\nregions\nIDH WT&quot;,
#                            &quot;met_is_plus&quot; = &quot;High\nmethylation\nregions\nIDH mutant&quot;)) +
#  xlab(&quot;Top 1000 differentially methylated or expressed locations\nin isogenic IDH1 mutant or wild-type tumors&quot;) + 
#  guides(fill=guide_legend(title=&quot;IDH status\nin isogenic\nxenografts&quot;),
#         colour=guide_legend(title=&quot;IDH status\nin isogenic\nxenografts&quot;)) +
#  stat_compare_means(size = 3, method = &quot;t.test&quot;, aes(label = paste0(&quot;p = &quot;, after_stat(p.format))))
#  
#met_is_cases_label &lt;- plot_grid(met_is_cases, labels=c(&quot;g&quot;), nrow=1, label_size = 20)
#ggsave(here(&quot;docs/figure/Fig_3G.jpg&quot;), plot = met_is_cases_label, device = &quot;jpeg&quot;, width = 12 , height= 15, units = &quot;in&quot;, dpi=300, scale = 0.82)</code></pre>
</div>
<div
id="figure-3h-cfdna-fragment-size-distributions-and-cumulative-representations"
class="section level2">
<h2>Figure 3H: cfDNA fragment size distributions and cumulative
representations</h2>
<pre class="r"><code>f &lt;- readRDS(here(&quot;data/fragments_somatic.rds&quot;))
f &lt;- f[which((f$w &gt;= 100) &amp; (f$w &lt;= 220)),]
f$group &lt;- substr(f$patient, 1, 5)
f[which(f$group == &quot;CGCRC&quot;),]$group &lt;- &quot;colorectal cancer&quot;
f[which(f$group == &quot;CGPLB&quot;),]$group &lt;- &quot;breast cancer&quot;
f[which(f$group == &quot;CGPLL&quot;),]$group &lt;- &quot;lung cancer&quot;
f[which(f$group == &quot;CGPLO&quot;),]$group &lt;- &quot;ovarian cancer&quot;
f$group &lt;- factor(f$group, levels=c(&quot;lung cancer&quot;, &quot;colorectal cancer&quot;, &quot;breast cancer&quot;, &quot;ovarian cancer&quot;))
f$mutation.stat &lt;- factor(f$mutation.stat, levels=c(TRUE, FALSE))

f$n &lt;- 1
t &lt;- f %&gt;%
  group_by(mutation.stat, w)%&gt;%
  summarize(n = sum(n)) </code></pre>
<pre><code>`summarise()` has grouped output by &#39;mutation.stat&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>t$rel &lt;- 0
t$rel_mod &lt;- 0
t[which(t$mutation.stat == TRUE),]$rel &lt;- t[which(t$mutation.stat == TRUE),]$n / sum(t[which(t$mutation.stat == TRUE),]$n)
t[which(t$mutation.stat == FALSE),]$rel &lt;- t[which(t$mutation.stat == FALSE),]$n / sum(t[which(t$mutation.stat == FALSE),]$n)
t[which(t$mutation.stat == TRUE),]$rel_mod &lt;- t[which(t$mutation.stat == TRUE),]$n / max(t[which(t$mutation.stat == TRUE),]$n)
t[which(t$mutation.stat == FALSE),]$rel_mod &lt;- t[which(t$mutation.stat == FALSE),]$n / max(t[which(t$mutation.stat == FALSE),]$n)
 weighted.mean(t[which(t$mutation.stat == FALSE),]$w, t[which(t$mutation.stat == FALSE),]$n) - weighted.mean(t[which(t$mutation.stat == TRUE),]$w, t[which(t$mutation.stat == TRUE),]$n)</code></pre>
<pre><code>[1] 3.934806</code></pre>
<pre class="r"><code>diff &lt;- t[which(t$mutation.stat == TRUE),]$rel - t[which(t$mutation.stat == FALSE),]$rel
f_diff &lt;- tibble(&quot;size&quot; = c(100:220),
                &quot;diff&quot; = diff)

group &lt;- ggplot(f, aes(x=w, group=mutation.stat, colour=mutation.stat)) +
  stat_ecdf(geom = &quot;line&quot;, pad = FALSE)+
  theme_classic(base_size = 20) +
  theme(text = element_text(size=20),
        axis.title.x = element_text(margin = margin(t = 20, b = 0)),
        legend.text = element_text(size = 12),
        legend.position = c(0.70,0.12),
        legend.title = element_blank(),
        legend.background = element_rect(fill= &quot;transparent&quot;),
        axis.title.y = element_text(margin = margin(r = 20, l = 0))) +
  scale_color_manual(values= c(&quot;red&quot;, &quot;#004766&quot;),
                     labels=c(&quot;Mutated (tumor-derived)&quot;, &quot;Wild-type (WBC)&quot;)) +
  ylab(&quot;Cumulative representation\nof cfDNA fragments&quot;) +
  xlab(&quot;cfDNA fragment size&quot;)


group2a &lt;- ggplot(t, aes(x=w, y=rel, group=mutation.stat)) +
  geom_line(aes(colour=mutation.stat)) +
  #geom_smooth(method = &quot;loess&quot;, se = FALSE, span = 0.5, colour=&quot;black&quot;) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=20),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 12),
        legend.position = c(0.25,0.95),
        legend.title = element_blank(),
        legend.background = element_rect(fill= &quot;transparent&quot;),
        axis.title.y = element_text(margin = margin(r = 20, l = 0)),
        ) +
  scale_color_manual(values= c(&quot;red&quot;, &quot;#004766&quot;),
                     labels=c(&quot;Mutated (tumor)&quot;, &quot;Wild-type (WBC)&quot;)) +
  ylab(&quot;Prevalence of\ncfDNA fragment sizes&quot;) 


ffinal &lt;- readRDS(here(&quot;data/cdf_multiple.rds&quot;))
ffinal$bigroup &lt;- 0
ffinal[which(ffinal$group == &quot;expr&quot;),]$bigroup &lt;- 1
ffinal[which(ffinal$group == &quot;unmeth&quot;),]$bigroup &lt;- 1
ffinal_diff &lt;- ffinal %&gt;%
  group_by(tss, size) %&gt;%
  summarise(diff_csum_n = rel_csum_n[bigroup == 1] - rel_csum_n[bigroup == 0],
            diff_n = rel_n[bigroup == 1] - rel_n[bigroup == 0])</code></pre>
<pre><code>`summarise()` has grouped output by &#39;tss&#39;. You can override using the `.groups`
argument.</code></pre>
<pre class="r"><code>ffinal$all_group &lt;- paste0(ffinal$tss, &quot;_&quot;, ffinal$group)

ffinal$label &lt;- &quot;NA&quot;
ffinal[which((ffinal$tss == &quot;tss_included_100bp-800bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;expr&quot;)),]$label &lt;- &quot;800bp - 100bp around TSS, 1000 most expressed genes&quot;
ffinal[which((ffinal$tss == &quot;tss_included_100bp-800bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;nexpr&quot;)),]$label &lt;- &quot;800bp - 100bp around TSS, 1000 least expressed genes&quot;
ffinal[which((ffinal$tss == &quot;tss_included_800bp-1000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;expr&quot;)),]$label &lt;- &quot;1000bp - 800bp around TSS, 1000 most expressed genes&quot;
ffinal[which((ffinal$tss == &quot;tss_included_800bp-1000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;nexpr&quot;)),]$label &lt;- &quot;1000bp - 800bp around TSS, 1000 least expressed genes&quot;
ffinal[which((ffinal$tss == &quot;cpg_included_800bp-1000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;meth&quot;)),]$label &lt;- &quot;1000bp - 800bp around CpG-island, 1000 most methylated CpG-islands&quot;
ffinal[which((ffinal$tss == &quot;cpg_included_800bp-1000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;unmeth&quot;)),]$label &lt;- &quot;1000bp - 800bp around CpG-island, 1000 least methylated CpG-islands&quot;

ffinal[which((ffinal$tss == &quot;tss_included_2500bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;expr&quot;)),]$label &lt;- &quot;2500bp around TSS, 1000 most expressed genes&quot;
ffinal[which((ffinal$tss == &quot;tss_included_2500bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;nexpr&quot;)),]$label &lt;- &quot;2500bp around TSS, 1000 least expressed genes&quot;
ffinal[which((ffinal$tss == &quot;tss_included_500000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;expr&quot;)),]$label &lt;- &quot;500000bp around TSS, 1000 most expressed genes&quot;
ffinal[which((ffinal$tss == &quot;tss_included_500000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;nexpr&quot;)),]$label &lt;- &quot;500000bp around TSS, 1000 least expressed genes&quot;

ffinal[which((ffinal$tss == &quot;cpg_included_500000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;meth&quot;)),]$label &lt;- &quot;500000bp around CpG-island, 1000 most methylated CpG-islands&quot;
ffinal[which((ffinal$tss == &quot;cpg_included_500000bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;unmeth&quot;)),]$label &lt;- &quot;500000bp around CpG-island, 1000 least methylated CpG-islands&quot;
ffinal[which((ffinal$tss == &quot;cpg_included_2500bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;meth&quot;)),]$label &lt;- &quot;2500bp around CpG-island, 1000 most methylated CpG-islands&quot;
ffinal[which((ffinal$tss == &quot;cpg_included_2500bp_top1000_bot1000.rds&quot;) &amp; (ffinal$group == &quot;unmeth&quot;)),]$label &lt;- &quot;2500bp around CpG-island, 1000 least methylated CpG-islands&quot;

ffinal$label &lt;- factor(ffinal$label, level=unique(ffinal$label)[c(1,4,2,3,14,12,13,8,11,9,10,7,5,6)])
tss_df &lt;- ffinal[which(ffinal$tss %in% c(&quot;tss_included_800bp-1000bp_top1000_bot1000.rds&quot;)),]
tss_df$rel_mod &lt;- 0
tss_df[which(tss_df$group == &quot;expr&quot;),]$rel_mod &lt;- tss_df[which(tss_df$group == &quot;expr&quot;),]$n / max(tss_df[which(tss_df$group == &quot;expr&quot;),]$n)
tss_df[which(tss_df$group == &quot;nexpr&quot;),]$rel_mod &lt;- tss_df[which(tss_df$group == &quot;nexpr&quot;),]$n / max(tss_df[which(tss_df$group == &quot;nexpr&quot;),]$n)
weighted.mean(tss_df[which(tss_df$group == &quot;nexpr&quot;),]$size, tss_df[which(tss_df$group == &quot;nexpr&quot;),]$n) - weighted.mean(tss_df[which(tss_df$group == &quot;expr&quot;),]$size, tss_df[which(tss_df$group == &quot;expr&quot;),]$n)</code></pre>
<pre><code>[1] 4.13887</code></pre>
<pre class="r"><code>tss_diff &lt;- tss_df[which(tss_df$group == &quot;expr&quot;),]$rel_n - tss_df[which(tss_df$group == &quot;nexpr&quot;),]$rel_n
tss_diff &lt;- tibble(&quot;size&quot; = c(100:220),
                &quot;diff&quot; = tss_diff)
tss &lt;- ggplot(tss_df, aes(x=size, y=rel_csum_n, group = label, colour = label)) +
  geom_line() +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=20),
        axis.title.x = element_text(margin = margin(t = 20, b = 0)),
        legend.text = element_text(size = 12),
        legend.position = c(0.70,0.12),
        legend.title = element_blank(),
        legend.background = element_rect(fill= &quot;transparent&quot;),
        legend.key = element_rect(colour = NA, fill = NA),
        axis.title.y = element_text(margin = margin(r = 20, l = 0))) +
  scale_color_manual(values= c(&quot;#1b7d0e&quot;, &quot;#52c922&quot;),
                     labels=c(&quot;TSS high expression&quot;, &quot;TSS low expression&quot;)) +
  ylab(&quot;&quot;) +
  xlab(&quot;cfDNA fragment size&quot;)


tss2a &lt;- ggplot(tss_df, aes(x=size, y=rel_n, group=group)) +
  geom_line(aes(colour=group)) +
  #geom_smooth(method = &quot;loess&quot;, se = FALSE, span = 0.5, colour=&quot;black&quot;) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=20),
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        legend.text = element_text(size = 12),
        legend.position = c(0.28,0.95),
        legend.title = element_blank(),
        legend.background = element_rect(fill= &quot;transparent&quot;)) +
  scale_color_manual(values= c(&quot;#1b7d0e&quot;, &quot;#52c922&quot;),
                     labels=c(&quot;TSS high expression&quot;, &quot;TSS low expression&quot;))


cpg_df &lt;- ffinal[which(ffinal$tss %in% c(&quot;cpg_included_800bp-1000bp_top1000_bot1000.rds&quot;)),]
cpg_diff &lt;- cpg_df[which(cpg_df$group == &quot;meth&quot;),]$rel_n - cpg_df[which(cpg_df$group == &quot;unmeth&quot;),]$rel_n
cpg_df$rel_mod &lt;- 0
cpg_df[which(cpg_df$group == &quot;meth&quot;),]$rel_mod &lt;- cpg_df[which(cpg_df$group == &quot;meth&quot;),]$n / max(cpg_df[which(cpg_df$group == &quot;meth&quot;),]$n)
cpg_df[which(cpg_df$group == &quot;unmeth&quot;),]$rel_mod &lt;- cpg_df[which(cpg_df$group == &quot;unmeth&quot;),]$n / max(cpg_df[which(cpg_df$group == &quot;unmeth&quot;),]$n)
weighted.mean(cpg_df[which(cpg_df$group == &quot;meth&quot;),]$size, cpg_df[which(cpg_df$group == &quot;meth&quot;),]$n) - weighted.mean(cpg_df[which(cpg_df$group == &quot;unmeth&quot;),]$size, cpg_df[which(cpg_df$group == &quot;unmeth&quot;),]$n)</code></pre>
<pre><code>[1] 2.158964</code></pre>
<pre class="r"><code>cpg_diff &lt;- tibble(&quot;size&quot; = c(100:220),
                &quot;diff&quot; = cpg_diff)
cpg_df$group &lt;- factor(cpg_df$group, levels = c(&quot;unmeth&quot;, &quot;meth&quot;))
cpg &lt;- ggplot(cpg_df, aes(x=size, y=rel_csum_n, group = group, colour = group)) +
  geom_line() +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=20),
        axis.title.x = element_text(margin = margin(t = 20, b = 0)),
        legend.text = element_text(size = 12),
        legend.position = c(0.70,0.12),
        legend.title = element_blank(),
        legend.background = element_rect(fill= &quot;transparent&quot;),
        axis.title.y = element_blank()) +
  scale_color_manual(values= c(&quot;#004766&quot;, &quot;#77C8DD&quot;),
                     labels=c(&quot;CpG low methylation&quot;, &quot;CpG high methylation&quot;)) +
  ylab(&quot;&quot;) +
  xlab(&quot;cfDNA fragment size&quot;)
  #ggtitle(&quot;2500bp around CpG-islands&quot;)


cpg2a &lt;- ggplot(cpg_df, aes(x=size, y=rel_n, group=group)) +
  geom_line(aes(colour=group)) +
  #geom_smooth(method = &quot;loess&quot;, se = FALSE, span = 0.5, colour=&quot;black&quot;) +
  theme_classic(base_size = 20) +
  theme(text = element_text(size=20),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        legend.text = element_text(size = 12),
        legend.position = c(0.28,0.95),
        legend.title = element_blank(),
        legend.background = element_rect(fill= &quot;transparent&quot;)) +
   scale_color_manual(values= c( &quot;#004766&quot;, &quot;#77C8DD&quot;),
                     labels=c(&quot;CpG low methylation&quot;, &quot;CpG high methylation&quot;))

av &lt;- plot_grid(group2a, group, labels=c(&quot;h&quot;, &quot;&quot;, &quot;&quot;), nrow=2, align = &quot;v&quot;, label_size = 20)

av1 &lt;- plot_grid(tss2a, tss, labels=c(&quot;&quot;, &quot;&quot;, &quot;&quot;), nrow=2, align = &quot;v&quot;, label_size = 20)
av2 &lt;- plot_grid(cpg2a, cpg, labels=c(&quot;&quot;, &quot;&quot;, &quot;&quot;), nrow=2, align = &quot;v&quot;, label_size = 20)

avv &lt;- plot_grid(av, av1, av2, ncol=3, rel_widths = c(1.05, 1, 1), align = &quot;hv&quot;, axis = &quot;bt&quot;,label_size = 20)

ggsave(here(&quot;docs/figure/Fig_3H.jpg&quot;), plot = avv, width = 20, height=10, units = &quot;in&quot;, dpi=600, bg = &quot;white&quot;, scale = 0.8)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] here_1.0.1           gtools_3.9.5         GenomicRanges_1.50.2
 [4] GenomeInfoDb_1.34.9  IRanges_2.32.0       S4Vectors_0.36.2    
 [7] BiocGenerics_0.44.0  lubridate_1.9.3      forcats_1.0.0       
[10] stringr_1.5.1        dplyr_1.1.4          purrr_1.0.2         
[13] readr_2.1.5          tidyr_1.3.1          tibble_3.2.1        
[16] ggplot2_3.4.4        tidyverse_2.0.0      cowplot_1.1.3       
[19] reshape2_1.4.4       workflowr_1.7.1     

loaded via a namespace (and not attached):
 [1] httr_1.4.7             sass_0.4.8             jsonlite_1.8.8        
 [4] bslib_0.6.1            getPass_0.2-4          highr_0.10            
 [7] GenomeInfoDbData_1.2.9 yaml_2.3.8             pillar_1.9.0          
[10] glue_1.7.0             digest_0.6.34          promises_1.2.1        
[13] XVector_0.38.0         colorspace_2.1-0       htmltools_0.5.7       
[16] httpuv_1.6.14          plyr_1.8.9             pkgconfig_2.0.3       
[19] zlibbioc_1.44.0        scales_1.3.0           processx_3.8.3        
[22] whisker_0.4.1          later_1.3.2            tzdb_0.4.0            
[25] timechange_0.3.0       git2r_0.33.0           generics_0.1.3        
[28] farver_2.1.1           cachem_1.0.8           withr_3.0.0           
[31] cli_3.6.2              magrittr_2.0.3         crayon_1.5.2          
[34] evaluate_0.23          ps_1.7.6               fs_1.6.3              
[37] fansi_1.0.6            textshaping_0.3.7      tools_4.2.3           
[40] hms_1.1.3              lifecycle_1.0.4        munsell_0.5.0         
[43] callr_3.7.3            compiler_4.2.3         jquerylib_0.1.4       
[46] systemfonts_1.0.5      rlang_1.1.3            grid_4.2.3            
[49] RCurl_1.98-1.14        rstudioapi_0.15.0      bitops_1.0-7          
[52] labeling_0.4.3         rmarkdown_2.25         gtable_0.3.4          
[55] R6_2.5.1               knitr_1.45             fastmap_1.1.1         
[58] utf8_1.2.4             rprojroot_2.0.4        ragg_1.2.7            
[61] stringi_1.8.3          Rcpp_1.0.12            vctrs_0.6.5           
[64] tidyselect_1.2.0       xfun_0.42             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
