<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Michael Noe" />

<meta name="date" content="2024-03-12" />

<title>Figure1.Rmd</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">noe2024</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure1.Rmd</h1>
<h4 class="author">Michael Noe</h4>
<h4 class="date">2024-03-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>noe2024/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024tree65ac5a530490a0f31dfd2a3aef73903ed07b27bdtargetblank65ac5a5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/cancer-genomics/noe2024/tree/65ac5a530490a0f31dfd2a3aef73903ed07b27bd" target="_blank">65ac5a5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024tree65ac5a530490a0f31dfd2a3aef73903ed07b27bdtargetblank65ac5a5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/cancer-genomics/noe2024/tree/65ac5a530490a0f31dfd2a3aef73903ed07b27bd" target="_blank">65ac5a5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/results/

Unstaged changes:
    Deleted:    analysis/.nfs00000000003fc5760000002d
    Deleted:    analysis/.nfs00000000003fc81300000027
    Deleted:    analysis/.nfs00000000003fc8ee00000029
    Deleted:    analysis/.nfs00000000003fc8f30000002a
    Deleted:    analysis/.nfs00000000003fc8f80000002b
    Deleted:    analysis/.nfs00000000003fc97f0000002c
    Deleted:    analysis/.nfs00000000003fca2d00000028
    Deleted:    analysis/.nfs00000000003fcac40000002e
    Modified:   analysis/Figure3.Rmd
    Modified:   analysis/SessionInfo.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Figure1.Rmd</code>) and HTML
(<code>docs/Figure1.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/65ac5a530490a0f31dfd2a3aef73903ed07b27bd/docs/Figure1.html" target="_blank">65ac5a5</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-11
</td>
<td>
Update supplementary tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/e2712c2174370e09a641030a5487b6a18338704f/docs/Figure1.html" target="_blank">e2712c2</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-03-06
</td>
<td>
Reorder figures: add S4,S6,S13
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/9d3f606bb5ef1c1a9c3c587cddd0f2250b76198f/docs/Figure1.html" target="_blank">9d3f606</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-24
</td>
<td>
Add Supp Fig 9, reorder Fig 3 figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/dc39d3fb10aa7b5335c6182588b4ef4a906fb458/analysis/Figure1.Rmd" target="_blank">dc39d3f</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-21
</td>
<td>
Add supplementary figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/dc39d3fb10aa7b5335c6182588b4ef4a906fb458/docs/Figure1.html" target="_blank">dc39d3f</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-21
</td>
<td>
Add supplementary figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/e1f36dcdb578c47e206926e77af40fe0d2c7053d/analysis/Figure1.Rmd" target="_blank">e1f36dc</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-15
</td>
<td>
Add supp. figs and tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/e1f36dcdb578c47e206926e77af40fe0d2c7053d/docs/Figure1.html" target="_blank">e1f36dc</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-02-15
</td>
<td>
Add supp. figs and tables
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/de3561665de4eb4a67334ac71cee69954c0f14f5/analysis/Figure1.Rmd" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/de3561665de4eb4a67334ac71cee69954c0f14f5/docs/Figure1.html" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="pre-requisites" class="section level2">
<h2>Pre-requisites</h2>
<p>These packages are necessary to process the data to make these
figures</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)
library(&quot;annotatr&quot;)
library(&quot;cowplot&quot;)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;

The following object is masked from &#39;package:lubridate&#39;:

    stamp</code></pre>
<pre class="r"><code>library(&quot;data.table&quot;)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last

The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(&quot;GenomicRanges&quot;)</code></pre>
<pre><code>Loading required package: stats4
Loading required package: BiocGenerics

Attaching package: &#39;BiocGenerics&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    intersect, setdiff, union

The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union

The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs

The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors

Attaching package: &#39;S4Vectors&#39;

The following objects are masked from &#39;package:data.table&#39;:

    first, second

The following objects are masked from &#39;package:lubridate&#39;:

    second, second&lt;-

The following objects are masked from &#39;package:dplyr&#39;:

    first, rename

The following object is masked from &#39;package:tidyr&#39;:

    expand

The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: &#39;IRanges&#39;

The following object is masked from &#39;package:data.table&#39;:

    shift

The following object is masked from &#39;package:lubridate&#39;:

    %within%

The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice

The following object is masked from &#39;package:purrr&#39;:

    reduce

Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>library(&quot;here&quot;)</code></pre>
<pre><code>here() starts at /Users/skoul/Documents/Projects/cfepi_wflow/noe2024</code></pre>
</div>
<div id="figure-1a" class="section level2">
<h2>Figure 1A</h2>
<pre class="r"><code>total &lt;- readRDS(here(&quot;data/end_motifs_cfdna_individual.rds&quot;))
enri &lt;- readRDS(here(&quot;data/end_motifs_cfdna_individual_200over_5perc.rds&quot;))
shea &lt;- readRDS(here(&quot;data/end_motifs_sheared_individual.rds&quot;))
theo &lt;- readRDS(here(&quot;data/theoretical_frequencies_hg19.rds&quot;))
theo$motif &lt;- paste0(substr(theo$motif,1,1), &quot; | &quot;, substr(theo$motif,2,3))
test &lt;- total %&gt;%
  group_by(Var1, group) %&gt;%
  summarise(mean = mean(FreqRatio))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Var1&#39;. You can override using the
`.groups` argument.</code></pre>
<pre class="r"><code>test$Var1 &lt;-  paste0(substr(test$Var1,1,1), &quot; | &quot;, substr(test$Var1,2,3))
total$Var1 &lt;- paste0(substr(total$Var1,1,1), &quot; | &quot;, substr(total$Var1,2,3))
enri$Var1 &lt;- paste0(substr(enri$Var1,1,1), &quot; | &quot;, substr(enri$Var1,2,3))
shea$Var1 &lt;- paste0(substr(shea$Var1,1,1), &quot; | &quot;, substr(shea$Var1,2,3))
total$totheo &lt;- 0
test$totheo &lt;- 0
enri$totheo &lt;- 0
shea$totheo &lt;- 0

mean(total[which(total$Var1 == &quot;A | CC&quot;),]$FreqRatio) + mean(total[which(total$Var1 == &quot;T | CC&quot;),]$FreqRatio)</code></pre>
<pre><code>[1] 0.1108725</code></pre>
<pre class="r"><code>mean(total[which(total$Var1 == &quot;A | CG&quot;),]$FreqRatio) + mean(total[which(total$Var1 == &quot;T | CG&quot;),]$FreqRatio)</code></pre>
<pre><code>[1] 0.01126663</code></pre>
<pre class="r"><code>mean(enri[which(enri$Var1 == &quot;A | CC&quot;),]$FreqRatio) + mean(enri[which(enri$Var1 == &quot;T | CC&quot;),]$FreqRatio)</code></pre>
<pre><code>[1] 0.7505879</code></pre>
<pre class="r"><code>mean(enri[which(enri$Var1 == &quot;A | CG&quot;),]$FreqRatio) + mean(enri[which(enri$Var1 == &quot;T | CG&quot;),]$FreqRatio)</code></pre>
<pre><code>[1] 0.02584186</code></pre>
<pre class="r"><code>for (i in as.character(unique(total$Var1))) {
  total[which(total$Var1 == i),]$totheo &lt;- total[which(total$Var1 == i),]$FreqRatio / theo[which(theo$motif == i),]$freq
  test[which(test$Var1 == i),]$totheo &lt;- test[which(test$Var1 == i),]$mean / theo[which(theo$motif == i),]$freq
  enri[which(enri$Var1 == i),]$totheo &lt;- enri[which(enri$Var1 == i),]$FreqRatio / theo[which(theo$motif == i),]$freq
  shea[which(shea$Var1 == i),]$totheo &lt;- shea[which(shea$Var1 == i),]$FreqRatio / theo[which(theo$motif == i),]$freq
}

enri$group &lt;- &quot;cfDNA at preferred sequences&quot;
shea$group &lt;- &quot;sonicated DNA&quot;
shea &lt;- shea[,c(1,2,3,6,4,5)]
colnames(enri) &lt;- c(&quot;Var1&quot;, &quot;Freq&quot;, &quot;id&quot;, &quot;group&quot;, &quot;FreqRatio&quot;, &quot;totheo&quot;)
test2 &lt;- test[which(test$group == &quot;normal_cfdna&quot;),]
test2 &lt;- test2[order(-test2$totheo),]

total[which(total$group == &quot;normal_cfdna&quot;), ]$group &lt;- &quot;cfDNA&quot;
total &lt;- rbind(total, enri, shea)
total$Var1 &lt;- factor(total$Var1, levels = as.character(test2$Var1))
total$group &lt;- factor(total$group, levels = c(&quot;cfDNA at preferred sequences&quot;, &quot;cfDNA&quot;, &quot;sonicated DNA&quot;))


mean(c(mean(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 == &quot;A | CC&quot;)),]$totheo),
     mean(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 == &quot;T | CC&quot;)),]$totheo)))</code></pre>
<pre><code>[1] 26.54543</code></pre>
<pre class="r"><code>mean(c(mean(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 == &quot;A | CG&quot;)),]$totheo),
     mean(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 == &quot;T | CG&quot;)),]$totheo)))</code></pre>
<pre><code>[1] 5.518233</code></pre>
<pre class="r"><code>t.test(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;A | CC&quot;, &quot;T | CC&quot;, &quot;A | CG&quot;, &quot;T | CG&quot;))),]$FreqRatio, theo[which(theo$motif %in% c(&quot;A | CC&quot;, &quot;T | CC&quot;, &quot;A | CG&quot;, &quot;T | CG&quot;)),]$freq)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;A | CC&quot;, &quot;T | CC&quot;, &quot;A | CG&quot;, &quot;T | CG&quot;))), ]$FreqRatio and theo[which(theo$motif %in% c(&quot;A | CC&quot;, &quot;T | CC&quot;, &quot;A | CG&quot;, &quot;T | CG&quot;)), ]$freq
t = 23.823, df = 86.498, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.1706277 0.2016944
sample estimates:
  mean of x   mean of y 
0.194107440 0.007946418 </code></pre>
<pre class="r"><code>t.test(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;A | CC&quot;))),]$FreqRatio, mu = theo[which(theo$motif %in% c(&quot;A | CC&quot;)),]$freq)</code></pre>
<pre><code>
    One Sample t-test

data:  total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;A | CC&quot;))), ]$FreqRatio
t = 2232.6, df = 231, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0.01162923
95 percent confidence interval:
 0.2104073 0.2107585
sample estimates:
mean of x 
0.2105829 </code></pre>
<pre class="r"><code>t.test(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;T | CC&quot;))),]$FreqRatio, mu = theo[which(theo$motif %in% c(&quot;T | CC&quot;)),]$freq)</code></pre>
<pre><code>
    One Sample t-test

data:  total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;T | CC&quot;))), ]$FreqRatio
t = 1480.2, df = 231, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0.0154363
95 percent confidence interval:
 0.5393068 0.5407032
sample estimates:
mean of x 
 0.540005 </code></pre>
<pre class="r"><code>t.test(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;A | CG&quot;))),]$FreqRatio, mu = theo[which(theo$motif %in% c(&quot;A | CG&quot;)),]$freq)</code></pre>
<pre><code>
    One Sample t-test

data:  total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;A | CG&quot;))), ]$FreqRatio
t = 443.12, df = 231, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0.002513012
95 percent confidence interval:
 0.01214071 0.01222670
sample estimates:
 mean of x 
0.01218371 </code></pre>
<pre class="r"><code>t.test(total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;T | CG&quot;))),]$FreqRatio, mu = theo[which(theo$motif %in% c(&quot;T | CG&quot;)),]$freq)</code></pre>
<pre><code>
    One Sample t-test

data:  total[which((total$group == &quot;cfDNA at preferred sequences&quot;) &amp; (total$Var1 %in% c(&quot;T | CG&quot;))), ]$FreqRatio
t = 502.11, df = 231, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0.002207123
95 percent confidence interval:
 0.01361322 0.01370309
sample estimates:
 mean of x 
0.01365816 </code></pre>
<pre class="r"><code>plot1A_plot &lt;- ggplot(total, aes(y=totheo, x=Var1, color=group)) +
  geom_point(position = position_dodge(width = 0), alpha=0.2, aes(color=group), size=0.1) +
  geom_boxplot(outlier.shape = NA, alpha=0.5, position = position_dodge(width = 0)) +
  scale_color_manual(values=c(&quot;#004766&quot;, &quot;#4FB553&quot;, &quot;#ff8c00&quot;),
                      breaks = c(&quot;cfDNA at preferred sequences&quot;, &quot;cfDNA&quot;, &quot;sonicated DNA&quot;),
                      labels= c(&quot;cfDNA at preferred ends&quot;, &quot;cfDNA&quot;, &quot;sheared&quot;)) + 
  #scale_fill_manual(values=c(&quot;#77C8DD&quot;, &quot;#77dd77&quot;, &quot;black&quot;)) + 
  theme_classic(base_size = 20) +
  scale_y_continuous(breaks = c(1,10,20,30),
                     limits = c(0,36.16585)
                     ) +  
  ylab(&quot;Frequency of end motif sequences to\ntheoretical genomic representation&quot;) +
  xlab(&quot;End motif sequences&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=1),
        text = element_text(size=16),
        legend.title = element_blank(),
        axis.title.y = element_text(margin = margin(r = 20, l = 0)),
        #axis.title.x = element_blank(),
        #axis.text.x.bottom = element_text(margin = margin(b = 0, t=0, r=0, l=0)),
        axis.title.x = element_text(margin = margin(t = 20, b = 20)),
        legend.position = c(.5, .80),
        plot.margin = unit(c(2,1,0,1), &quot;cm&quot;)
        )

n_table_1 &lt;- data.frame(&quot;nucleotide&quot; =  substr(as.character(test2$Var1), 1, 1),
                        &quot;position_row&quot; = 1,
                        &quot;position_column&quot; = c(1:64))
n_table_2 &lt;- data.frame(&quot;nucleotide&quot; =  substr(as.character(test2$Var1),  5, 5),
                        &quot;position_row&quot; = 2,
                        &quot;position_column&quot; = c(1:64))
n_table_3 &lt;- data.frame(&quot;nucleotide&quot; =  substr(as.character(test2$Var1),  6, 6),
                        &quot;position_row&quot; = 3,
                        &quot;position_column&quot; = c(1:64))
nucleotide_table &lt;- rbind(n_table_1, n_table_2, n_table_3)
nucleotide_table$nucleotide &lt;- factor(nucleotide_table$nucleotide, levels = c(&quot;A&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;))
plot1A_axis &lt;- ggplot(nucleotide_table, aes(x=position_column, y=position_row)) +
  geom_tile(aes(fill=nucleotide)) +
  theme_classic() +
  theme(text = element_text(size=16),
        axis.line = element_blank(),
        axis.title = element_blank(),
        axis.text = element_blank(),
        axis.ticks = element_blank(),
        legend.position = &quot;bottom&quot;,
        legend.title=element_blank(),
        panel.border = element_blank()) + 
  scale_x_discrete(expand=c(0,0))+
  scale_y_discrete(expand=c(0,0))+
  scale_fill_manual(values= c(&quot;#77dd77&quot;, &quot;#00660a&quot;, &quot;#0288D1&quot;, &quot;#0D47A1&quot;))
plot1A &lt;- plot_grid(plot1A_plot, plot1A_axis, nrow=2, rel_heights = c(1,0.2), label_size = 20, align = &quot;v&quot;)
#ggsave(here(&quot;docs/figure/temp_preferredcfDNA_cfDNA_sheared.jpg&quot;), plot = plot1A, device = &quot;jpeg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=300, scale = 0.8)</code></pre>
</div>
<div id="figure-1b" class="section level2">
<h2>Figure 1B</h2>
<pre class="r"><code>combination_all &lt;- readRDS(here(&quot;data/cfdna_ratio_motif.rds&quot;))
combination &lt;- combination_all[which(nchar(combination_all$motif) == 3),]

combin &lt;- tibble()
for (i in unique(combination$motif)) {
  temp &lt;- combination[which(combination$motif == i),]
  temp &lt;- temp[which(temp$ratio &lt;= 0.638),]
  new &lt;- tibble(motif = i,
                ratio = seq(0, 0.638, 0.0001))
  new$num &lt;- 0
  new$num &lt;- as.numeric(NA)
  new[match(as.character(temp$ratio), as.character(new$ratio)),]$num &lt;- as.numeric(temp$num)
  new$num &lt;- nafill(new$num,&quot;const&quot;, fill=0)
  combin &lt;- rbind(combin, new)
}
#cfdna_ratios &lt;- combin[which(combin$ratio &gt;= 0.06),c(1,3)] %&gt;%
#  group_by(motif) %&gt;%
#  summarise_all(funs(sum(., na.rm = FALSE))) %&gt;%
#  mutate(freq = num / sum(num))

combin &lt;- combin %&gt;%
  dplyr::group_by(ratio) %&gt;%
  mutate(freq = num / sum(num))

combin$motif_group &lt;- &quot;rest motifs&quot;
combin[which(combin$motif == &quot;ACC&quot;),]$motif_group &lt;- &quot;A | CC&quot;
combin[which(combin$motif == &quot;TCC&quot;),]$motif_group &lt;- &quot;T | CC&quot;
combin[which(combin$motif == &quot;ACG&quot;),]$motif_group &lt;- &quot;A | CG&quot;
combin[which(combin$motif == &quot;TCG&quot;),]$motif_group &lt;- &quot;T | CG&quot;
combin$motif_group &lt;- factor(combin$motif_group, levels = c(&quot;rest motifs&quot;, &quot;T | CG&quot;, &quot;A | CG&quot;, &quot;T | CC&quot;, &quot;A | CC&quot;))
plot &lt;- combin[,c(5,2,3,4)] %&gt;%
  group_by(motif_group,ratio) %&gt;%
  summarise_each(funs(sum))</code></pre>
<pre><code>Warning: `summarise_each()` was deprecated in dplyr 0.7.0.
ℹ Please use `across()` instead.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>Warning: `funs()` was deprecated in dplyr 0.8.0.
ℹ Please use a list of either functions or lambdas:

# Simple named list: list(mean = mean, median = median)

# Auto named with `tibble::lst()`: tibble::lst(mean, median)

# Using lambdas list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre class="r"><code>plot &lt;- plot[which(plot$ratio &lt;= 0.3),]

dat &lt;- plot
ratios &lt;- dat %&gt;%
    group_by(motif_group) %&gt;%
    summarize(min=min(ratio),
              max=max(ratio))
newx &lt;- data.frame(ratio=seq(0, 0.3, by=0.001))
smoother &lt;- function(df, span=3/4, newx){
    fit &lt;- loess(freq~ratio, df, span=span)
    pred &lt;- predict(fit, newdata=newx)
    pred &lt;- tibble(ratio=newx$ratio, freq=pred)
    pred
}
span &lt;- 1/10
dat2 &lt;- dat %&gt;%
    filter(ratio &lt;= 0.3) %&gt;%
    group_by(motif_group) %&gt;%
    nest()
dat2$smoothed &lt;- map(dat2$data, smoother, newx=newx, span=span)
dat3 &lt;- unnest(dat2, &quot;smoothed&quot;)

plot &lt;- dat3
dat4 &lt;- dat3 %&gt;%
    ungroup() %&gt;%
    group_by(ratio) %&gt;%
    mutate(freq=freq/sum(freq)) %&gt;%
    ungroup() %&gt;%
    group_by(motif_group)
plot&lt;- dat4
dat5 &lt;- dat4 %&gt;%
    ungroup() %&gt;%
    mutate(motif_group=fct_rev(motif_group))
plot &lt;- dat5
dat6 &lt;- dat5 %&gt;%
    filter(motif_group != &quot;rest motifs&quot;) %&gt;%
    mutate(motif_group=droplevels(motif_group))
dat.list &lt;- dat6 %&gt;%
    mutate(ymin=0, ymax=freq) %&gt;%
    dplyr::select(-data) %&gt;%
    group_by(motif_group) %&gt;%
    nest()
names(dat.list$data) &lt;- dat.list$motif_group
## for T|CC, the minimum is the frequency of A|CC
##           the maximum is the frequency of T|CC + frequency of A|CC
tmp &lt;- dat.list$data[[&quot;T | CC&quot;]] %&gt;%
    mutate(ymin=dat.list$data[[&quot;A | CC&quot;]]$ymax,
           ymax=freq + ymin)
dat.list$data[[&quot;T | CC&quot;]] &lt;- tmp
## for A|CG, the minimum is the ymax of T|CC
##           the maximum is the frequency of A|CG + ymin
tmp &lt;- dat.list$data[[&quot;A | CG&quot;]] %&gt;%
    mutate(ymin=dat.list$data[[&quot;T | CC&quot;]]$ymax,
           ymax=freq + ymin)
dat.list$data[[&quot;A | CG&quot;]] &lt;- tmp
tmp &lt;- dat.list$data[[&quot;T | CG&quot;]] %&gt;%
    mutate(ymin=dat.list$data[[&quot;A | CG&quot;]]$ymax,
           ymax=freq + ymin)
dat.list$data[[&quot;T | CG&quot;]] &lt;- tmp
dat7 &lt;- dat.list %&gt;%
    unnest(&quot;data&quot;)
ylabel &lt;- &quot;Probability of end motif sequence&quot;
plot1B &lt;- dat7 %&gt;%
     ggplot(aes(x = ratio, y=freq, fill = motif_group)) +
     geom_ribbon(aes(ymin=ymin, ymax=ymax, fill=motif_group)) +
     scale_fill_manual(&quot;&quot;, values = c(&quot;A | CC&quot; = &quot;#77C8DD&quot;, &quot;T | CC&quot; = &quot;#004766&quot;, &quot;A | CG&quot; = &quot;#77dd77&quot;, &quot;T | CG&quot; = &quot;#00660a&quot;, &quot;rest motifs&quot; = &quot;white&quot;), breaks = c(&quot;T | CG&quot;, &quot;A | CG&quot;, &quot;T | CC&quot;, &quot;A | CC&quot;)) +
     ylab(ylabel) +
     xlab(&quot;Preferrence of position for cfDNA ends (%)&quot;) +
     theme_classic(base_size=20) +
     theme(text = element_text(size=16),
           axis.title.y = element_text(margin = margin(r = 20, l = 0)),
           axis.title.x = element_text(margin = margin(t = 45, b = 0)),
           legend.position = c(0.2,0.8),
           legend.title = element_blank(),
           plot.margin = unit(c(1,1,1,1), &quot;cm&quot;)) +
     scale_y_continuous(limits=c(0, 1)) +
     scale_x_continuous(labels = scales::percent, limits = c(0,0.15), expand = c(0, 0.005))

#ggsave(here(&quot;docs/figure/temp_cfdna_ratio_motif.jpg&quot;), plot = plot1B, device = &quot;jpeg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=300, scale = .8)</code></pre>
</div>
<div id="figure-1c" class="section level2">
<h2>Figure 1C</h2>
<pre class="r"><code>file &lt;- &quot;data/positons_conserved_200over_5perc.rds&quot;
meth_file &lt;- &quot;data/Moss_ratios.rds&quot;
t &lt;- readRDS(here(file))
m &lt;- readRDS(here(meth_file))
t$perc &lt;- 0
t[which(t$pos == &quot;start&quot;),]$perc &lt;- t[which(t$pos == &quot;start&quot;),]$start_perc
t[which(t$pos == &quot;end&quot;),]$perc &lt;- t[which(t$pos == &quot;end&quot;),]$end_perc
tgr &lt;- GRanges(paste0(t$seqnames, &quot;:&quot;, t$location))
tgr$atcc &lt;- 0
tgr$atcg &lt;- 0
tgr[c(which((t$pos == &quot;start&quot;) &amp; (t$motif %in% c(&quot;ACC&quot;, &quot;TCC&quot;))), which((t$pos == &quot;end&quot;) &amp; (t$motif %in% c(&quot;GGT&quot;, &quot;GGA&quot;))))]$atcc &lt;- 1
tgr[c(which((t$pos == &quot;start&quot;) &amp; (t$motif %in% c(&quot;ACG&quot;, &quot;TCG&quot;))), which((t$pos == &quot;end&quot;) &amp; (t$motif %in% c(&quot;CGT&quot;, &quot;CGA&quot;))))]$atcg &lt;- 1
m &lt;- m[,c(2,3,5)]
mgr &lt;- GRanges(paste0(m$chr, &quot;:&quot;, m$pos))
mgr$beta &lt;- m$beta

aoi &lt;- builtin_annotations()
aoi &lt;- aoi[grep(&quot;hg19&quot;, aoi)]
aoi &lt;- aoi[-c(grep(&quot;Nhlf&quot;, aoi), grep(&quot;Nhek&quot;, aoi), grep(&quot;Huvec&quot;, aoi), grep(&quot;Hsmm&quot;, aoi), grep(&quot;Gm12878&quot;, aoi), grep(&quot;H1hesc&quot;, aoi), grep(&quot;Hepg2&quot;, aoi), grep(&quot;Hmec&quot;, aoi))]
aoi &lt;- aoi[-which(aoi == &quot;hg19_lncrna_gencode&quot;)]
annotations = build_annotations(genome = &#39;hg19&#39;, annotations = aoi)</code></pre>
<pre><code>Building enhancers...</code></pre>
<pre><code>
Attaching package: &#39;XVector&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    compact</code></pre>
<pre><code>
Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    strsplit</code></pre>
<pre><code>Building hmms...</code></pre>
<pre><code>Downloading chromHMM track for K562</code></pre>
<pre><code>Loading required package: GenomicFeatures</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre><code></code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre><code>Building promoters...</code></pre>
<pre><code>Building 1to5kb upstream of TSS...</code></pre>
<pre><code>Building intergenic...</code></pre>
<pre><code>Building cds...</code></pre>
<pre><code>Building 5UTRs...</code></pre>
<pre><code>Building 3UTRs...</code></pre>
<pre><code>Building exons...</code></pre>
<pre><code>Building first exons...</code></pre>
<pre><code>Building introns...</code></pre>
<pre><code>Building intron exon boundaries...</code></pre>
<pre><code>Building exon intron boundaries...</code></pre>
<pre><code>snapshotDate(): 2022-10-31</code></pre>
<pre><code>Building CpG islands...</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>Building CpG shores...</code></pre>
<pre><code>Building CpG shelves...</code></pre>
<pre><code>Building inter-CpG-islands...</code></pre>
<pre class="r"><code>annotations &lt;- annotations[which(seqnames(annotations) %in% c(paste0(&quot;chr&quot;, c(1:22))))]
o &lt;- as_tibble(findOverlaps(mgr, annotations))
o$beta &lt;- mgr[o$queryHits]$beta
o &lt;- o %&gt;%
  group_by(subjectHits) %&gt;%
  summarise(beta = mean(beta))
annotations$beta &lt;- 2</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[o$subjectHits,]$beta &lt;- o$beta</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.

Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations_lowbeta &lt;- annotations[which(annotations$beta &lt;= 0.3),]
annotations_highbeta &lt;- annotations[which(annotations$beta &gt;= 0.7),]
annotations_lowbeta$type &lt;- paste0(annotations_lowbeta$type, &quot;_unmeth&quot;)
annotations_highbeta$type &lt;- paste0(annotations_highbeta$type, &quot;_meth&quot;)</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations$group2 &lt;- &quot;all&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations_lowbeta$group2 &lt;- &quot;unmethylated&quot;
annotations_highbeta$group2 &lt;- &quot;methylated&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations &lt;- c(annotations, annotations_lowbeta, annotations_highbeta)</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 2 out-of-bound ranges located on sequence
  chr17. Note that ranges located on a sequence whose length is unknown
  (NA) or on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations &lt;- annotations[which(annotations$group2 == &quot;all&quot;)]
annotations$group3 &lt;- &quot;NA&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_genes_intergenic&quot;)]$group3 &lt;- &quot;Intergenic&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_cpg_inter&quot;)]$group3 &lt;- &quot;CpG - open sea&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_cpg_islands&quot;)]$group3 &lt;- &quot;CpG - islands&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_genes_firstexons&quot;)]$group3 &lt;- &quot;First exon&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_genes_exons&quot;)]$group3 &lt;- &quot;Exons&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_genes_introns&quot;)]$group3 &lt;- &quot;Introns&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_genes_promoters&quot;)]$group3 &lt;- &quot;Promoters&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[which(annotations$type == &quot;hg19_enhancers_fantom&quot;)]$group3 &lt;- &quot;Enhancers&quot;</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>o &lt;- findOverlaps(tgr, annotations)
o &lt;- as_tibble(o)
ot &lt;- table(o$subjectHits)
annotations$recur &lt;- 0</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations[as.numeric(as.character(names(ot)))]$recur &lt;- as.numeric(ot)</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.

Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annotations$frequency &lt;- annotations$recur / width(annotations)</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>annot &lt;- as_tibble(annotations)
annot_sum &lt;- annot %&gt;%
  group_by(group3) %&gt;%
  summarise(frequency = mean(frequency))
annot_sum &lt;- annot_sum[order(-annot_sum$frequency),]
annot_sum$group3 &lt;- factor(annot_sum$group3, levels = c(annot_sum$group3))
annotations$group3 &lt;- factor(annotations$group3, levels = c(annot_sum$group3))</code></pre>
<pre><code>Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 1 out-of-bound range located on sequence chr17.
  Note that ranges located on a sequence whose length is unknown (NA) or
  on a circular sequence are not considered out-of-bound (use
  seqlengths() and isCircular() to get the lengths and circularity flags
  of the underlying sequences). You can use trim() to trim these ranges.
  See ?`trim,GenomicRanges-method` for more information.</code></pre>
<pre class="r"><code>plot1C &lt;- ggplot(annot_sum[which(annot_sum$group3 %in% c(&quot;First exon&quot;, &quot;CpG - islands&quot;, &quot;Promoters&quot;, &quot;Intergenic&quot;, &quot;CpG - open sea&quot;, &quot;Exons&quot;, &quot;Introns&quot;, &quot;Enhancers&quot;)),], aes(x=group3, y=frequency, fill=frequency)) +
  geom_bar(stat = &quot;identity&quot;) +
  theme_classic(base_size = 20) +
  theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=1)) +
  ylab(&quot;Frequency of preferred ends&quot;) +
  xlab(&quot;&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=1),
        text = element_text(size=16),
        legend.title = element_blank(),
        legend.position = &quot;none&quot;,
        axis.title.y = element_text(margin = margin(r = 20, l = 20)),
        axis.title.x = element_blank(),
        plot.margin = unit(c(2,1,0.5,1), &quot;cm&quot;),
        strip.background = element_blank()
        ) +
  scale_fill_gradient(low =  &quot;#004766&quot;, high = &quot;#77c8dd&quot;) + 
  scale_y_continuous(labels = scales::percent)

#ggsave(here(&quot;docs/figure/recurrent_pos_groups2.jpg&quot;), plot = plot1C, device = &quot;jpeg&quot;,width = 20, height=12,units = &quot;in&quot;, dpi=300)</code></pre>
</div>
<div id="figure-1-combine-figure-1a-figure-1b-figure-1c"
class="section level2">
<h2>Figure 1: combine Figure 1A, Figure 1B, Figure 1C</h2>
<pre class="r"><code>dir.create(here(&quot;docs/figure&quot;), showWarnings=FALSE)
plot_combi1BCD &lt;- plot_grid(plot1B, plot1C, NULL, ncol=3, rel_widths = c(1,1,1), labels=c(&quot;b&quot;, &quot;c&quot;, &quot;d&quot;), label_size = 20, align = &quot;v&quot;)
plot_combi1ABCD &lt;- plot_grid(plot1A, plot_combi1BCD, nrow=2, rel_heights = c(1,1), labels=c(&#39;a&#39;, &#39;&#39;), label_size = 20)
ggsave(here(&quot;docs/figure/Fig_1.jpg&quot;), plot = plot_combi1ABCD, device = &quot;jpeg&quot;,width =180, height=108, units = &quot;mm&quot;, dpi=600, bg = &quot;white&quot;, scale = 3)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] org.Hs.eg.db_3.16.0                    
 [2] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [3] GenomicFeatures_1.50.4                 
 [4] AnnotationDbi_1.60.2                   
 [5] Biobase_2.58.0                         
 [6] BSgenome.Hsapiens.UCSC.hg19_1.4.3      
 [7] BSgenome_1.66.3                        
 [8] rtracklayer_1.58.0                     
 [9] Biostrings_2.66.0                      
[10] XVector_0.38.0                         
[11] here_1.0.1                             
[12] GenomicRanges_1.50.2                   
[13] GenomeInfoDb_1.34.9                    
[14] IRanges_2.32.0                         
[15] S4Vectors_0.36.2                       
[16] BiocGenerics_0.44.0                    
[17] data.table_1.15.0                      
[18] cowplot_1.1.3                          
[19] annotatr_1.24.0                        
[20] lubridate_1.9.3                        
[21] forcats_1.0.0                          
[22] stringr_1.5.1                          
[23] dplyr_1.1.4                            
[24] purrr_1.0.2                            
[25] readr_2.1.5                            
[26] tidyr_1.3.1                            
[27] tibble_3.2.1                           
[28] ggplot2_3.4.4                          
[29] tidyverse_2.0.0                        
[30] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] AnnotationHub_3.6.0           BiocFileCache_2.11.1         
  [3] systemfonts_1.0.5             plyr_1.8.9                   
  [5] BiocParallel_1.32.6           digest_0.6.34                
  [7] htmltools_0.5.7               fansi_1.0.6                  
  [9] magrittr_2.0.3                memoise_2.0.1                
 [11] tzdb_0.4.0                    matrixStats_1.1.0            
 [13] vroom_1.6.5                   timechange_0.3.0             
 [15] prettyunits_1.2.0             colorspace_2.1-0             
 [17] blob_1.2.4                    rappdirs_0.3.3               
 [19] textshaping_0.3.7             xfun_0.42                    
 [21] callr_3.7.3                   crayon_1.5.2                 
 [23] RCurl_1.98-1.14               jsonlite_1.8.8               
 [25] glue_1.7.0                    gtable_0.3.4                 
 [27] zlibbioc_1.44.0               DelayedArray_0.24.0          
 [29] scales_1.3.0                  DBI_1.2.1                    
 [31] Rcpp_1.0.12                   xtable_1.8-4                 
 [33] progress_1.2.3                bit_4.0.5                    
 [35] httr_1.4.7                    ellipsis_0.3.2               
 [37] pkgconfig_2.0.3               XML_3.99-0.16.1              
 [39] farver_2.1.1                  sass_0.4.8                   
 [41] dbplyr_2.4.0                  utf8_1.2.4                   
 [43] tidyselect_1.2.0              labeling_0.4.3               
 [45] rlang_1.1.3                   reshape2_1.4.4               
 [47] later_1.3.2                   munsell_0.5.0                
 [49] BiocVersion_3.16.0            tools_4.2.3                  
 [51] cachem_1.0.8                  cli_3.6.2                    
 [53] generics_0.1.3                RSQLite_2.3.5                
 [55] evaluate_0.23                 fastmap_1.1.1                
 [57] yaml_2.3.8                    ragg_1.2.7                   
 [59] processx_3.8.3                knitr_1.45                   
 [61] bit64_4.0.5                   fs_1.6.3                     
 [63] KEGGREST_1.38.0               whisker_0.4.1                
 [65] mime_0.12                     xml2_1.3.6                   
 [67] biomaRt_2.54.1                compiler_4.2.3               
 [69] rstudioapi_0.15.0             filelock_1.0.3               
 [71] curl_5.2.0                    png_0.1-8                    
 [73] interactiveDisplayBase_1.36.0 bslib_0.6.1                  
 [75] stringi_1.8.3                 ps_1.7.6                     
 [77] lattice_0.22-5                Matrix_1.6-5                 
 [79] vctrs_0.6.5                   pillar_1.9.0                 
 [81] lifecycle_1.0.4               BiocManager_1.30.22          
 [83] jquerylib_0.1.4               bitops_1.0-7                 
 [85] httpuv_1.6.14                 R6_2.5.1                     
 [87] BiocIO_1.8.0                  promises_1.2.1               
 [89] codetools_0.2-19              SummarizedExperiment_1.28.0  
 [91] rprojroot_2.0.4               rjson_0.2.21                 
 [93] withr_3.0.0                   regioneR_1.30.0              
 [95] GenomicAlignments_1.34.1      Rsamtools_2.14.0             
 [97] GenomeInfoDbData_1.2.9        parallel_4.2.3               
 [99] hms_1.1.3                     grid_4.2.3                   
[101] rmarkdown_2.25                MatrixGenerics_1.10.0        
[103] git2r_0.33.0                  getPass_0.2-4                
[105] shiny_1.8.0                   restfulr_0.0.15              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
