<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">noe2024</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>noe2024/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024treec05a4efb44aa04659dc25aaa0437d53413f3f21etargetblankc05a4efa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/cancer-genomics/noe2024/tree/c05a4efb44aa04659dc25aaa0437d53413f3f21e" target="_blank">c05a4ef</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomcancergenomicsnoe2024treec05a4efb44aa04659dc25aaa0437d53413f3f21etargetblankc05a4efa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/cancer-genomics/noe2024/tree/c05a4efb44aa04659dc25aaa0437d53413f3f21e" target="_blank">c05a4ef</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/results/

Untracked files:
    Untracked:  analysis/Figure3E.Rmd
    Untracked:  analysis/Supplementary_Figure1.Rmd
    Untracked:  analysis/Supplementary_Figure10.Rmd
    Untracked:  analysis/Supplementary_Figure2.Rmd
    Untracked:  analysis/Supplementary_Figure3.Rmd
    Untracked:  analysis/Supplementary_Figure4.Rmd
    Untracked:  analysis/Supplementary_Figure7.Rmd
    Untracked:  code/Expand_Fig4C_Rmd.R
    Untracked:  code/Pre_Figure3.rmd
    Untracked:  code/Pre_Supplementary_Figure3.rmd
    Untracked:  code/Pre_Supplementary_Figure4.rmd
    Untracked:  code/SuppTable1.Rmd
    Untracked:  code/SuppTable2.Rmd
    Untracked:  code/SuppTable4.Rmd
    Untracked:  code/SuppTable5.Rmd
    Untracked:  code/SuppTable6.Rmd
    Untracked:  code/get_Moss_et_al.R
    Untracked:  code/todos/
    Untracked:  data/Moss_et_al-NatComm-2018/
    Untracked:  data/diff_startfrags_3bp.rds
    Untracked:  data/diff_startfrags_4bp.rds
    Untracked:  data/healthy_cpggroups.rds
    Untracked:  data/motifs_cfdna_144_167.rds
    Untracked:  data/selected_lucas_files.rds
    Untracked:  data/selected_validation_files.rds
    Untracked:  output/Fig_4C_all_folds.pdf
    Untracked:  output/fold01_ens.csv
    Untracked:  output/fold02_ens.csv
    Untracked:  output/fold03_ens.csv
    Untracked:  output/fold04_ens.csv
    Untracked:  output/fold05_ens.csv
    Untracked:  output/fold06_ens.csv
    Untracked:  output/fold07_ens.csv
    Untracked:  output/fold08_ens.csv
    Untracked:  output/fold09_ens.csv
    Untracked:  output/supplementary_tables/

Unstaged changes:
    Modified:   analysis/Figure1.Rmd
    Modified:   analysis/Figure2.Rmd
    Modified:   analysis/Figure3.Rmd
    Modified:   analysis/Figure4.Rmd
    Modified:   analysis/Figure4C.Rmd
    Modified:   analysis/SessionInfo.Rmd
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Deleted:    code/F3e.Rmd
    Deleted:    code/SF10.Rmd
    Deleted:    code/SF7.Rmd
    Modified:   output/All_Scores.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/de3561665de4eb4a67334ac71cee69954c0f14f5/analysis/index.Rmd" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/cancer-genomics/noe2024/de3561665de4eb4a67334ac71cee69954c0f14f5/docs/index.html" target="_blank">de35616</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-30
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/cancer-genomics/noe2024/blob/c3db0ae18fa99d0941882439a942d9ccd70f5db0/analysis/index.Rmd" target="_blank">c3db0ae</a>
</td>
<td>
Shashikant Koul
</td>
<td>
2024-01-29
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="citation" class="section level1">
<h1>Citation</h1>
<p>Michaël Noë, Akshaya V. Annapragada, Zacharia H. Foda, Jamie Medina,
Dimitrios Mathios, Stephen Cristiano, Christopher Cherry, Daniel C.
Bruhm, Noushin Niknafs, Vilmos Adleff, Leonardo Ferreira, Hari Easwaran,
Stephen Baylin, Jillian Phallen, Robert B. Scharpf, Victor E.
Velculescu, DNA methylation and gene expression as factors of
genome-wide cell-free DNA fragmentation.</p>
</div>
<div id="abstract" class="section level1">
<h1>Abstract</h1>
<p>Circulating cell-free DNA (cfDNA) is emerging as a diagnostic avenue
for cancer detection, but the characteristics and origins of cfDNA
fragmentation in the blood are poorly understood. We evaluated the
effect of DNA methylation and gene expression on naturally occurring
genome- wide cfDNA fragmentation through analysis of plasma from 969
individuals, including 182 with cancer. cfDNA fragment ends occurring at
preferred locations genome-wide more frequently contained CCs or CGs,
and fragments ending with CGs or CCGs were enriched or depleted,
respectively, at methylated CpG positions, consistent with structural
models showing increased interaction of CG fragment ends with
nucleosomes. CpG methylation and reduced gene expression were
independently associated with higher levels and larger sizes of cfDNA
fragments, and reflected differences in cfDNA fragmentation in
tissue-specific pathways. Tumor-related hypomethylation and increased
gene expression were associated with global decrease in cfDNA fragment
size that may explain the overall smaller cfDNA fragments observed in
human cancers. Cancer-specific methylation at CpGs of pancreatic cancer
patients was associated with genome-wide changes in cfDNA fragment ends
in patients with cancers. These results provide a connection between
epigenetic changes and cfDNA fragmentation that may have implications
for disease detection.</p>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The analyses in this repository use data from previous studies
(Christiano et al., Nature, 2019 and Mathios et al., Nature
Communications, 2021). The data for the samples analyzed in these
studies was deposited at the European Genome-Phenome Archive (EGA). The
GitHub-repositories for these studies explain how the data was analysed
until a GenomicRanges (GRanges) object was made with the fragment start-
and end-positions.</p>
<p>For the analyses in this repository, we start with the GRanges files,
which contain cell-free DNA fragment information.</p>
</div>
<div id="requirements" class="section level1">
<h1>Requirements</h1>
<p>The R-packages used to process the samples and to construct the
figures are provided <a href="../docs/SessionInfo.html">here</a>. These
are available on Bioconductor or CRAN.</p>
</div>
<div id="figures" class="section level1">
<h1>Figures</h1>
<ul>
<li>Figure 1: <a href="../docs/Figure1.html">HTML</a>, <a
href="../analysis/Figure1.Rmd">Code</a>, <a
href="../docs/figure/Fig_1.jpg">JPG</a></li>
<li>Figure 2: <a href="../docs/Figure2.html">HTML</a>, <a
href="../analysis/Figure2.Rmd">Code</a>, <a
href="../docs/figure/Fig_2.jpg">JPG</a></li>
<li>Figure 3: <a href="../docs/Figure3.html">HTML</a>, <a
href="../analysis/Figure3.Rmd">Code</a>
<ul>
<li>A,B: <a href="../docs/figure/Fig_3AB.jpg">JPG</a></li>
<li>C,D: <a href="../docs/figure/Fig_3CD.jpg">JPG</a></li>
<li>E: <a href="../docs/Figure3E.html">HTML</a>, <a
href="../analysis/Figure3E.Rmd">Code</a>, <a
href="../docs/figure/Fig_3E_multivar.jpg">JPG</a></li>
<li>F: <a href="../docs/figure/Fig_3F.jpg">JPG</a><br />
</li>
</ul></li>
<li>Figure 4:
<ul>
<li>A,B,C: <a href="../docs/Figure4.html">HTML</a>, <a
href="../analysis/Figure4.Rmd">Code</a>, <a
href="../docs/figure/Fig_4.jpg">JPG</a></li>
<li>C: <a href="../docs/Figure4C.html">HTML</a>, <a
href="../analysis/Figure4C.Rmd">Code</a>, <a
href="../docs/figure/Fig_4C.jpg">JPG</a></li>
</ul></li>
</ul>
</div>
<div id="supplementary-figures" class="section level1">
<h1>Supplementary Figures</h1>
<ul>
<li>Supp. Fig. 1: <a href="../docs/Supplementary_Figure1.html">HTML</a>,
<a href="../analysis/Supplementary_Figure1.Rmd">Code</a>, <a
href="../docs/figure/Supplementary_Fig_1.jpg">JPG</a></li>
<li>Supp. Fig. 2: <a href="../docs/Supplementary_Figure2.html">HTML</a>,
<a href="../analysis/Supplementary_Figure2.Rmd">Code</a>, <a
href="../docs/figure/Supplementary_Fig_2.jpg">JPG</a></li>
<li>Supp. Fig. 3: <a href="../docs/Supplementary_Figure3.html">HTML</a>,
<a href="../analysis/Supplementary_Figure3.Rmd">Code</a>, <a
href="../docs/figure/Supplementary_Fig_3.jpg">JPG</a></li>
<li>Supp. Fig. 4: <a href="../docs/Supplementary_Figure4.html">HTML</a>,
<a href="../analysis/Supplementary_Figure4.Rmd">Code</a>, <a
href="../docs/figure/Supplementary_Fig_4.jpg">JPG</a></li>
<li>Supp. Fig. 5: <a href="">HTML</a>, <a href="">Code</a>, <a
href="">JPG</a> Empty</li>
<li>Supp. Fig. 6: <a href="">HTML</a>, <a href="">Code</a>, <a
href="">JPG</a> Empty</li>
<li>Supp. Fig. 7: <a href="../docs/Supplementary_Figure7.html">HTML</a>,
<a href="../analysis/Supplementary_Figure7.Rmd">Code</a></li>
<li>Supp. Fig. 8: <a href="">HTML</a>, <a href="">Code</a>, <a
href="">JPG</a> Empty</li>
<li>Supp. Fig. 9: <a href="">HTML</a>, <a href="">Code</a>, <a
href="">JPG</a> Empty</li>
<li>Supp. Fig. 10: <a
href="../docs/Supplementary_Figure10.html">HTML</a>, <a
href="../analysis/Supplementary_Figure10.Rmd">Code</a></li>
</ul>
</div>
<div id="supplementary-tables" class="section level1">
<h1>Supplementary Tables</h1>
<ul>
<li>Supp. Table 1: Refer to supplementary data in Christiano et al.,
Nature, 2019 and Mathios et al., Nature Communications, 2021</li>
<li>Supp. Table 2: <a href="../code/SuppTable2.Rmd">Code</a></li>
<li>Supp. Table 3: This table was created manually. Refer to the Methods
section of this paper.</li>
<li>Supp. Table 4: <a href="../code/SuppTable4.Rmd">Code</a> Need
GRanges to run.</li>
<li>Supp. Table 5: <a href="../code/SuppTable5.Rmd">Code</a> Need Rds
files from the previous step to run.</li>
<li>Supp. Table 6: <a href="../code/SuppTable6.Rmd">Code</a> Need chrX
fragments data to run.</li>
<li>Supp. Table 7: <a href="">Code</a> Empty</li>
<li>Supp. Table 8: <a href="">Code</a> Empty</li>
<li>Supp. Table 9: <a href="">Code</a> Empty</li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
