### Required Libraries
```{r}
library(tidyverse)
library(fgsea)
library(ggplot2)
library(ggpubr)
library(data.table)
library(ComplexHeatmap)
library(here)
```

### Custom functions read in here. See .Rmd for details.
```{r, include=F}
# Make custom GSEA plot
custom_enrichment <- function(ranks, genes, ylims, title=''){
    library(ggplot2)
    prep_df <- function(peaks, ids){
        tar_ranges <- 2:(length(peaks$tops) - 1)
        x <- rep(ids, each=2)
        y <- rbind(peaks$bots[tar_ranges], peaks$top[tar_ranges]) %>%
            c()
        data.frame(x=c(0, length(ranks), x), y=c(0, 0, y))
    }
    
    # Prep data
    ranks <- sort(ranks, decreasing=T)
    ids <- genes %>%
        intersect(names(ranks)) %>%
        match(names(ranks)) %>%
        sort()
    peaks <- custom_get_peaks(ranks, ids)
    df <- prep_df(peaks, ids)

    # Generate plot with real data
    p <- ggplot(df, aes(x=x, y=y)) + 
        geom_line(color="dark blue") + 
        geom_hline(yintercept=max(df$y), colour="red", linetype="dashed") + 
        geom_hline(yintercept=min(df$y), colour="red", linetype="dashed") + 
        geom_hline(yintercept=0, colour="black") + 
        theme_classic(base_size=20) +
        geom_segment(data=df[-1:-2, ], mapping=aes(x=x, y=ylims[1]+.01, xend=x, yend=ylims[1]+0.05), size=0.2) +
        theme(plot.title = element_text(size=15)) +
        labs(x="Gene Rank", y="Enrichment Score", title=title) + 
        ylim(ylims[1], ylims[2])

    # Add lines for random permutations
    for(i in 1:100){
        new_ids <- sample(1:length(ranks), length(ids)) %>%
            sort()
        peaks <- custom_get_peaks(ranks, new_ids)
        df <- prep_df(peaks, ids)
        p$layers <- c(geom_line(data=df, color='light grey', size=.1), p$layers)
    }
    p
}

custom_calc_gsea_stats <- function(ranks, ids, nperm=0){
    # Get peaks and leading edge genes
    peaks <- custom_get_peaks(ranks, ids)
    if(peaks$stat > 0){
        mid <- which(peaks$tops == peaks$stat) %>%
            na.omit() %>%
            head(1)
        le <- names(peaks$tops)[1:mid] %>%
            na.omit()
    } else {
        mid <- which(peaks$bots == peaks$stat) %>%
            na.omit() %>%
            tail(1)
        le <- names(peaks$bots)[length(peaks$bots):mid] %>%
            na.omit()
    }
    outs <- list(Leading.Edge=le, Enrichment.Score=peaks$stat)

    # Calculate pval if nperm defined
    if(nperm != 0){
        stats <- sapply(1:nperm, function(x){
            new_ids <- sample(1:length(ranks), length(ids)) %>%
                sort()
            custom_get_peaks(ranks, new_ids)$stat
        })
        n_greater <- sum(abs(stats) > abs(peaks$stat))
        if(n_greater == 0) n_greater <- 1
        outs$P <- n_greater / nperm
    }
    outs
}

custom_get_peaks <- function(ranks, ids){
    # Sort ids
    ids <- sort(ids)

    # Get correct step size for enrichment scores
    step_up <- 1 / length(ids)
    step_dn <- 1 / (length(ranks))

    # Calculate enrichment scores before and after each hit
    tops <- 0
    bots <- 0
    prev_id <- 1
    for(id in ids){
        bots <- c(bots, tail(tops, 1) + step_dn * (prev_id - id))
        tops <- c(tops, tail(bots, 1) + step_up)
        prev_id <- id
    }
    tops <- c(tops, 0)
    bots <- c(bots, 0)
    
    # Calc stat as the min/max of bot/top
    names(tops) <- names(bots) <- c(NA, names(ranks)[ids], NA)
    stat <- ifelse(abs(min(bots)) > max(tops), min(bots), max(tops))

    # Return
    list(tops=tops, bots=bots, stat=stat)
}

custom_gsea <- function(ranks, sets){
    # Prep to run
    df <- c()
    ranks <- sort(ranks, decreasing=T)

    # Step through each set
    for(set in names(sets)){
        # Get index of genes in pathway
        ids <- sets[[set]] %>%
            intersect(names(ranks)) %>%
            match(names(ranks)) %>%
            sort()

        # Calc relevant GSEA info and add to data frame
        gsea <- custom_calc_gsea_stats(ranks, ids, 1e4)
        df <- rbind(df, tibble(Gene.Set=set, P.val=gsea$P, 
            Enrichment.Score=gsea$Enrichment.Score, 
            Leading.Edge=list(gsea$Leading.Edge)))
    }

    # Finalize analyses and return
    df$P.adj <- p.adjust(df$P.val, 'BH')
    df
}

# Make GSEA plots for all ranks for given pathways
compare_gsea_plots <- function(path, sets, all_ranks, ylims){
    plots <- lapply(1:length(all_ranks), function(i){
        custom_enrichment(all_ranks[[i]], sets[[path]], ylims,
            paste(names(all_ranks)[i]))
    })
    title <- gsub('_', ' ', path, fixed=T) %>%
        str_to_title()
    ggarrange(plotlist=plots, ncol=2, nrow=2) %>%
        annotate_figure(top=text_grob(title, size=20)) %>%
        print()
    return()
}
```

### Generate GSEA Enrichment plots
```{r}
# Read in analysis data
# GSEA results for each modality
comp_table <- read.table(here('data/giraffe/gsea_table.csv'), sep=',', header=T)

# Gene ranks calculated by modality
all_ranks <- readRDS(here('data/giraffe/ranks.RDS'))

# Read in gene sets
sets <- c(
    gmtPathways(here('data/giraffe/gene_sets/c2.cp.kegg.v7.5.1.symbols.gmt')),
    gmtPathways(here('data/giraffe/gene_sets/h.all.v7.5.1.symbols.gmt'))
)
len <- sapply(sets, length)
sets <- sets[len > 150]

# Plot target gene set
comp_table <- arrange(comp_table, CpG.Coverage.P.val)
i <- 1
ymax <- max(comp_table[i, c(3, 6, 9, 12)])
ymin <- min(comp_table[i, c(3, 6, 9, 12)])
compare_gsea_plots(comp_table$Gene.Set[i], sets, all_ranks, c(ymin-.1, ymax))
```

### Create GSEA Heatmap
```{r, fig.height=10}
tab <- fread(here('data/giraffe/gsea_table.csv')) %>%
    filter(RNA.Expression.P.val < .1 & Methylation.P.val < .1 &
        TSS.Coverage.P.val < .1 & CpG.Coverage.P.val < .1) %>%
    select(Gene.Set, 'WBC RNA expression'='RNA.Expression.Enrichment.Score', 
        'WBC DNA Methylation'='Methylation.Enrichment.Score',
        'cfDNA TSS coverage'='TSS.Coverage.Enrichment.Score',
        'cfDNA CpG coverage'='CpG.Coverage.Enrichment.Score') %>%
    data.frame() %>%
    mutate(Gene.Set=str_to_title(gsub('_', ' ', Gene.Set))) %>%
    mutate(Gene.Set=gsub('Hallmark.|Kegg.', '', Gene.Set)) %>%
    column_to_rownames('Gene.Set') %>%
    arrange(cfDNA.CpG.coverage) %>%
    data.matrix()
colnames(tab) <- gsub('.', ' ', colnames(tab), fixed=T)
tab <- tab[!rownames(tab) %in% "Kras Signaling Up", ]

rename <- c(
    'Fatty Acid Metabolism'='Fatty Acid Met',
    'Oxidative Phosphorylation'='Ox Phos',
    'Kras Signaling Dn'='Kras Sig Dn',
    'Calcium Signaling Pathway'='Calcium Signaling',
    'Neuroactive Ligand Receptor Interaction'='Neuro LR Interaction'
)
mid <- match(names(rename), rownames(tab))
rownames(tab)[mid] <- unname(rename)
tab <- tab[order(tab[, 1], decreasing=T), ]

Heatmap(tab, cluster_rows=F, cluster_columns=F, 
    rect_gp=gpar(col='white', lwd=2),
    column_title_gp=gpar(fontsize=16, fontface='bold'),
    name='GSEA Enrichment Score',
    row_names_side='left',
    height=nrow(tab) * unit(0.3, 'inches'), width=ncol(tab) * unit(0.3, 'inches'))
```

### Run linear model fitting RNA and methylation to coverage.
```{r, fig.height=4}
# Read in table methylation, expression, and coverage
tab <- readRDS(here('data/giraffe/table.rds')) %>%
  column_to_rownames('transcript') %>%
  mutate(rna_expression=log10(rna_expression + 1), coverage=log10(coverage + 1)) %>%
  mutate(rna_expression=scale(rna_expression), beta=scale(beta)) %>%
  filter(!is.na(beta))

# Fit linear model
model <- lm(coverage ~ beta * rna_expression, tab)

# Get confidence intervals for coefficients
ci_rna <- confint(model, 'rna_expression', level=0.95)
ci_beta <- confint(model, 'beta', level=0.95)
df <- data.frame(Variable=c('RNA', 'Methylation'), 
  min_ci=c(ci_rna[1], ci_beta[1]),
  max_ci=c(ci_rna[2], ci_beta[2]),
  mean=c(model$coefficients['rna_expression'], model$coefficients['beta']))

# Plot CIs.
ggplot(df, aes(y=Variable, color=Variable)) +
  geom_point(aes(x=mean), size=2) +
  geom_errorbar(aes(xmin=min_ci, xmax=max_ci, width=0.4)) +
  geom_vline(xintercept=0, linetype='dashed') +
  theme_classic() +
  theme(legend.position='none') +
  ylab('') +
  xlab('Regression Coefficient') +
  xlim(-0.2, 0.05)
```

### Repeat linear model fitting to fragment size.
```{r, fig.height=4}
# Read in table methylation, expression, and sizes
tab <- readRDS(here('data/giraffe/methylation_expression_size.rds')) %>%
  column_to_rownames('transcript') %>%
  mutate(rna_expression=log10(rna_expression + 1)) %>%
  mutate(rna_expression=scale(rna_expression), beta=scale(beta)) %>%
  filter(!is.na(beta))

# Fit linear model
model <- lm(size ~ beta * rna_expression, tab)

# Get confidence intervals for coefficients
ci_rna <- confint(model, 'rna_expression', level=0.95)
ci_beta <- confint(model, 'beta', level=0.95)
df <- data.frame(Variable=c('RNA', 'Methylation'), 
  min_ci=c(ci_rna[1], ci_beta[1]),
  max_ci=c(ci_rna[2], ci_beta[2]),
  mean=c(model$coefficients['rna_expression'], model$coefficients['beta']))

# Plot CIs
ggplot(df, aes(y=Variable, color=Variable)) +
  geom_point(aes(x=mean), size=2) +
  geom_errorbar(aes(xmin=min_ci, xmax=max_ci, width=0.4)) +
  geom_vline(xintercept=0, linetype='dashed') +
  theme_classic() +
  theme(legend.position='none') +
  ylab('') +
  xlab('Regression Coefficient') +
  xlim(-0.3, 0.2)
```
