### Required Libraries
```{r}
library(SummarizedExperiment)
library(ggplot2)
library(ggpubr)
library(GenomicRanges)
library(here)
```

### Read in and find overlap of RNA and Coverage data.
NOTE: gene_conv.RDS is a saved output from biomaRt.
```{r}
# Read in WB data
wb_expr <- readRDS(here('data/giraffe/gtex_wb_counts.RDS'))
wb_expr <- apply(wb_expr, 2, function(x){
  x*1e6/sum(x)
})

# Read in healthy coverage data for TSS
cov_se <- cbind(
  readRDS(here('data/giraffe/lucas_healthy_tss_bins.RDS'))
)

# Get HGNC conversions
conv <- readRDS(here('data/giraffe/gene_conv.RDS'))
rowRanges(cov_se)$hgnc <- conv$hgnc[match(rowRanges(cov_se)$gene, conv$hs.ens)]
ol <- intersect(rowRanges(cov_se)$hgnc, rownames(wb_expr))
cov_mat <- assay(cov_se)[match(ol, rowRanges(cov_se)$hgnc), ]
cov_mat <- apply(cov_mat, 2, function(x){
  x*1e6/sum(x)
})
rownames(cov_mat) <- ol

# Subset to overlap and get ave expr and cov
wb_expr <- rowMeans(wb_expr[ol,])
cov <- rowMeans(cov_mat[ol,])
```


### Plot coverage vs. RNA expression
```{r}
df <- data.frame(cbind('Coverage'=log2(cov), 'Expression'=log2(wb_expr)))
ggplot(df, aes(x=Coverage, y=Expression)) +
  geom_point(size=0.5, color='grey') +
  geom_density2d(color='black') +
  theme_classic(base_size=20) +
  xlab("Mean Coverage (Log2 CPM)") +
  ylab("Mean Expression (Log2 CPM)") +
  stat_cor(size=6) +
  xlim(2, 8)
```

### Read in coverage and methylation data and find overlap.
```{r}
# Read in methylation
all_met <- readRDS(here('data/giraffe/methylation_healthy.rds')) %>%
  mutate(start=pos, end=pos)
beta_mat <- all_met[, 4:11] %>% data.matrix()

# Read in healthy coverage data for TSS
cov_se <- readRDS(here('data/giraffe/lucas_healthy_tss_bins.RDS'))

# Find overlapping regions with methylation data
met_granges <- makeGRangesFromDataFrame(all_met)
ol <- findOverlaps(met_granges, rowRanges(cov_se))

# Subset to overlap and get ave expr and cov
met_betas <- rowMeans(beta_mat[queryHits(ol),])
cov <- assay(cov_se)[subjectHits(ol), ] %>%
  rowMeans()
```


### Plot coverage vs. RNA expression
```{r}
df <- data.frame(cbind('Coverage'=cov, 'Beta'=met_betas))
df$Rank_Coverage <- order(df$Coverage)
df$Rank_Beta <- order(df$Beta)

breaks <- c(.003, .005, .007, .01, .02, .04, .08, .15)
ggplot(df, aes(x=Coverage, y=Beta)) +
  geom_point(size=0.1, color='grey') +
  geom_density2d(color='black', breaks=breaks) +
  theme_classic(base_size=20) +
  xlab("Mean Coverage (CPM)") +
  ylab("Mean Beta") +
  stat_cor(size=6) +
  xlim(0, 210) +
  scale_y_continuous(breaks=c(0, .25, .5, .75, 1), limits=c(0, 1.06))
dev.off()
```